<!DOCTYPE html>

<html lang="en" data-content_root="../">
  <head>
    <meta charset="utf-8" />
    <meta name="viewport" content="width=device-width, initial-scale=1.0" /><meta name="viewport" content="width=device-width, initial-scale=1" />

    <title>pinder.core.index package &#8212; pinder 0.1.dev1+g8ad1ead documentation</title>
    <link rel="stylesheet" type="text/css" href="../_static/pygments.css?v=fa44fd50" />
    <link rel="stylesheet" type="text/css" href="../_static/basic_mod.css?v=59943e95" />
    <link rel="stylesheet" type="text/css" href="../_static/mystnb.4510f1fc1dee50b3e5859aac5469c37c29e427902b24a333a5f9fcb2f0b3ac41.css" />
    <link rel="stylesheet" type="text/css" href="../_static/custom.css?v=28804041" />
    <script src="../_static/documentation_options.js?v=40280a1a"></script>
    <script src="../_static/doctools.js?v=9a2dae69"></script>
    <script src="../_static/sphinx_highlight.js?v=dc90522c"></script>
    <script src="https://cdnjs.cloudflare.com/ajax/libs/require.js/2.3.4/require.min.js"></script>
    <script src="../_static/js/theme.js"></script>
    <script src="../_static/js/petite-vue.js"></script>
    <link rel="icon" href="../_static/favicon.ico"/>
    <link rel="index" title="Index" href="../genindex.html" />
    <link rel="search" title="Search" href="../search.html" />
    <link rel="next" title="pinder.core.loader package" href="pinder.core.loader.html" />
    <link rel="prev" title="pinder.core package" href="pinder.core.html" /> 
  </head><body data-dark_mode_code_blocks="true">

<div id="top_nav">
    

    <nav>
        
            
        

        <p id="toggle_sidebar">
            <a href="#" title="Toggle sidebar">|||</a>
        </p>
        <h1><a href="../index.html" title="Go to homepage"><img src="../_static/pinder.png"/></a></h1>
            <a id="source_link" href="https://github.com/pinder-org/pinder/">
    
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 496 512">
            <path fill="white" d="M 244.8,8 C 106.1,8 0,113.3 0,252 c 0,110.9 69.8,205.8 169.5,239.2 12.8,2.3 17.3,-5.6 17.3,-12.1 0,-6.2 -0.3,-40.4 -0.3,-61.4 0,0 -70,15 -84.7,-29.8 0,0 -11.4,-29.1 -27.8,-36.6 0,0 -22.9,-15.7 1.6,-15.4 0,0 24.9,2 38.6,25.8 21.9,38.6 58.6,27.5 72.9,20.9 2.3,-16 8.8,-27.1 16,-33.7 -55.9,-6.2 -112.3,-14.3 -112.3,-110.5 0,-27.5 7.6,-41.3 23.6,-58.9 -2.6,-6.5 -11.1,-33.3 2.6,-67.9 20.9,-6.5 69,27 69,27 20,-5.6 41.5,-8.5 62.8,-8.5 21.3,0 42.8,2.9 62.8,8.5 0,0 48.1,-33.6 69,-27 13.7,34.7 5.2,61.4 2.6,67.9 16,17.7 25.8,31.5 25.8,58.9 0,96.5 -58.9,104.2 -114.8,110.5 9.2,7.9 17,22.9 17,46.4 0,33.7 -0.3,75.4 -0.3,83.6 0,6.5 4.6,14.4 17.3,12.1 C 428.2,457.8 496,362.9 496,252 496,113.3 383.5,8 244.8,8 Z"/>
        </svg>
    
</a>
        

        <a id="mode_toggle" href="#" @click.prevent="handleClick" :title="mode">
    <template v-if="mode == 'light'">
        <svg width="100%" height="100%" viewBox="0 0 79 80" version="1.1" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" xml:space="preserve" style="fill-rule:evenodd;clip-rule:evenodd;stroke-linejoin:round;stroke-miterlimit:2;"><g id="mode_light"><rect id="Bounds" x="0" y="-0" width="78.623" height="79.049" style="fill:none;"/><circle cx="39.311" cy="39.524" r="15.734" style="fill:#fff;"/><g id="beams"><g id="beam"><path id="beam1" serif:id="beam" d="M44.212,4.901c0,-2.705 -2.196,-4.901 -4.901,-4.901c-2.704,-0 -4.9,2.196 -4.9,4.901l-0,9.614c-0,2.705 2.196,4.901 4.9,4.901c2.705,0 4.901,-2.196 4.901,-4.901l0,-9.614Z" style="fill:#fff;"/></g><g id="beam2" serif:id="beam"><path id="beam3" serif:id="beam" d="M67.48,18.073c1.913,-1.912 1.913,-5.018 0,-6.931c-1.912,-1.912 -5.018,-1.912 -6.931,0l-6.798,6.799c-1.912,1.912 -1.912,5.018 0,6.931c1.913,1.912 5.018,1.912 6.931,-0l6.798,-6.799Z" style="fill:#fff;"/></g><g id="beam4" serif:id="beam"><path id="beam5" serif:id="beam" d="M25.728,61.108c1.912,-1.913 1.912,-5.018 -0,-6.931c-1.913,-1.913 -5.019,-1.913 -6.931,-0l-6.799,6.798c-1.912,1.913 -1.912,5.019 0,6.931c1.913,1.913 5.019,1.913 6.931,0l6.799,-6.798Z" style="fill:#fff;"/></g><g id="beam6" serif:id="beam"><path id="beam7" serif:id="beam" d="M60.682,54.177c-1.913,-1.913 -5.018,-1.913 -6.931,-0c-1.912,1.913 -1.912,5.018 0,6.931l6.798,6.798c1.913,1.913 5.019,1.913 6.931,0c1.913,-1.912 1.913,-5.018 0,-6.931l-6.798,-6.798Z" style="fill:#fff;"/></g><g id="beam8" serif:id="beam"><path id="beam9" serif:id="beam" d="M4.901,34.623c-2.705,0 -4.901,2.196 -4.901,4.901c0,2.705 2.196,4.901 4.901,4.901l9.614,0c2.705,0 4.901,-2.196 4.901,-4.901c0,-2.705 -2.196,-4.901 -4.901,-4.901l-9.614,0Z" style="fill:#fff;"/></g><g id="beam10" serif:id="beam"><path id="beam11" serif:id="beam" d="M44.212,64.534c0,-2.705 -2.196,-4.901 -4.901,-4.901c-2.704,-0 -4.9,2.196 -4.9,4.901l-0,9.614c-0,2.705 2.196,4.901 4.9,4.901c2.705,-0 4.901,-2.196 4.901,-4.901l0,-9.614Z" style="fill:#fff;"/></g><g id="beam12" serif:id="beam"><path id="beam13" serif:id="beam" d="M18.929,11.142c-1.912,-1.912 -5.018,-1.912 -6.931,0c-1.912,1.913 -1.912,5.019 0,6.931l6.799,6.799c1.912,1.912 5.018,1.912 6.931,-0c1.912,-1.913 1.912,-5.019 -0,-6.931l-6.799,-6.799Z" style="fill:#fff;"/></g><g id="beam14" serif:id="beam"><path id="beam15" serif:id="beam" d="M64.108,34.623c-2.705,0 -4.901,2.196 -4.901,4.901c-0,2.705 2.196,4.901 4.901,4.901l9.614,0c2.705,0 4.901,-2.196 4.901,-4.901c-0,-2.705 -2.196,-4.901 -4.901,-4.901l-9.614,0Z" style="fill:#fff;"/></g></g></g></svg>
    </template>

    <template v-if="mode == 'dark'">
        <svg width="100%" height="100%" viewBox="0 0 79 80" version="1.1" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" xml:space="preserve" style="fill-rule:evenodd;clip-rule:evenodd;stroke-linejoin:round;stroke-miterlimit:2;"><g id="mode_dark"><rect id="Bounds" x="0" y="-0" width="78.623" height="79.049" style="fill:none;"/><circle cx="39.311" cy="39.524" r="15.734" style="fill:#fff;"/><g id="beams"><g id="beam"><path id="beam1" serif:id="beam" d="M44.212,14.515c0,-2.705 -2.196,-4.901 -4.901,-4.901c-2.704,0 -4.901,2.196 -4.901,4.901c0,2.705 2.197,4.901 4.901,4.901c2.705,0 4.901,-2.196 4.901,-4.901Z" style="fill:#fff;"/></g><g id="beam2" serif:id="beam"><path id="beam3" serif:id="beam" d="M60.662,24.892c1.902,-1.902 1.902,-4.99 0,-6.892l-0.04,-0.039c-1.901,-1.902 -4.989,-1.902 -6.891,-0c-1.901,1.901 -1.901,4.989 0,6.891l0.04,0.04c1.902,1.901 4.989,1.901 6.891,-0Z" style="fill:#fff;"/></g><g id="beam4" serif:id="beam"><path id="beam5" serif:id="beam" d="M25.732,61.103c1.91,-1.91 1.91,-5.011 0,-6.921l-0.009,-0.01c-1.91,-1.91 -5.012,-1.91 -6.921,-0c-1.91,1.91 -1.91,5.011 -0,6.921l0.01,0.01c1.909,1.91 5.011,1.91 6.92,-0Z" style="fill:#fff;"/></g><g id="beam6" serif:id="beam"><path id="beam7" serif:id="beam" d="M60.672,54.167c-1.907,-1.907 -5.004,-1.907 -6.911,0l-0.02,0.02c-1.907,1.907 -1.907,5.004 0,6.911c1.907,1.907 5.004,1.907 6.911,-0l0.02,-0.02c1.907,-1.907 1.907,-5.004 0,-6.911Z" style="fill:#fff;"/></g><g id="beam8" serif:id="beam"><path id="beam9" serif:id="beam" d="M14.52,34.623c-2.702,0 -4.896,2.194 -4.896,4.896l0,0.01c0,2.702 2.194,4.896 4.896,4.896c2.702,0 4.896,-2.194 4.896,-4.896l-0,-0.01c-0,-2.702 -2.194,-4.896 -4.896,-4.896Z" style="fill:#fff;"/></g><g id="beam10" serif:id="beam"><path id="beam11" serif:id="beam" d="M44.212,64.534c0,-2.705 -2.196,-4.901 -4.901,-4.901c-2.704,-0 -4.901,2.196 -4.901,4.901c0,2.704 2.197,4.9 4.901,4.9c2.705,0 4.901,-2.196 4.901,-4.9Z" style="fill:#fff;"/></g><g id="beam12" serif:id="beam"><path id="beam13" serif:id="beam" d="M25.73,17.943c-1.911,-1.911 -5.015,-1.911 -6.926,0l-0.005,0.005c-1.911,1.911 -1.911,5.015 0,6.926c1.911,1.911 5.015,1.911 6.926,0l0.005,-0.005c1.911,-1.911 1.911,-5.014 -0,-6.926Z" style="fill:#fff;"/></g><g id="beam14" serif:id="beam"><path id="beam15" serif:id="beam" d="M64.098,34.623c-2.699,0 -4.891,2.192 -4.891,4.892l-0,0.019c-0,2.699 2.192,4.891 4.891,4.891c2.7,0 4.892,-2.192 4.892,-4.891l0,-0.019c0,-2.7 -2.192,-4.892 -4.892,-4.892Z" style="fill:#fff;"/></g></g></g></svg>
    </template>

    <template v-if="mode == 'darkest'">
        <svg width="100%" height="100%" viewBox="0 0 79 80" version="1.1" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" xml:space="preserve" style="fill-rule:evenodd;clip-rule:evenodd;stroke-linejoin:round;stroke-miterlimit:2;"><g id="mode_darkest"><rect id="Bounds" x="0" y="-0" width="78.623" height="79.049" style="fill:none;"/><path d="M39.315,23.791c8.684,-0 15.734,7.05 15.734,15.733c0,8.684 -7.05,15.734 -15.734,15.734c-8.683,0 -15.733,-7.05 -15.733,-15.734c-0,-8.683 7.05,-15.733 15.733,-15.733Zm0,4.737c6.069,0 10.997,4.927 10.997,10.996c-0,6.069 -4.928,10.996 -10.997,10.996c-6.068,0 -10.996,-4.927 -10.996,-10.996c0,-6.069 4.928,-10.996 10.996,-10.996Z" style="fill:#fff;"/><g id="beams"><g id="beam"><path id="beam1" serif:id="beam" d="M44.216,14.515c0,-2.705 -2.196,-4.901 -4.901,-4.901c-2.704,0 -4.9,2.196 -4.9,4.901c-0,2.705 2.196,4.901 4.9,4.901c2.705,0 4.901,-2.196 4.901,-4.901Z" style="fill:#fff;"/></g><g id="beam2" serif:id="beam"><path id="beam3" serif:id="beam" d="M60.666,24.892c1.902,-1.902 1.902,-4.99 0,-6.892l-0.04,-0.039c-1.901,-1.902 -4.989,-1.902 -6.891,-0c-1.901,1.901 -1.901,4.989 0,6.891l0.04,0.04c1.902,1.901 4.99,1.901 6.891,-0Z" style="fill:#fff;"/></g><g id="beam4" serif:id="beam"><path id="beam5" serif:id="beam" d="M25.737,61.103c1.909,-1.91 1.909,-5.011 -0,-6.921l-0.01,-0.01c-1.91,-1.91 -5.011,-1.91 -6.921,-0c-1.91,1.91 -1.91,5.011 -0,6.921l0.01,0.01c1.91,1.91 5.011,1.91 6.921,-0Z" style="fill:#fff;"/></g><g id="beam6" serif:id="beam"><path id="beam7" serif:id="beam" d="M60.676,54.167c-1.907,-1.907 -5.004,-1.907 -6.911,0l-0.02,0.02c-1.907,1.907 -1.907,5.004 0,6.911c1.907,1.907 5.004,1.907 6.911,-0l0.02,-0.02c1.907,-1.907 1.907,-5.004 0,-6.911Z" style="fill:#fff;"/></g><g id="beam8" serif:id="beam"><path id="beam9" serif:id="beam" d="M14.524,34.623c-2.702,0 -4.896,2.194 -4.896,4.896l0,0.01c0,2.702 2.194,4.896 4.896,4.896c2.702,0 4.896,-2.194 4.896,-4.896l0,-0.01c0,-2.702 -2.194,-4.896 -4.896,-4.896Z" style="fill:#fff;"/></g><g id="beam10" serif:id="beam"><path id="beam11" serif:id="beam" d="M44.216,64.534c0,-2.705 -2.196,-4.901 -4.901,-4.901c-2.704,-0 -4.9,2.196 -4.9,4.901c-0,2.704 2.196,4.9 4.9,4.9c2.705,0 4.901,-2.196 4.901,-4.9Z" style="fill:#fff;"/></g><g id="beam12" serif:id="beam"><path id="beam13" serif:id="beam" d="M25.734,17.943c-1.911,-1.911 -5.015,-1.911 -6.926,0l-0.005,0.005c-1.911,1.911 -1.911,5.015 0,6.926c1.911,1.911 5.015,1.911 6.926,0l0.005,-0.005c1.911,-1.911 1.911,-5.014 0,-6.926Z" style="fill:#fff;"/></g><g id="beam14" serif:id="beam"><path id="beam15" serif:id="beam" d="M64.103,34.623c-2.7,0 -4.892,2.192 -4.892,4.892l-0,0.019c-0,2.699 2.192,4.891 4.892,4.891c2.699,0 4.891,-2.192 4.891,-4.891l0,-0.019c0,-2.7 -2.192,-4.892 -4.891,-4.892Z" style="fill:#fff;"/></g></g></g></svg>
    </template>
</a>

<script>
(function() {
    const LOCAL_STORAGE_KEY = 'piccoloThemeMode'

    var initialMode = localStorage.getItem(LOCAL_STORAGE_KEY)

    if (initialMode) {
        // Make sure the value in local storage is valid
        if (['light', 'dark', 'darkest'].indexOf(initialMode) == -1) {
            initialMode = 'light'
            localStorage.setItem(LOCAL_STORAGE_KEY, initialMode)
        }
    } else {
        // Check if the client prefers dark mode
        if (window.matchMedia && window.matchMedia('(prefers-color-scheme: dark)').matches) {
            initialMode = 'dark'
        } else {
            initialMode = 'light'
        }
        localStorage.setItem(LOCAL_STORAGE_KEY, initialMode)
    }

    document.documentElement.dataset.mode = initialMode

    PetiteVue.createApp({
        'mode': initialMode,
        handleClick() {
            let currentMode = this.mode

            if (currentMode == 'light') {
                this.mode = 'dark'
            } else if (currentMode == 'dark') {
                this.mode = 'darkest'
            } else if (currentMode == 'darkest') {
                this.mode = 'light'
            }

            document.documentElement.dataset.mode = this.mode
            localStorage.setItem(LOCAL_STORAGE_KEY, this.mode)

            console.log(this.mode)
        }
    }).mount('#mode_toggle')
})()
</script>
            <p class="mobile_search_link">
                <a href="../search.html" title="Search">
                    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 65 64" fill-rule="evenodd" stroke-linejoin="round" stroke-miterlimit="2">
                        <path d="M14.873 40.009c-2.315-3.943-3.642-8.532-3.642-13.429C11.231 11.91 23.141 0 37.811 0s26.58 11.91 26.58 26.58-11.91 26.58-26.58 26.58a26.44 26.44 0 0 1-14.277-4.161L9.739 62.794a3.12 3.12 0 0 1-4.413 0L.913 58.382c-1.217-1.218-1.217-3.196 0-4.413l13.96-13.96zM37.811 8.054c10.225 0 18.526 8.301 18.526 18.526s-8.301 18.526-18.526 18.526-18.526-8.301-18.526-18.526S27.586 8.054 37.811 8.054z" fill="#fff" />
                    </svg>
                </a>
            </p>
        

        <div class="searchbox_wrapper">
            
<search id="searchbox" style="display: none" role="search">
  <h3 id="searchlabel">Quick search</h3>
    <div class="searchformwrapper">
    <form class="search" action="../search.html" method="get">
      <input type="text" name="q" aria-labelledby="searchlabel" autocomplete="off" autocorrect="off" autocapitalize="off" spellcheck="false"/>
      <input type="submit" value="Go" />
    </form>
    </div>
</search>
<script>document.getElementById('searchbox').style.display = "block"</script>
        </div>
    </nav>
</div>

    
      <div class="sphinxsidebar" role="navigation" aria-label="Main">
        <div class="sphinxsidebarwrapper"><p class="caption" role="heading"><span class="caption-text">Contents:</span></p>
<ul class="current">
<li class="toctree-l1"><a class="reference internal" href="../readme_link.html">Getting started</a></li>
<li class="toctree-l1"><a class="reference internal" href="../examples.html">Examples</a></li>
<li class="toctree-l1 current"><a class="reference internal" href="pinder.html">API reference</a><ul class="current">
<li class="toctree-l2 current"><a class="reference internal" href="pinder.core.html">pinder.core package</a><ul class="current">
<li class="toctree-l3 current"><a class="current reference internal" href="#">pinder.core.index package</a></li>
<li class="toctree-l3"><a class="reference internal" href="pinder.core.loader.html">pinder.core.loader package</a></li>
<li class="toctree-l3"><a class="reference internal" href="pinder.core.structure.html">pinder.core.structure package</a></li>
<li class="toctree-l3"><a class="reference internal" href="pinder.core.submit.html">pinder.core.submit package</a></li>
<li class="toctree-l3"><a class="reference internal" href="pinder.core.utils.html">pinder.core.utils package</a></li>
</ul>
</li>
<li class="toctree-l2"><a class="reference internal" href="pinder.data.html">pinder.data package</a></li>
<li class="toctree-l2"><a class="reference internal" href="pinder.eval.html">pinder.eval package</a></li>
<li class="toctree-l2"><a class="reference internal" href="pinder.methods.html">pinder.methods package</a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="../example_readme.html">Pinder abstractions</a></li>
<li class="toctree-l1"><a class="reference internal" href="../faq_link.html">Frequently asked questions</a></li>
<li class="toctree-l1"><a class="reference internal" href="../limitations_link.html">Limitations</a></li>
<li class="toctree-l1"><a class="reference internal" href="../changelog_link.html">Dataset changelog</a></li>
</ul>

        </div>
      </div>


    <div class="document">
      <div class="documentwrapper">
        <div class="bodywrapper">
          <div class="body" role="main">
            
  <section id="pinder-core-index-package">
<h1>pinder.core.index package<a class="headerlink" href="#pinder-core-index-package" title="Link to this heading">¶</a></h1>
<section id="submodules">
<h2>Submodules<a class="headerlink" href="#submodules" title="Link to this heading">¶</a></h2>
</section>
<section id="module-pinder.core.index.id">
<span id="pinder-core-index-id-module"></span><h2>pinder.core.index.id module<a class="headerlink" href="#module-pinder.core.index.id" title="Link to this heading">¶</a></h2>
<dl class="py class">
<dt class="sig sig-object py" id="pinder.core.index.id.Protein">
<em class="property"><span class="pre">class</span><span class="w"> </span></em><span class="sig-prename descclassname"><span class="pre">pinder.core.index.id.</span></span><span class="sig-name descname"><span class="pre">Protein</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">source</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">str</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">uniprot</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">str</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">chain</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">str</span><span class="w"> </span><span class="p"><span class="pre">|</span></span><span class="w"> </span><span class="pre">None</span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">'A'</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">from_residue</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">int</span><span class="w"> </span><span class="p"><span class="pre">|</span></span><span class="w"> </span><span class="pre">None</span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">to_residue</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">int</span><span class="w"> </span><span class="p"><span class="pre">|</span></span><span class="w"> </span><span class="pre">None</span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">None</span></span></em><span class="sig-paren">)</span><a class="reference internal" href="../_modules/pinder/core/index/id.html#Protein"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#pinder.core.index.id.Protein" title="Link to this definition">¶</a></dt>
<dd><p>Bases: <code class="xref py py-class docutils literal notranslate"><span class="pre">object</span></code></p>
<p>A class to represent a Protein.</p>
<p class="rubric">Examples</p>
<div class="doctest highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="n">protein</span> <span class="o">=</span> <span class="n">Protein</span><span class="p">(</span><span class="n">source</span><span class="o">=</span><span class="s1">&#39;6q0r&#39;</span><span class="p">,</span> <span class="n">chain</span><span class="o">=</span><span class="s1">&#39;B&#39;</span><span class="p">,</span> <span class="n">uniprot</span><span class="o">=</span><span class="s1">&#39;Q66K64&#39;</span><span class="p">)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="nb">str</span><span class="p">(</span><span class="n">protein</span><span class="p">)</span> <span class="o">==</span> <span class="s1">&#39;6q0r__B_Q66K64&#39;</span>
<span class="go">True</span>
</pre></div>
</div>
<div class="doctest highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="n">protein</span> <span class="o">=</span> <span class="n">Protein</span><span class="p">(</span><span class="n">source</span><span class="o">=</span><span class="s1">&#39;af2&#39;</span><span class="p">,</span> <span class="n">chain</span><span class="o">=</span><span class="s1">&#39;A&#39;</span><span class="p">,</span> <span class="n">uniprot</span><span class="o">=</span><span class="s1">&#39;Q14498&#39;</span><span class="p">,</span> <span class="n">from_residue</span><span class="o">=</span><span class="mi">1</span><span class="p">,</span> <span class="n">to_residue</span><span class="o">=</span><span class="mi">100</span><span class="p">)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="nb">str</span><span class="p">(</span><span class="n">protein</span><span class="p">)</span> <span class="o">==</span> <span class="s1">&#39;af2__A_Q14498_1_100&#39;</span>
<span class="go">False</span>
</pre></div>
</div>
<dl class="field-list simple">
<dt class="field-odd">Attributes<span class="colon">:</span></dt>
<dd class="field-odd"><dl class="simple">
<dt><strong>source</strong><span class="classifier">str</span></dt><dd><p>The source of the protein.</p>
</dd>
<dt><strong>uniprot</strong><span class="classifier">str</span></dt><dd><p>The uniprot ID of the protein.</p>
</dd>
<dt><strong>chain</strong><span class="classifier">Optional[str]</span></dt><dd><p>The chain of the protein, can be None.</p>
</dd>
<dt><strong>from_residue</strong><span class="classifier">Optional[int]</span></dt><dd><p>The starting residue of the protein, can be None.</p>
</dd>
<dt><strong>to_residue</strong><span class="classifier">Optional[int]</span></dt><dd><p>The ending residue of the protein, can be None.</p>
</dd>
</dl>
</dd>
</dl>
<p class="rubric">Methods</p>
<table class="docutils align-default">
<tbody>
<tr class="row-odd"><td><p><strong>__str__():</strong></p></td>
<td><p>Returns a string representation of the protein.</p></td>
</tr>
</tbody>
</table>
<dl class="py attribute">
<dt class="sig sig-object py" id="pinder.core.index.id.Protein.source">
<span class="sig-name descname"><span class="pre">source</span></span><em class="property"><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="pre">str</span></em><a class="headerlink" href="#pinder.core.index.id.Protein.source" title="Link to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="py attribute">
<dt class="sig sig-object py" id="pinder.core.index.id.Protein.uniprot">
<span class="sig-name descname"><span class="pre">uniprot</span></span><em class="property"><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="pre">str</span></em><a class="headerlink" href="#pinder.core.index.id.Protein.uniprot" title="Link to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="py attribute">
<dt class="sig sig-object py" id="pinder.core.index.id.Protein.chain">
<span class="sig-name descname"><span class="pre">chain</span></span><em class="property"><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="pre">str</span><span class="w"> </span><span class="p"><span class="pre">|</span></span><span class="w"> </span><span class="pre">None</span></em><em class="property"><span class="w"> </span><span class="p"><span class="pre">=</span></span><span class="w"> </span><span class="pre">'A'</span></em><a class="headerlink" href="#pinder.core.index.id.Protein.chain" title="Link to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="py attribute">
<dt class="sig sig-object py" id="pinder.core.index.id.Protein.from_residue">
<span class="sig-name descname"><span class="pre">from_residue</span></span><em class="property"><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="pre">int</span><span class="w"> </span><span class="p"><span class="pre">|</span></span><span class="w"> </span><span class="pre">None</span></em><em class="property"><span class="w"> </span><span class="p"><span class="pre">=</span></span><span class="w"> </span><span class="pre">None</span></em><a class="headerlink" href="#pinder.core.index.id.Protein.from_residue" title="Link to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="py attribute">
<dt class="sig sig-object py" id="pinder.core.index.id.Protein.to_residue">
<span class="sig-name descname"><span class="pre">to_residue</span></span><em class="property"><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="pre">int</span><span class="w"> </span><span class="p"><span class="pre">|</span></span><span class="w"> </span><span class="pre">None</span></em><em class="property"><span class="w"> </span><span class="p"><span class="pre">=</span></span><span class="w"> </span><span class="pre">None</span></em><a class="headerlink" href="#pinder.core.index.id.Protein.to_residue" title="Link to this definition">¶</a></dt>
<dd></dd></dl>

</dd></dl>

<dl class="py class">
<dt class="sig sig-object py" id="pinder.core.index.id.Monomer">
<em class="property"><span class="pre">class</span><span class="w"> </span></em><span class="sig-prename descclassname"><span class="pre">pinder.core.index.id.</span></span><span class="sig-name descname"><span class="pre">Monomer</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">proteins</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">list</span><span class="p"><span class="pre">[</span></span><a class="reference internal" href="#pinder.core.index.id.Protein" title="pinder.core.index.id.Protein"><span class="pre">Protein</span></a><span class="p"><span class="pre">]</span></span></span></em>, <em class="sig-param"><span class="n"><span class="pre">side</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">str</span><span class="w"> </span><span class="p"><span class="pre">|</span></span><span class="w"> </span><span class="pre">None</span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">None</span></span></em><span class="sig-paren">)</span><a class="reference internal" href="../_modules/pinder/core/index/id.html#Monomer"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#pinder.core.index.id.Monomer" title="Link to this definition">¶</a></dt>
<dd><p>Bases: <code class="xref py py-class docutils literal notranslate"><span class="pre">object</span></code></p>
<p>A class to represent a Monomer.</p>
<p class="rubric">Examples</p>
<div class="doctest highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="n">filename</span> <span class="o">=</span> <span class="s2">&quot;6q0r__B_Q66K64.pdb&quot;</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">parsed_monomer</span> <span class="o">=</span> <span class="n">Monomer</span><span class="o">.</span><span class="n">from_string</span><span class="p">(</span><span class="n">filename</span><span class="p">)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="nb">print</span><span class="p">(</span><span class="n">parsed_monomer</span><span class="p">)</span>
<span class="go">6q0r__B_Q66K64</span>
<span class="gp">&gt;&gt;&gt; </span><span class="k">assert</span> <span class="n">filename</span> <span class="o">==</span> <span class="nb">str</span><span class="p">(</span><span class="n">parsed_monomer</span><span class="p">)</span> <span class="o">+</span> <span class="s1">&#39;.pdb&#39;</span>
</pre></div>
</div>
<div class="doctest highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="nb">print</span><span class="p">(</span><span class="n">Monomer</span><span class="p">([</span><span class="n">Protein</span><span class="p">(</span><span class="n">source</span><span class="o">=</span><span class="s1">&#39;af2&#39;</span><span class="p">,</span> <span class="n">chain</span><span class="o">=</span><span class="s1">&#39;A&#39;</span><span class="p">,</span> <span class="n">uniprot</span><span class="o">=</span><span class="s1">&#39;Q14498&#39;</span><span class="p">,</span> <span class="n">from_residue</span><span class="o">=</span><span class="mi">1</span><span class="p">,</span> <span class="n">to_residue</span><span class="o">=</span><span class="mi">100</span><span class="p">)]))</span>
<span class="go">af2__Q14498_1_100</span>
<span class="gp">&gt;&gt;&gt; </span><span class="nb">print</span><span class="p">(</span><span class="n">Monomer</span><span class="p">([</span><span class="n">Protein</span><span class="p">(</span><span class="n">source</span><span class="o">=</span><span class="s1">&#39;af2&#39;</span><span class="p">,</span> <span class="n">chain</span><span class="o">=</span><span class="s1">&#39;A&#39;</span><span class="p">,</span> <span class="n">uniprot</span><span class="o">=</span><span class="s1">&#39;Q14498&#39;</span><span class="p">,</span> <span class="n">from_residue</span><span class="o">=</span><span class="mi">1</span><span class="p">,</span> <span class="n">to_residue</span><span class="o">=</span><span class="mi">100</span><span class="p">)],</span> <span class="n">side</span><span class="o">=</span><span class="s1">&#39;L&#39;</span><span class="p">))</span>
<span class="go">af2__Q14498_1_100-L</span>
<span class="gp">&gt;&gt;&gt; </span><span class="nb">print</span><span class="p">(</span><span class="n">Monomer</span><span class="p">([</span><span class="n">Protein</span><span class="p">(</span><span class="n">source</span><span class="o">=</span><span class="s1">&#39;6q0r&#39;</span><span class="p">,</span> <span class="n">chain</span><span class="o">=</span><span class="s1">&#39;B&#39;</span><span class="p">,</span> <span class="n">uniprot</span><span class="o">=</span><span class="s1">&#39;Q66K64&#39;</span><span class="p">)],</span> <span class="n">side</span><span class="o">=</span><span class="s1">&#39;R&#39;</span><span class="p">))</span>
<span class="go">6q0r__B_Q66K64-R</span>
<span class="gp">&gt;&gt;&gt; </span><span class="nb">print</span><span class="p">(</span><span class="n">Monomer</span><span class="o">.</span><span class="n">from_string</span><span class="p">(</span><span class="s2">&quot;af2__Q14498_1_100-R.pdb&quot;</span><span class="p">))</span>
<span class="go">af2__Q14498_1_100-R</span>
<span class="gp">&gt;&gt;&gt; </span><span class="nb">print</span><span class="p">(</span><span class="n">Monomer</span><span class="p">([</span><span class="n">Protein</span><span class="p">(</span><span class="n">source</span><span class="o">=</span><span class="s1">&#39;af2&#39;</span><span class="p">,</span> <span class="n">chain</span><span class="o">=</span><span class="s1">&#39;A&#39;</span><span class="p">,</span> <span class="n">uniprot</span><span class="o">=</span><span class="s1">&#39;Q14498&#39;</span><span class="p">)],</span> <span class="n">side</span><span class="o">=</span><span class="s1">&#39;R&#39;</span><span class="p">))</span>
<span class="go">af2__Q14498-R</span>
<span class="gp">&gt;&gt;&gt; </span><span class="nb">print</span><span class="p">(</span><span class="n">Monomer</span><span class="o">.</span><span class="n">from_string</span><span class="p">(</span><span class="s2">&quot;af2__Q14498-R.pdb&quot;</span><span class="p">))</span>
<span class="go">af2__Q14498-R</span>
</pre></div>
</div>
<dl class="field-list simple">
<dt class="field-odd">Attributes<span class="colon">:</span></dt>
<dd class="field-odd"><dl class="simple">
<dt><strong>proteins</strong><span class="classifier">List[Protein]</span></dt><dd><p>A list of Protein objects that make up the monomer.</p>
</dd>
<dt><strong>side</strong><span class="classifier">Optional[str]</span></dt><dd><p>The side of the monomer, can be None.</p>
</dd>
</dl>
</dd>
</dl>
<p class="rubric">Methods</p>
<table class="docutils align-default">
<tbody>
<tr class="row-odd"><td><p><strong>__str__():</strong></p></td>
<td><p>Returns a string representation of the monomer.</p></td>
</tr>
<tr class="row-even"><td><p><strong>from_string(monomer_str: str):</strong></p></td>
<td><p>Parses a monomer string into a Monomer object.</p></td>
</tr>
</tbody>
</table>
<dl class="py attribute">
<dt class="sig sig-object py" id="pinder.core.index.id.Monomer.proteins">
<span class="sig-name descname"><span class="pre">proteins</span></span><em class="property"><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="pre">list</span><span class="p"><span class="pre">[</span></span><a class="reference internal" href="#pinder.core.index.id.Protein" title="pinder.core.index.id.Protein"><span class="pre">Protein</span></a><span class="p"><span class="pre">]</span></span></em><a class="headerlink" href="#pinder.core.index.id.Monomer.proteins" title="Link to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="py attribute">
<dt class="sig sig-object py" id="pinder.core.index.id.Monomer.side">
<span class="sig-name descname"><span class="pre">side</span></span><em class="property"><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="pre">str</span><span class="w"> </span><span class="p"><span class="pre">|</span></span><span class="w"> </span><span class="pre">None</span></em><em class="property"><span class="w"> </span><span class="p"><span class="pre">=</span></span><span class="w"> </span><span class="pre">None</span></em><a class="headerlink" href="#pinder.core.index.id.Monomer.side" title="Link to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="pinder.core.index.id.Monomer.from_string">
<em class="property"><span class="pre">classmethod</span><span class="w"> </span></em><span class="sig-name descname"><span class="pre">from_string</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">monomer_str</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">str</span></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">&#x2192;</span> <span class="sig-return-typehint"><a class="reference internal" href="#pinder.core.index.id.Monomer" title="pinder.core.index.id.Monomer"><span class="pre">Monomer</span></a></span></span><a class="reference internal" href="../_modules/pinder/core/index/id.html#Monomer.from_string"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#pinder.core.index.id.Monomer.from_string" title="Link to this definition">¶</a></dt>
<dd><p>parse a monomer string into a Monomer object
supports when monomer is a .pdb file or a part of a complex</p>
</dd></dl>

</dd></dl>

<dl class="py class">
<dt class="sig sig-object py" id="pinder.core.index.id.Dimer">
<em class="property"><span class="pre">class</span><span class="w"> </span></em><span class="sig-prename descclassname"><span class="pre">pinder.core.index.id.</span></span><span class="sig-name descname"><span class="pre">Dimer</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">monomer1</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference internal" href="#pinder.core.index.id.Monomer" title="pinder.core.index.id.Monomer"><span class="pre">Monomer</span></a></span></em>, <em class="sig-param"><span class="n"><span class="pre">monomer2</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference internal" href="#pinder.core.index.id.Monomer" title="pinder.core.index.id.Monomer"><span class="pre">Monomer</span></a></span></em><span class="sig-paren">)</span><a class="reference internal" href="../_modules/pinder/core/index/id.html#Dimer"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#pinder.core.index.id.Dimer" title="Link to this definition">¶</a></dt>
<dd><p>Bases: <code class="xref py py-class docutils literal notranslate"><span class="pre">object</span></code></p>
<p>A class used to represent a Dimer, which is a complex of two Monomers.</p>
<p class="rubric">Examples</p>
<div class="doctest highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="n">monomer1</span> <span class="o">=</span> <span class="n">Monomer</span><span class="o">.</span><span class="n">from_string</span><span class="p">(</span><span class="s2">&quot;6q0r__B_Q66K64&quot;</span><span class="p">)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">monomer2</span> <span class="o">=</span> <span class="n">Monomer</span><span class="o">.</span><span class="n">from_string</span><span class="p">(</span><span class="s2">&quot;6q0r__D_Q14498&quot;</span><span class="p">)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">dimer</span> <span class="o">=</span> <span class="n">Dimer</span><span class="p">(</span><span class="n">monomer1</span><span class="p">,</span> <span class="n">monomer2</span><span class="p">)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="k">assert</span> <span class="nb">str</span><span class="p">(</span><span class="n">dimer</span><span class="p">)</span> <span class="o">==</span> <span class="s2">&quot;6q0r__B_Q66K64--6q0r__D_Q14498&quot;</span>
<span class="gp">&gt;&gt;&gt; </span><span class="nb">print</span><span class="p">(</span><span class="n">dimer</span><span class="p">)</span>
<span class="go">6q0r__B_Q66K64--6q0r__D_Q14498</span>
</pre></div>
</div>
<dl class="field-list simple">
<dt class="field-odd">Attributes<span class="colon">:</span></dt>
<dd class="field-odd"><dl class="simple">
<dt><strong>monomer1</strong><span class="classifier">Monomer</span></dt><dd><p>The first monomer in the dimer</p>
</dd>
<dt><strong>monomer2</strong><span class="classifier">Monomer</span></dt><dd><p>The second monomer in the dimer</p>
</dd>
</dl>
</dd>
</dl>
<p class="rubric">Methods</p>
<table class="docutils align-default">
<tbody>
<tr class="row-odd"><td><p><strong>__str__():</strong></p></td>
<td><p>Returns a string representation of the dimer.</p></td>
</tr>
</tbody>
</table>
<dl class="py attribute">
<dt class="sig sig-object py" id="pinder.core.index.id.Dimer.monomer1">
<span class="sig-name descname"><span class="pre">monomer1</span></span><em class="property"><span class="p"><span class="pre">:</span></span><span class="w"> </span><a class="reference internal" href="#pinder.core.index.id.Monomer" title="pinder.core.index.id.Monomer"><span class="pre">Monomer</span></a></em><a class="headerlink" href="#pinder.core.index.id.Dimer.monomer1" title="Link to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="py attribute">
<dt class="sig sig-object py" id="pinder.core.index.id.Dimer.monomer2">
<span class="sig-name descname"><span class="pre">monomer2</span></span><em class="property"><span class="p"><span class="pre">:</span></span><span class="w"> </span><a class="reference internal" href="#pinder.core.index.id.Monomer" title="pinder.core.index.id.Monomer"><span class="pre">Monomer</span></a></em><a class="headerlink" href="#pinder.core.index.id.Dimer.monomer2" title="Link to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="pinder.core.index.id.Dimer.from_string">
<em class="property"><span class="pre">classmethod</span><span class="w"> </span></em><span class="sig-name descname"><span class="pre">from_string</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">dimer_str</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">str</span></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">&#x2192;</span> <span class="sig-return-typehint"><a class="reference internal" href="#pinder.core.index.id.Dimer" title="pinder.core.index.id.Dimer"><span class="pre">Dimer</span></a></span></span><a class="reference internal" href="../_modules/pinder/core/index/id.html#Dimer.from_string"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#pinder.core.index.id.Dimer.from_string" title="Link to this definition">¶</a></dt>
<dd></dd></dl>

</dd></dl>

</section>
<section id="module-pinder.core.index.system">
<span id="pinder-core-index-system-module"></span><h2>pinder.core.index.system module<a class="headerlink" href="#module-pinder.core.index.system" title="Link to this heading">¶</a></h2>
<dl class="py class">
<dt class="sig sig-object py" id="pinder.core.index.system.FolderNames">
<em class="property"><span class="pre">class</span><span class="w"> </span></em><span class="sig-prename descclassname"><span class="pre">pinder.core.index.system.</span></span><span class="sig-name descname"><span class="pre">FolderNames</span></span><a class="reference internal" href="../_modules/pinder/core/index/system.html#FolderNames"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#pinder.core.index.system.FolderNames" title="Link to this definition">¶</a></dt>
<dd><p>Bases: <code class="xref py py-class docutils literal notranslate"><span class="pre">object</span></code></p>
<dl class="py attribute">
<dt class="sig sig-object py" id="pinder.core.index.system.FolderNames.apo">
<span class="sig-name descname"><span class="pre">apo</span></span><em class="property"><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="pre">str</span></em><em class="property"><span class="w"> </span><span class="p"><span class="pre">=</span></span><span class="w"> </span><span class="pre">'apo'</span></em><a class="headerlink" href="#pinder.core.index.system.FolderNames.apo" title="Link to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="py attribute">
<dt class="sig sig-object py" id="pinder.core.index.system.FolderNames.holo">
<span class="sig-name descname"><span class="pre">holo</span></span><em class="property"><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="pre">str</span></em><em class="property"><span class="w"> </span><span class="p"><span class="pre">=</span></span><span class="w"> </span><span class="pre">'holo'</span></em><a class="headerlink" href="#pinder.core.index.system.FolderNames.holo" title="Link to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="py attribute">
<dt class="sig sig-object py" id="pinder.core.index.system.FolderNames.predicted">
<span class="sig-name descname"><span class="pre">predicted</span></span><em class="property"><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="pre">str</span></em><em class="property"><span class="w"> </span><span class="p"><span class="pre">=</span></span><span class="w"> </span><span class="pre">'predicted'</span></em><a class="headerlink" href="#pinder.core.index.system.FolderNames.predicted" title="Link to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="py attribute">
<dt class="sig sig-object py" id="pinder.core.index.system.FolderNames.alphafold">
<span class="sig-name descname"><span class="pre">alphafold</span></span><em class="property"><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="pre">str</span></em><em class="property"><span class="w"> </span><span class="p"><span class="pre">=</span></span><span class="w"> </span><span class="pre">'predicted'</span></em><a class="headerlink" href="#pinder.core.index.system.FolderNames.alphafold" title="Link to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="py attribute">
<dt class="sig sig-object py" id="pinder.core.index.system.FolderNames.af2">
<span class="sig-name descname"><span class="pre">af2</span></span><em class="property"><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="pre">str</span></em><em class="property"><span class="w"> </span><span class="p"><span class="pre">=</span></span><span class="w"> </span><span class="pre">'predicted'</span></em><a class="headerlink" href="#pinder.core.index.system.FolderNames.af2" title="Link to this definition">¶</a></dt>
<dd></dd></dl>

</dd></dl>

<dl class="py class">
<dt class="sig sig-object py" id="pinder.core.index.system.PinderSystem">
<em class="property"><span class="pre">class</span><span class="w"> </span></em><span class="sig-prename descclassname"><span class="pre">pinder.core.index.system.</span></span><span class="sig-name descname"><span class="pre">PinderSystem</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">entry</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">str</span><span class="w"> </span><span class="p"><span class="pre">|</span></span><span class="w"> </span><a class="reference internal" href="#pinder.core.index.utils.IndexEntry" title="pinder.core.index.utils.IndexEntry"><span class="pre">IndexEntry</span></a></span></em>, <em class="sig-param"><span class="n"><span class="pre">apo_receptor_pdb_code</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">str</span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">''</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">apo_ligand_pdb_code</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">str</span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">''</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">metadata</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference internal" href="#pinder.core.index.utils.MetadataEntry" title="pinder.core.index.utils.MetadataEntry"><span class="pre">MetadataEntry</span></a><span class="w"> </span><span class="p"><span class="pre">|</span></span><span class="w"> </span><span class="pre">None</span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">dataset_path</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">Path</span><span class="w"> </span><span class="p"><span class="pre">|</span></span><span class="w"> </span><span class="pre">None</span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="o"><span class="pre">**</span></span><span class="n"><span class="pre">kwargs</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">dict</span><span class="p"><span class="pre">[</span></span><span class="pre">str</span><span class="p"><span class="pre">,</span></span><span class="w"> </span><span class="pre">Any</span><span class="p"><span class="pre">]</span></span></span></em><span class="sig-paren">)</span><a class="reference internal" href="../_modules/pinder/core/index/system.html#PinderSystem"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#pinder.core.index.system.PinderSystem" title="Link to this definition">¶</a></dt>
<dd><p>Bases: <code class="xref py py-class docutils literal notranslate"><span class="pre">object</span></code></p>
<p>Represents a system within the Pinder framework designed to handle and process
structural data. It provides functionality to load, align, and analyze
protein structures within the context of a Pinder index entry.</p>
<p>Upon initialization, the system prepares directories for different structure types
(holo, apo, predicted) and loads the respective protein structures.</p>
<p>Methods include creating complexes, calculating RMSD, difficulty assessment,
and updating substructure presence based on filtering criteria.</p>
<dl class="py attribute">
<dt class="sig sig-object py" id="pinder.core.index.system.PinderSystem.entry">
<span class="sig-name descname"><span class="pre">entry</span></span><a class="headerlink" href="#pinder.core.index.system.PinderSystem.entry" title="Link to this definition">¶</a></dt>
<dd><p>An index entry object containing primary metadata for the system.</p>
<dl class="field-list simple">
<dt class="field-odd">Type<span class="colon">:</span></dt>
<dd class="field-odd"><p><a class="reference internal" href="#pinder.core.index.utils.IndexEntry" title="pinder.core.index.utils.IndexEntry">IndexEntry</a></p>
</dd>
</dl>
</dd></dl>

<dl class="py attribute">
<dt class="sig sig-object py" id="pinder.core.index.system.PinderSystem.pdbs_path">
<span class="sig-name descname"><span class="pre">pdbs_path</span></span><a class="headerlink" href="#pinder.core.index.system.PinderSystem.pdbs_path" title="Link to this definition">¶</a></dt>
<dd><p>Path to the directory containing PDB files.</p>
<dl class="field-list simple">
<dt class="field-odd">Type<span class="colon">:</span></dt>
<dd class="field-odd"><p>Path</p>
</dd>
</dl>
</dd></dl>

<dl class="py attribute">
<dt class="sig sig-object py" id="pinder.core.index.system.PinderSystem.mappings_path">
<span class="sig-name descname"><span class="pre">mappings_path</span></span><a class="headerlink" href="#pinder.core.index.system.PinderSystem.mappings_path" title="Link to this definition">¶</a></dt>
<dd><p>Path to the directory containing Parquet mapping files.</p>
<dl class="field-list simple">
<dt class="field-odd">Type<span class="colon">:</span></dt>
<dd class="field-odd"><p>Path</p>
</dd>
</dl>
</dd></dl>

<dl class="py attribute">
<dt class="sig sig-object py" id="pinder.core.index.system.PinderSystem.native">
<span class="sig-name descname"><span class="pre">native</span></span><a class="headerlink" href="#pinder.core.index.system.PinderSystem.native" title="Link to this definition">¶</a></dt>
<dd><p>The native (ground-truth) structure of the system.</p>
<dl class="field-list simple">
<dt class="field-odd">Type<span class="colon">:</span></dt>
<dd class="field-odd"><p><a class="reference internal" href="pinder.core.loader.html#pinder.core.loader.structure.Structure" title="pinder.core.loader.structure.Structure">Structure</a></p>
</dd>
</dl>
</dd></dl>

<dl class="py attribute">
<dt class="sig sig-object py" id="pinder.core.index.system.PinderSystem.holo_dir">
<span class="sig-name descname"><span class="pre">holo_dir</span></span><a class="headerlink" href="#pinder.core.index.system.PinderSystem.holo_dir" title="Link to this definition">¶</a></dt>
<dd><p>Directory for the holo structures.</p>
<dl class="field-list simple">
<dt class="field-odd">Type<span class="colon">:</span></dt>
<dd class="field-odd"><p>Path</p>
</dd>
</dl>
</dd></dl>

<dl class="py attribute">
<dt class="sig sig-object py" id="pinder.core.index.system.PinderSystem.apo_dir">
<span class="sig-name descname"><span class="pre">apo_dir</span></span><a class="headerlink" href="#pinder.core.index.system.PinderSystem.apo_dir" title="Link to this definition">¶</a></dt>
<dd><p>Directory for the apo structures.</p>
<dl class="field-list simple">
<dt class="field-odd">Type<span class="colon">:</span></dt>
<dd class="field-odd"><p>Path</p>
</dd>
</dl>
</dd></dl>

<dl class="py attribute">
<dt class="sig sig-object py" id="pinder.core.index.system.PinderSystem.pred_dir">
<span class="sig-name descname"><span class="pre">pred_dir</span></span><a class="headerlink" href="#pinder.core.index.system.PinderSystem.pred_dir" title="Link to this definition">¶</a></dt>
<dd><p>Directory for the predicted structures.</p>
<dl class="field-list simple">
<dt class="field-odd">Type<span class="colon">:</span></dt>
<dd class="field-odd"><p>Path</p>
</dd>
</dl>
</dd></dl>

<dl class="py attribute">
<dt class="sig sig-object py" id="pinder.core.index.system.PinderSystem.holo_receptor">
<span class="sig-name descname"><span class="pre">holo_receptor</span></span><a class="headerlink" href="#pinder.core.index.system.PinderSystem.holo_receptor" title="Link to this definition">¶</a></dt>
<dd><p>The holo form of the receptor.</p>
<dl class="field-list simple">
<dt class="field-odd">Type<span class="colon">:</span></dt>
<dd class="field-odd"><p><a class="reference internal" href="pinder.core.loader.html#pinder.core.loader.structure.Structure" title="pinder.core.loader.structure.Structure">Structure</a></p>
</dd>
</dl>
</dd></dl>

<dl class="py attribute">
<dt class="sig sig-object py" id="pinder.core.index.system.PinderSystem.holo_ligand">
<span class="sig-name descname"><span class="pre">holo_ligand</span></span><a class="headerlink" href="#pinder.core.index.system.PinderSystem.holo_ligand" title="Link to this definition">¶</a></dt>
<dd><p>The holo form of the ligand.</p>
<dl class="field-list simple">
<dt class="field-odd">Type<span class="colon">:</span></dt>
<dd class="field-odd"><p><a class="reference internal" href="pinder.core.loader.html#pinder.core.loader.structure.Structure" title="pinder.core.loader.structure.Structure">Structure</a></p>
</dd>
</dl>
</dd></dl>

<dl class="py attribute">
<dt class="sig sig-object py" id="pinder.core.index.system.PinderSystem.apo_receptor">
<span class="sig-name descname"><span class="pre">apo_receptor</span></span><a class="headerlink" href="#pinder.core.index.system.PinderSystem.apo_receptor" title="Link to this definition">¶</a></dt>
<dd><p>The apo form of the receptor.</p>
<dl class="field-list simple">
<dt class="field-odd">Type<span class="colon">:</span></dt>
<dd class="field-odd"><p><a class="reference internal" href="pinder.core.loader.html#pinder.core.loader.structure.Structure" title="pinder.core.loader.structure.Structure">Structure</a></p>
</dd>
</dl>
</dd></dl>

<dl class="py attribute">
<dt class="sig sig-object py" id="pinder.core.index.system.PinderSystem.apo_ligand">
<span class="sig-name descname"><span class="pre">apo_ligand</span></span><a class="headerlink" href="#pinder.core.index.system.PinderSystem.apo_ligand" title="Link to this definition">¶</a></dt>
<dd><p>The apo form of the ligand.</p>
<dl class="field-list simple">
<dt class="field-odd">Type<span class="colon">:</span></dt>
<dd class="field-odd"><p><a class="reference internal" href="pinder.core.loader.html#pinder.core.loader.structure.Structure" title="pinder.core.loader.structure.Structure">Structure</a></p>
</dd>
</dl>
</dd></dl>

<dl class="py attribute">
<dt class="sig sig-object py" id="pinder.core.index.system.PinderSystem.pred_receptor">
<span class="sig-name descname"><span class="pre">pred_receptor</span></span><a class="headerlink" href="#pinder.core.index.system.PinderSystem.pred_receptor" title="Link to this definition">¶</a></dt>
<dd><p>The predicted structure of the receptor.</p>
<dl class="field-list simple">
<dt class="field-odd">Type<span class="colon">:</span></dt>
<dd class="field-odd"><p><a class="reference internal" href="pinder.core.loader.html#pinder.core.loader.structure.Structure" title="pinder.core.loader.structure.Structure">Structure</a></p>
</dd>
</dl>
</dd></dl>

<dl class="py attribute">
<dt class="sig sig-object py" id="pinder.core.index.system.PinderSystem.pred_ligand">
<span class="sig-name descname"><span class="pre">pred_ligand</span></span><a class="headerlink" href="#pinder.core.index.system.PinderSystem.pred_ligand" title="Link to this definition">¶</a></dt>
<dd><p>The predicted structure of the ligand.</p>
<dl class="field-list simple">
<dt class="field-odd">Type<span class="colon">:</span></dt>
<dd class="field-odd"><p><a class="reference internal" href="pinder.core.loader.html#pinder.core.loader.structure.Structure" title="pinder.core.loader.structure.Structure">Structure</a></p>
</dd>
</dl>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="pinder.core.index.system.PinderSystem.filter_common_uniprot_res">
<span class="sig-name descname"><span class="pre">filter_common_uniprot_res</span></span><span class="sig-paren">(</span><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">&#x2192;</span> <span class="sig-return-typehint"><span class="pre">None</span></span></span><a class="reference internal" href="../_modules/pinder/core/index/system.html#PinderSystem.filter_common_uniprot_res"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#pinder.core.index.system.PinderSystem.filter_common_uniprot_res" title="Link to this definition">¶</a></dt>
<dd><p>Filters the loaded protein structures for common UniProt residues, ensuring that
comparisons between structures are made on a common set of residues.</p>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="pinder.core.index.system.PinderSystem.create_masked_bound_unbound_complexes">
<span class="sig-name descname"><span class="pre">create_masked_bound_unbound_complexes</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">monomer_types</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">Sequence</span><span class="p"><span class="pre">[</span></span><span class="pre">str</span><span class="p"><span class="pre">]</span></span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">['apo',</span> <span class="pre">'predicted']</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">remove_differing_atoms</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">bool</span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">True</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">renumber_residues</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">bool</span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">False</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">remove_differing_annotations</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">bool</span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">False</span></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">&#x2192;</span> <span class="sig-return-typehint"><span class="pre">tuple</span><span class="p"><span class="pre">[</span></span><a class="reference internal" href="pinder.core.loader.html#pinder.core.loader.structure.Structure" title="pinder.core.loader.structure.Structure"><span class="pre">Structure</span></a><span class="p"><span class="pre">,</span></span><span class="w"> </span><a class="reference internal" href="pinder.core.loader.html#pinder.core.loader.structure.Structure" title="pinder.core.loader.structure.Structure"><span class="pre">Structure</span></a><span class="p"><span class="pre">,</span></span><span class="w"> </span><a class="reference internal" href="pinder.core.loader.html#pinder.core.loader.structure.Structure" title="pinder.core.loader.structure.Structure"><span class="pre">Structure</span></a><span class="p"><span class="pre">]</span></span></span></span><a class="reference internal" href="../_modules/pinder/core/index/system.html#PinderSystem.create_masked_bound_unbound_complexes"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#pinder.core.index.system.PinderSystem.create_masked_bound_unbound_complexes" title="Link to this definition">¶</a></dt>
<dd><p>Create dimer complexes for apo and predicted cropped to common holo substructures.</p>
<p>The method applies a pairwise masking procedure which crops both unbound and bound structures
such that they have equal numbers of residues and atom counts.</p>
<p>Note: this method may result in very distorted holo (ground-truth) structures if
the unbound monomer structures have little to no sequence and atoms in common.
Unless you need all monomer types to be equal shapes, the <cite>PinderSystem.create_complex</cite> method
or pure-superposition without masking (Structure.superimpose) is more appropriate.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>monomer_types</strong> (<em>Sequence</em><em>[</em><em>str</em><em>]</em>) – The unbound monomer types to consider (apo, predicted, or both).</p></li>
<li><p><strong>remove_differing_atoms</strong> (<em>bool</em>) – Whether to remove non-overlappings atoms that may still be present even after sequence-based alignment.</p></li>
<li><p><strong>renumber_residues</strong> (<em>bool</em>) – Whether to renumber the residues in the receptor and ligand <cite>Structure</cite>’s to match numbering of the holo counterparts.</p></li>
<li><p><strong>remove_differing_annotations</strong> (<em>bool</em>) – Whether to remove annotation categories (set to empty str or default value for the category type).
This is useful if you need to perform biotite.structure.filter_intersection on the resulting structures.
Note: this can have unintended side-effects like stripping the <cite>element</cite> attribute on structures.
By default, the annotation categories are removed if they don’t match in order to define the intersecting atom mask,
after which the original structure annotations are preserved by applying the intersecting mask to the original AtomArray.
Default is False.</p></li>
</ul>
</dd>
<dt class="field-even">Returns<span class="colon">:</span></dt>
<dd class="field-even"><p>A tuple of the cropped holo, apo, and predicted Structure objects, respectively.</p>
</dd>
<dt class="field-odd">Return type<span class="colon">:</span></dt>
<dd class="field-odd"><p>tuple[<a class="reference internal" href="pinder.core.loader.html#pinder.core.loader.structure.Structure" title="pinder.core.loader.structure.Structure">Structure</a>, <a class="reference internal" href="pinder.core.loader.html#pinder.core.loader.structure.Structure" title="pinder.core.loader.structure.Structure">Structure</a>, <a class="reference internal" href="pinder.core.loader.html#pinder.core.loader.structure.Structure" title="pinder.core.loader.structure.Structure">Structure</a>]</p>
</dd>
</dl>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="pinder.core.index.system.PinderSystem.create_complex">
<span class="sig-name descname"><span class="pre">create_complex</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">receptor</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference internal" href="pinder.core.loader.html#pinder.core.loader.structure.Structure" title="pinder.core.loader.structure.Structure"><span class="pre">Structure</span></a></span></em>, <em class="sig-param"><span class="n"><span class="pre">ligand</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference internal" href="pinder.core.loader.html#pinder.core.loader.structure.Structure" title="pinder.core.loader.structure.Structure"><span class="pre">Structure</span></a></span></em>, <em class="sig-param"><span class="n"><span class="pre">remove_differing_atoms</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">bool</span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">True</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">renumber_residues</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">bool</span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">False</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">remove_differing_annotations</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">bool</span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">False</span></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">&#x2192;</span> <span class="sig-return-typehint"><a class="reference internal" href="pinder.core.loader.html#pinder.core.loader.structure.Structure" title="pinder.core.loader.structure.Structure"><span class="pre">Structure</span></a></span></span><a class="reference internal" href="../_modules/pinder/core/index/system.html#PinderSystem.create_complex"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#pinder.core.index.system.PinderSystem.create_complex" title="Link to this definition">¶</a></dt>
<dd><p>Creates a complex from the receptor and ligand structures.</p>
<p>The complex is created by aligning the monomers to their
respective holo forms and combining them into a single structure.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>receptor</strong> (<a class="reference internal" href="pinder.core.loader.html#pinder.core.loader.structure.Structure" title="pinder.core.loader.structure.Structure"><em>Structure</em></a>) – The receptor structure.</p></li>
<li><p><strong>ligand</strong> (<a class="reference internal" href="pinder.core.loader.html#pinder.core.loader.structure.Structure" title="pinder.core.loader.structure.Structure"><em>Structure</em></a>) – The ligand structure.</p></li>
<li><p><strong>remove_differing_atoms</strong> (<em>bool</em>) – Whether to remove non-overlappings atoms that may still be present even after sequence-based alignment.</p></li>
<li><p><strong>renumber_residues</strong> (<em>bool</em>) – Whether to renumber the residues in the receptor and ligand <cite>Structure</cite>’s to match numbering of the holo counterparts.</p></li>
<li><p><strong>remove_differing_annotations</strong> (<em>bool</em>) – Whether to remove annotation categories (set to empty str or default value for the category type).
This is useful if you need to perform biotite.structure.filter_intersection on the resulting structures.
Note: this can have unintended side-effects like stripping the <cite>element</cite> attribute on structures.
By default, the annotation categories are removed if they don’t match in order to define the intersecting atom mask,
after which the original structure annotations are preserved by applying the intersecting mask to the original AtomArray.
Default is False.</p></li>
</ul>
</dd>
<dt class="field-even">Returns<span class="colon">:</span></dt>
<dd class="field-even"><p>A new Structure instance representing the complex.</p>
</dd>
<dt class="field-odd">Return type<span class="colon">:</span></dt>
<dd class="field-odd"><p><a class="reference internal" href="pinder.core.loader.html#pinder.core.loader.structure.Structure" title="pinder.core.loader.structure.Structure">Structure</a></p>
</dd>
</dl>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="pinder.core.index.system.PinderSystem.create_apo_complex">
<span class="sig-name descname"><span class="pre">create_apo_complex</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">remove_differing_atoms</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">bool</span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">True</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">renumber_residues</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">bool</span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">False</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">remove_differing_annotations</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">bool</span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">False</span></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">&#x2192;</span> <span class="sig-return-typehint"><a class="reference internal" href="pinder.core.loader.html#pinder.core.loader.structure.Structure" title="pinder.core.loader.structure.Structure"><span class="pre">Structure</span></a></span></span><a class="reference internal" href="../_modules/pinder/core/index/system.html#PinderSystem.create_apo_complex"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#pinder.core.index.system.PinderSystem.create_apo_complex" title="Link to this definition">¶</a></dt>
<dd><p>Creates an apo complex using the receptor and ligand structures.
Falls back to the holo structures if apo structures are not available.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>remove_differing_atoms</strong> (<em>bool</em>) – Whether to remove non-overlappings atoms that may still be present even after sequence-based alignment.</p></li>
<li><p><strong>renumber_residues</strong> (<em>bool</em>) – Whether to renumber the residues in the apo receptor and ligand <cite>Structure</cite>’s to match numbering of the holo counterparts.</p></li>
<li><p><strong>remove_differing_annotations</strong> (<em>bool</em>) – Whether to remove annotation categories (set to empty str or default value for the category type).
This is useful if you need to perform biotite.structure.filter_intersection on the resulting structures.
Note: this can have unintended side-effects like stripping the <cite>element</cite> attribute on structures.
By default, the annotation categories are removed if they don’t match in order to define the intersecting atom mask,
after which the original structure annotations are preserved by applying the intersecting mask to the original AtomArray.
Default is False.</p></li>
</ul>
</dd>
<dt class="field-even">Returns<span class="colon">:</span></dt>
<dd class="field-even"><p>A new Structure instance representing the apo complex.</p>
</dd>
<dt class="field-odd">Return type<span class="colon">:</span></dt>
<dd class="field-odd"><p><a class="reference internal" href="pinder.core.loader.html#pinder.core.loader.structure.Structure" title="pinder.core.loader.structure.Structure">Structure</a></p>
</dd>
</dl>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="pinder.core.index.system.PinderSystem.create_pred_complex">
<span class="sig-name descname"><span class="pre">create_pred_complex</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">remove_differing_atoms</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">bool</span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">True</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">renumber_residues</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">bool</span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">False</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">remove_differing_annotations</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">bool</span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">False</span></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">&#x2192;</span> <span class="sig-return-typehint"><a class="reference internal" href="pinder.core.loader.html#pinder.core.loader.structure.Structure" title="pinder.core.loader.structure.Structure"><span class="pre">Structure</span></a></span></span><a class="reference internal" href="../_modules/pinder/core/index/system.html#PinderSystem.create_pred_complex"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#pinder.core.index.system.PinderSystem.create_pred_complex" title="Link to this definition">¶</a></dt>
<dd><p>Creates a predicted complex using the receptor and ligand structures.
Falls back to the holo structures if predicted structures are not available.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>remove_differing_atoms</strong> (<em>bool</em>) – Whether to remove non-overlappings atoms that may still be present even after sequence-based alignment.</p></li>
<li><p><strong>renumber_residues</strong> (<em>bool</em>) – Whether to renumber the residues in the predicted receptor and ligand <cite>Structure</cite>’s to match numbering of the holo counterparts.</p></li>
<li><p><strong>remove_differing_annotations</strong> (<em>bool</em>) – Whether to remove annotation categories (set to empty str or default value for the category type).
This is useful if you need to perform biotite.structure.filter_intersection on the resulting structures.
Note: this can have unintended side-effects like stripping the <cite>element</cite> attribute on structures.
By default, the annotation categories are removed if they don’t match in order to define the intersecting atom mask,
after which the original structure annotations are preserved by applying the intersecting mask to the original AtomArray.
Default is False.</p></li>
</ul>
</dd>
<dt class="field-even">Returns<span class="colon">:</span></dt>
<dd class="field-even"><p>A new Structure instance representing the predicted complex.</p>
</dd>
<dt class="field-odd">Return type<span class="colon">:</span></dt>
<dd class="field-odd"><p><a class="reference internal" href="pinder.core.loader.html#pinder.core.loader.structure.Structure" title="pinder.core.loader.structure.Structure">Structure</a></p>
</dd>
</dl>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="pinder.core.index.system.PinderSystem.unbound_rmsd">
<span class="sig-name descname"><span class="pre">unbound_rmsd</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">monomer_name</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference internal" href="pinder.core.structure.html#pinder.core.structure.models.MonomerName" title="pinder.core.structure.models.MonomerName"><span class="pre">MonomerName</span></a></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">&#x2192;</span> <span class="sig-return-typehint"><span class="pre">dict</span><span class="p"><span class="pre">[</span></span><span class="pre">str</span><span class="p"><span class="pre">,</span></span><span class="w"> </span><span class="pre">float</span><span class="p"><span class="pre">]</span></span></span></span><a class="reference internal" href="../_modules/pinder/core/index/system.html#PinderSystem.unbound_rmsd"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#pinder.core.index.system.PinderSystem.unbound_rmsd" title="Link to this definition">¶</a></dt>
<dd><p>Calculates the RMSD of the unbound receptor and ligand with respect to
their holo forms for a given monomer state (apo or predicted).</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><p><strong>monomer_name</strong> (<a class="reference internal" href="pinder.core.structure.html#pinder.core.structure.models.MonomerName" title="pinder.core.structure.models.MonomerName"><em>MonomerName</em></a>) – Enum representing the monomer state.</p>
</dd>
<dt class="field-even">Returns<span class="colon">:</span></dt>
<dd class="field-even"><p>A dictionary with RMSD values for the receptor and ligand.</p>
</dd>
<dt class="field-odd">Return type<span class="colon">:</span></dt>
<dd class="field-odd"><p>dict[str, float]</p>
</dd>
</dl>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="pinder.core.index.system.PinderSystem.unbound_difficulty">
<span class="sig-name descname"><span class="pre">unbound_difficulty</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">monomer_name</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference internal" href="pinder.core.structure.html#pinder.core.structure.models.MonomerName" title="pinder.core.structure.models.MonomerName"><span class="pre">MonomerName</span></a></span></em>, <em class="sig-param"><span class="n"><span class="pre">contact_rad</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">float</span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">5.0</span></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">&#x2192;</span> <span class="sig-return-typehint"><span class="pre">dict</span><span class="p"><span class="pre">[</span></span><span class="pre">str</span><span class="p"><span class="pre">,</span></span><span class="w"> </span><span class="pre">float</span><span class="w"> </span><span class="p"><span class="pre">|</span></span><span class="w"> </span><span class="pre">str</span><span class="p"><span class="pre">]</span></span></span></span><a class="reference internal" href="../_modules/pinder/core/index/system.html#PinderSystem.unbound_difficulty"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#pinder.core.index.system.PinderSystem.unbound_difficulty" title="Link to this definition">¶</a></dt>
<dd><p>Assesses the difficulty of docking the unbound structures based on the given monomer
state (apo or predicted).</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><p><strong>monomer_name</strong> (<a class="reference internal" href="pinder.core.structure.html#pinder.core.structure.models.MonomerName" title="pinder.core.structure.models.MonomerName"><em>MonomerName</em></a>) – Enum representing the monomer state.</p>
</dd>
<dt class="field-even">Returns<span class="colon">:</span></dt>
<dd class="field-even"><p>A dictionary with difficulty assessment metrics.</p>
</dd>
<dt class="field-odd">Return type<span class="colon">:</span></dt>
<dd class="field-odd"><p>dict[str, Union[float, str]]</p>
</dd>
</dl>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="pinder.core.index.system.PinderSystem.apo_monomer_difficulty">
<span class="sig-name descname"><span class="pre">apo_monomer_difficulty</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">monomer_name</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference internal" href="pinder.core.structure.html#pinder.core.structure.models.MonomerName" title="pinder.core.structure.models.MonomerName"><span class="pre">MonomerName</span></a></span></em>, <em class="sig-param"><span class="n"><span class="pre">body</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">str</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">contact_rad</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">float</span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">5.0</span></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">&#x2192;</span> <span class="sig-return-typehint"><span class="pre">dict</span><span class="p"><span class="pre">[</span></span><span class="pre">str</span><span class="p"><span class="pre">,</span></span><span class="w"> </span><span class="pre">float</span><span class="w"> </span><span class="p"><span class="pre">|</span></span><span class="w"> </span><span class="pre">str</span><span class="p"><span class="pre">]</span></span></span></span><a class="reference internal" href="../_modules/pinder/core/index/system.html#PinderSystem.apo_monomer_difficulty"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#pinder.core.index.system.PinderSystem.apo_monomer_difficulty" title="Link to this definition">¶</a></dt>
<dd><p>Evaluates the difficulty of docking for an individual apo monomer.</p>
<p>Takes the specified body of the monomer (receptor or ligand) into account.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>monomer_name</strong> (<a class="reference internal" href="pinder.core.structure.html#pinder.core.structure.models.MonomerName" title="pinder.core.structure.models.MonomerName"><em>MonomerName</em></a>) – Enum representing the monomer state.</p></li>
<li><p><strong>body</strong> (<em>str</em>) – String indicating which body (‘receptor’ or ‘ligand’) to use.</p></li>
</ul>
</dd>
<dt class="field-even">Returns<span class="colon">:</span></dt>
<dd class="field-even"><p>A dictionary with docking difficulty metrics for the specified monomer body.</p>
</dd>
<dt class="field-odd">Return type<span class="colon">:</span></dt>
<dd class="field-odd"><p>dict[str, Union[float, str]]</p>
</dd>
</dl>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="pinder.core.index.system.PinderSystem.download_entry">
<span class="sig-name descname"><span class="pre">download_entry</span></span><span class="sig-paren">(</span><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">&#x2192;</span> <span class="sig-return-typehint"><span class="pre">None</span></span></span><a class="reference internal" href="../_modules/pinder/core/index/system.html#PinderSystem.download_entry"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#pinder.core.index.system.PinderSystem.download_entry" title="Link to this definition">¶</a></dt>
<dd><p>Downloads data associated with an entry for the PinderSystem instance.</p>
<p>It checks for the existence of PDB and Parquet files and downloads missing
files from the Pinder bucket to the local dataset root directory.</p>
</dd></dl>

<dl class="py property">
<dt class="sig sig-object py" id="pinder.core.index.system.PinderSystem.filepaths">
<em class="property"><span class="pre">property</span><span class="w"> </span></em><span class="sig-name descname"><span class="pre">filepaths</span></span><em class="property"><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="pre">dict</span><span class="p"><span class="pre">[</span></span><span class="pre">str</span><span class="p"><span class="pre">,</span></span><span class="w"> </span><span class="pre">str</span><span class="w"> </span><span class="p"><span class="pre">|</span></span><span class="w"> </span><span class="pre">None</span><span class="p"><span class="pre">]</span></span></em><a class="headerlink" href="#pinder.core.index.system.PinderSystem.filepaths" title="Link to this definition">¶</a></dt>
<dd><p>Retrieves the file paths for the structural data associated with the system.</p>
<dl class="field-list simple">
<dt class="field-odd">Returns<span class="colon">:</span></dt>
<dd class="field-odd"><p>A dictionary with keys for each structure type
(e.g., ‘holo_receptor’, ‘holo_ligand’) and values as the
corresponding file paths or None if not available.</p>
</dd>
<dt class="field-even">Return type<span class="colon">:</span></dt>
<dd class="field-even"><p>dict[str, Optional[str]]</p>
</dd>
</dl>
</dd></dl>

<dl class="py property">
<dt class="sig sig-object py" id="pinder.core.index.system.PinderSystem.metadata">
<em class="property"><span class="pre">property</span><span class="w"> </span></em><span class="sig-name descname"><span class="pre">metadata</span></span><em class="property"><span class="p"><span class="pre">:</span></span><span class="w"> </span><a class="reference internal" href="#pinder.core.index.utils.MetadataEntry" title="pinder.core.index.utils.MetadataEntry"><span class="pre">MetadataEntry</span></a><span class="w"> </span><span class="p"><span class="pre">|</span></span><span class="w"> </span><span class="pre">None</span></em><a class="headerlink" href="#pinder.core.index.system.PinderSystem.metadata" title="Link to this definition">¶</a></dt>
<dd><p>Retrieves the additional metadata associated with an IndexEntry.</p>
<dl class="field-list simple">
<dt class="field-odd">Returns<span class="colon">:</span></dt>
<dd class="field-odd"><p>A MetadataEntry object if the metadata exists, otherwise None.</p>
</dd>
<dt class="field-even">Return type<span class="colon">:</span></dt>
<dd class="field-even"><p><a class="reference internal" href="#pinder.core.index.utils.MetadataEntry" title="pinder.core.index.utils.MetadataEntry">MetadataEntry</a> | None</p>
</dd>
</dl>
</dd></dl>

<dl class="py property">
<dt class="sig sig-object py" id="pinder.core.index.system.PinderSystem.pymol_script">
<em class="property"><span class="pre">property</span><span class="w"> </span></em><span class="sig-name descname"><span class="pre">pymol_script</span></span><em class="property"><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="pre">str</span></em><a class="headerlink" href="#pinder.core.index.system.PinderSystem.pymol_script" title="Link to this definition">¶</a></dt>
<dd><p>Constructs a PyMOL script to visualize the loaded structures in the PinderSystem.</p>
<dl class="field-list simple">
<dt class="field-odd">Returns<span class="colon">:</span></dt>
<dd class="field-odd"><p>A string representing the PyMOL script commands for visualizing the structures.</p>
</dd>
<dt class="field-even">Return type<span class="colon">:</span></dt>
<dd class="field-even"><p>str</p>
</dd>
</dl>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="pinder.core.index.system.PinderSystem.load_alt_apo_structure">
<span class="sig-name descname"><span class="pre">load_alt_apo_structure</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">alt_pdbs</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">list</span><span class="p"><span class="pre">[</span></span><span class="pre">str</span><span class="p"><span class="pre">]</span></span></span></em>, <em class="sig-param"><span class="n"><span class="pre">code</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">str</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">canon_code</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">str</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">chain_id</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">str</span><span class="w"> </span><span class="p"><span class="pre">|</span></span><span class="w"> </span><span class="pre">None</span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">None</span></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">&#x2192;</span> <span class="sig-return-typehint"><a class="reference internal" href="pinder.core.loader.html#pinder.core.loader.structure.Structure" title="pinder.core.loader.structure.Structure"><span class="pre">Structure</span></a><span class="w"> </span><span class="p"><span class="pre">|</span></span><span class="w"> </span><span class="pre">None</span></span></span><a class="reference internal" href="../_modules/pinder/core/index/system.html#PinderSystem.load_alt_apo_structure"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#pinder.core.index.system.PinderSystem.load_alt_apo_structure" title="Link to this definition">¶</a></dt>
<dd><p>Loads an alternate apo structure based on the provided PDB codes, if available.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>alt_pdbs</strong> (<em>List</em><em>[</em><em>str</em><em>]</em>) – A list of alternative PDB file paths.</p></li>
<li><p><strong>code</strong> (<em>str</em>) – The specific code to identify the alternate apo structure.</p></li>
<li><p><strong>canon_code</strong> (<em>str</em>) – The canonical code for the apo structure.</p></li>
</ul>
</dd>
<dt class="field-even">Returns<span class="colon">:</span></dt>
<dd class="field-even"><p>The loaded Structure object if found, otherwise None.</p>
</dd>
<dt class="field-odd">Return type<span class="colon">:</span></dt>
<dd class="field-odd"><p>Optional[<a class="reference internal" href="pinder.core.loader.html#pinder.core.loader.structure.Structure" title="pinder.core.loader.structure.Structure">Structure</a>]</p>
</dd>
</dl>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="pinder.core.index.system.PinderSystem.load_structure">
<em class="property"><span class="pre">static</span><span class="w"> </span></em><span class="sig-name descname"><span class="pre">load_structure</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">pdb_file</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">Path</span><span class="w"> </span><span class="p"><span class="pre">|</span></span><span class="w"> </span><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">chain_id</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">str</span><span class="w"> </span><span class="p"><span class="pre">|</span></span><span class="w"> </span><span class="pre">None</span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">None</span></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">&#x2192;</span> <span class="sig-return-typehint"><a class="reference internal" href="pinder.core.loader.html#pinder.core.loader.structure.Structure" title="pinder.core.loader.structure.Structure"><span class="pre">Structure</span></a><span class="w"> </span><span class="p"><span class="pre">|</span></span><span class="w"> </span><span class="pre">None</span></span></span><a class="reference internal" href="../_modules/pinder/core/index/system.html#PinderSystem.load_structure"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#pinder.core.index.system.PinderSystem.load_structure" title="Link to this definition">¶</a></dt>
<dd><p>Loads a structure from a PDB file if it exists and is valid.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><p><strong>pdb_file</strong> (<em>Optional</em><em>[</em><em>Path</em><em>]</em>) – The file path to the PDB file.</p>
</dd>
<dt class="field-even">Returns<span class="colon">:</span></dt>
<dd class="field-even"><p>The loaded Structure object if the file is valid, otherwise None.</p>
</dd>
<dt class="field-odd">Return type<span class="colon">:</span></dt>
<dd class="field-odd"><p>Optional[<a class="reference internal" href="pinder.core.loader.html#pinder.core.loader.structure.Structure" title="pinder.core.loader.structure.Structure">Structure</a>]</p>
</dd>
</dl>
</dd></dl>

</dd></dl>

</section>
<section id="module-pinder.core.index.utils">
<span id="pinder-core-index-utils-module"></span><h2>pinder.core.index.utils module<a class="headerlink" href="#module-pinder.core.index.utils" title="Link to this heading">¶</a></h2>
<dl class="py function">
<dt class="sig sig-object py" id="pinder.core.index.utils.get_pinder_location">
<span class="sig-prename descclassname"><span class="pre">pinder.core.index.utils.</span></span><span class="sig-name descname"><span class="pre">get_pinder_location</span></span><span class="sig-paren">(</span><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">&#x2192;</span> <span class="sig-return-typehint"><span class="pre">Path</span></span></span><a class="reference internal" href="../_modules/pinder/core/index/utils.html#get_pinder_location"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#pinder.core.index.utils.get_pinder_location" title="Link to this definition">¶</a></dt>
<dd><p>Determines the base directory location for the Pinder data.</p>
<p>First, check the environment for PINDER_DATA_DIR. If provided, assume
it is the full path to the directory containing the index, pdbs/, and mappings/.
Otherwise, the PINDER_BASE_DIR environment variable is checked, and if unset,
falls back to the default XDG_DATA_HOME location (~/.local/share), and
appends the current version of the Pinder release. The Pinder release can be
controlled via the PINDER_RELEASE environment variable.</p>
<dl class="field-list simple">
<dt class="field-odd">Returns<span class="colon">:</span></dt>
<dd class="field-odd"><p>The path to the base directory for Pinder data.</p>
</dd>
<dt class="field-even">Return type<span class="colon">:</span></dt>
<dd class="field-even"><p>Path</p>
</dd>
</dl>
</dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="pinder.core.index.utils.get_pinder_bucket_root">
<span class="sig-prename descclassname"><span class="pre">pinder.core.index.utils.</span></span><span class="sig-name descname"><span class="pre">get_pinder_bucket_root</span></span><span class="sig-paren">(</span><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">&#x2192;</span> <span class="sig-return-typehint"><span class="pre">str</span></span></span><a class="reference internal" href="../_modules/pinder/core/index/utils.html#get_pinder_bucket_root"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#pinder.core.index.utils.get_pinder_bucket_root" title="Link to this definition">¶</a></dt>
<dd><p>Constructs the root bucket path for the Pinder data in Google Cloud Storage
based on the PINDER_RELEASE environment variable.</p>
<dl class="field-list simple">
<dt class="field-odd">Returns<span class="colon">:</span></dt>
<dd class="field-odd"><p>The root bucket path as a string.</p>
</dd>
<dt class="field-even">Return type<span class="colon">:</span></dt>
<dd class="field-even"><p>str</p>
</dd>
</dl>
</dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="pinder.core.index.utils.get_index_location">
<span class="sig-prename descclassname"><span class="pre">pinder.core.index.utils.</span></span><span class="sig-name descname"><span class="pre">get_index_location</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">csv_name</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">str</span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">'index.parquet'</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">remote</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">bool</span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">False</span></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">&#x2192;</span> <span class="sig-return-typehint"><span class="pre">Path</span><span class="w"> </span><span class="p"><span class="pre">|</span></span><span class="w"> </span><span class="pre">str</span></span></span><a class="reference internal" href="../_modules/pinder/core/index/utils.html#get_index_location"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#pinder.core.index.utils.get_index_location" title="Link to this definition">¶</a></dt>
<dd><p>Gets the file path for the Pinder index CSV/Parquet file, either locally or remotely.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>csv_name</strong> (<em>str</em>) – The name of the CSV/Parquet file. Defaults to “index.parquet”.</p></li>
<li><p><strong>remote</strong> (<em>bool</em>) – A flag to determine if the remote file path should be returned.
Defaults to False.</p></li>
</ul>
</dd>
<dt class="field-even">Returns<span class="colon">:</span></dt>
<dd class="field-even"><p>The path to the index CSV/Parquet file, either as a Path object (local)
or a string (remote).</p>
</dd>
<dt class="field-odd">Return type<span class="colon">:</span></dt>
<dd class="field-odd"><p>Union[Path, str]</p>
</dd>
</dl>
</dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="pinder.core.index.utils.get_index">
<span class="sig-prename descclassname"><span class="pre">pinder.core.index.utils.</span></span><span class="sig-name descname"><span class="pre">get_index</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">csv_name</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">str</span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">'index.parquet'</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">update</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">bool</span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">False</span></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">&#x2192;</span> <span class="sig-return-typehint"><span class="pre">DataFrame</span></span></span><a class="reference internal" href="../_modules/pinder/core/index/utils.html#get_index"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#pinder.core.index.utils.get_index" title="Link to this definition">¶</a></dt>
<dd><p>Retrieves the Pinder index as a pandas DataFrame. If the index is not already
loaded, it reads from the local CSV/Parquet file, or downloads it if not present.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>csv_name</strong> (<em>str</em>) – The name of the CSV/Parquet file to load. Defaults to “index.parquet”.</p></li>
<li><p><strong>update</strong> (<em>bool</em>) – Whether to force update index on disk even if it exists.
Default is False.</p></li>
</ul>
</dd>
<dt class="field-even">Returns<span class="colon">:</span></dt>
<dd class="field-even"><p>The Pinder index as a DataFrame.</p>
</dd>
<dt class="field-odd">Return type<span class="colon">:</span></dt>
<dd class="field-odd"><p>pd.DataFrame</p>
</dd>
</dl>
</dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="pinder.core.index.utils.get_metadata">
<span class="sig-prename descclassname"><span class="pre">pinder.core.index.utils.</span></span><span class="sig-name descname"><span class="pre">get_metadata</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">csv_name</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">str</span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">'metadata.parquet'</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">update</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">bool</span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">False</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">extra_glob</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">str</span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">'metadata-*.csv.gz'</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">extra_data</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference internal" href="#pinder.core.index.utils.SupplementaryData" title="pinder.core.index.utils.SupplementaryData"><span class="pre">SupplementaryData</span></a><span class="w"> </span><span class="p"><span class="pre">|</span></span><span class="w"> </span><span class="pre">tuple</span><span class="p"><span class="pre">[</span></span><a class="reference internal" href="#pinder.core.index.utils.SupplementaryData" title="pinder.core.index.utils.SupplementaryData"><span class="pre">SupplementaryData</span></a><span class="p"><span class="pre">]</span></span><span class="w"> </span><span class="p"><span class="pre">|</span></span><span class="w"> </span><span class="pre">tuple</span><span class="p"><span class="pre">[</span></span><span class="p"><span class="pre">(</span></span><span class="p"><span class="pre">)</span></span><span class="p"><span class="pre">]</span></span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">()</span></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">&#x2192;</span> <span class="sig-return-typehint"><span class="pre">DataFrame</span></span></span><a class="reference internal" href="../_modules/pinder/core/index/utils.html#get_metadata"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#pinder.core.index.utils.get_metadata" title="Link to this definition">¶</a></dt>
<dd><p>Retrieves the Pinder index metadata as a pandas DataFrame.</p>
<p>If the metadata is not already loaded, it reads from the local CSV/Parquet file,
or downloads it if not present.</p>
<p>Also attempts to read extra metadata. We assume that all extra CSV
files contain an <cite>id</cite> field.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>csv_name</strong> (<em>str</em>) – The name of the CSV file to load. Defaults to “metadata.parquet”.</p></li>
<li><p><strong>update</strong> (<em>bool</em>) – Whether to force update index on disk even if it exists. Default is False.</p></li>
<li><p><strong>extra_glob</strong> (<em>str</em>) – The pattern to match extra metadata CSV files. Defaults to “metadata-*.csv.gz”</p></li>
</ul>
</dd>
<dt class="field-even">Returns<span class="colon">:</span></dt>
<dd class="field-even"><p>The Pinder metadata as a DataFrame.</p>
</dd>
<dt class="field-odd">Return type<span class="colon">:</span></dt>
<dd class="field-odd"><p>pd.DataFrame</p>
</dd>
</dl>
</dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="pinder.core.index.utils.get_extra_metadata">
<span class="sig-prename descclassname"><span class="pre">pinder.core.index.utils.</span></span><span class="sig-name descname"><span class="pre">get_extra_metadata</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">local_location</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">str</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">remote_location</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">str</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">glob_pattern</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">str</span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">'metadata-*.csv.gz'</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">extra_data</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">tuple</span><span class="p"><span class="pre">[</span></span><a class="reference internal" href="#pinder.core.index.utils.SupplementaryData" title="pinder.core.index.utils.SupplementaryData"><span class="pre">SupplementaryData</span></a><span class="p"><span class="pre">]</span></span><span class="w"> </span><span class="p"><span class="pre">|</span></span><span class="w"> </span><span class="pre">tuple</span><span class="p"><span class="pre">[</span></span><span class="p"><span class="pre">(</span></span><span class="p"><span class="pre">)</span></span><span class="p"><span class="pre">]</span></span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">()</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">update</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">bool</span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">False</span></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">&#x2192;</span> <span class="sig-return-typehint"><span class="pre">DataFrame</span><span class="w"> </span><span class="p"><span class="pre">|</span></span><span class="w"> </span><span class="pre">None</span></span></span><a class="reference internal" href="../_modules/pinder/core/index/utils.html#get_extra_metadata"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#pinder.core.index.utils.get_extra_metadata" title="Link to this definition">¶</a></dt>
<dd><p>Retrieves extra metadata as a pandas DataFrame.</p>
<p>If the metadata is not already loaded, it reads from local CSV files,
or downloads them if not present.</p>
<p>We assume that all CSV files contain an <cite>id</cite> field.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>local_location</strong> (<em>str</em>) – The filepath to the local directory containing CSV files.</p></li>
<li><p><strong>remote_location</strong> (<em>str</em>) – The filepath to the remote location containing CSV files.</p></li>
<li><p><strong>glob_pattern</strong> (<em>str</em>) – The pattern to match extra metadata CSV files. Defaults to
“metadata-*.csv.gz”</p></li>
<li><p><strong>update</strong> (<em>bool</em>) – Whether to force update index on disk even if it exists.
Default is False.</p></li>
</ul>
</dd>
<dt class="field-even">Returns<span class="colon">:</span></dt>
<dd class="field-even"><p>The Pinder metadata as a DataFrame.</p>
</dd>
<dt class="field-odd">Return type<span class="colon">:</span></dt>
<dd class="field-odd"><p>pd.DataFrame</p>
</dd>
</dl>
</dd></dl>

<dl class="py class">
<dt class="sig sig-object py" id="pinder.core.index.utils.SupplementaryData">
<em class="property"><span class="pre">class</span><span class="w"> </span></em><span class="sig-prename descclassname"><span class="pre">pinder.core.index.utils.</span></span><span class="sig-name descname"><span class="pre">SupplementaryData</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">value</span></span></em><span class="sig-paren">)</span><a class="reference internal" href="../_modules/pinder/core/index/utils.html#SupplementaryData"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#pinder.core.index.utils.SupplementaryData" title="Link to this definition">¶</a></dt>
<dd><p>Bases: <code class="xref py py-class docutils literal notranslate"><span class="pre">str</span></code>, <code class="xref py py-class docutils literal notranslate"><span class="pre">Enum</span></code></p>
<p>An enumeration.</p>
<dl class="py attribute">
<dt class="sig sig-object py" id="pinder.core.index.utils.SupplementaryData.sequence_database">
<span class="sig-name descname"><span class="pre">sequence_database</span></span><em class="property"><span class="w"> </span><span class="p"><span class="pre">=</span></span><span class="w"> </span><span class="pre">'sequence_database.parquet'</span></em><a class="headerlink" href="#pinder.core.index.utils.SupplementaryData.sequence_database" title="Link to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="py attribute">
<dt class="sig sig-object py" id="pinder.core.index.utils.SupplementaryData.supplementary_metadata">
<span class="sig-name descname"><span class="pre">supplementary_metadata</span></span><em class="property"><span class="w"> </span><span class="p"><span class="pre">=</span></span><span class="w"> </span><span class="pre">'supplementary_metadata.parquet'</span></em><a class="headerlink" href="#pinder.core.index.utils.SupplementaryData.supplementary_metadata" title="Link to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="py attribute">
<dt class="sig sig-object py" id="pinder.core.index.utils.SupplementaryData.entity_metadata">
<span class="sig-name descname"><span class="pre">entity_metadata</span></span><em class="property"><span class="w"> </span><span class="p"><span class="pre">=</span></span><span class="w"> </span><span class="pre">'entity_metadata.parquet'</span></em><a class="headerlink" href="#pinder.core.index.utils.SupplementaryData.entity_metadata" title="Link to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="py attribute">
<dt class="sig sig-object py" id="pinder.core.index.utils.SupplementaryData.chain_metadata">
<span class="sig-name descname"><span class="pre">chain_metadata</span></span><em class="property"><span class="w"> </span><span class="p"><span class="pre">=</span></span><span class="w"> </span><span class="pre">'chain_metadata.parquet'</span></em><a class="headerlink" href="#pinder.core.index.utils.SupplementaryData.chain_metadata" title="Link to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="py attribute">
<dt class="sig sig-object py" id="pinder.core.index.utils.SupplementaryData.ecod_metadata">
<span class="sig-name descname"><span class="pre">ecod_metadata</span></span><em class="property"><span class="w"> </span><span class="p"><span class="pre">=</span></span><span class="w"> </span><span class="pre">'ecod_metadata.parquet'</span></em><a class="headerlink" href="#pinder.core.index.utils.SupplementaryData.ecod_metadata" title="Link to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="py attribute">
<dt class="sig sig-object py" id="pinder.core.index.utils.SupplementaryData.enzyme_classification_metadata">
<span class="sig-name descname"><span class="pre">enzyme_classification_metadata</span></span><em class="property"><span class="w"> </span><span class="p"><span class="pre">=</span></span><span class="w"> </span><span class="pre">'enzyme_classification_metadata.parquet'</span></em><a class="headerlink" href="#pinder.core.index.utils.SupplementaryData.enzyme_classification_metadata" title="Link to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="py attribute">
<dt class="sig sig-object py" id="pinder.core.index.utils.SupplementaryData.interface_annotations">
<span class="sig-name descname"><span class="pre">interface_annotations</span></span><em class="property"><span class="w"> </span><span class="p"><span class="pre">=</span></span><span class="w"> </span><span class="pre">'interface_annotations.parquet'</span></em><a class="headerlink" href="#pinder.core.index.utils.SupplementaryData.interface_annotations" title="Link to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="py attribute">
<dt class="sig sig-object py" id="pinder.core.index.utils.SupplementaryData.sabdab_metadata">
<span class="sig-name descname"><span class="pre">sabdab_metadata</span></span><em class="property"><span class="w"> </span><span class="p"><span class="pre">=</span></span><span class="w"> </span><span class="pre">'sabdab_metadata.parquet'</span></em><a class="headerlink" href="#pinder.core.index.utils.SupplementaryData.sabdab_metadata" title="Link to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="py attribute">
<dt class="sig sig-object py" id="pinder.core.index.utils.SupplementaryData.monomer_neff">
<span class="sig-name descname"><span class="pre">monomer_neff</span></span><em class="property"><span class="w"> </span><span class="p"><span class="pre">=</span></span><span class="w"> </span><span class="pre">'monomer_neff.parquet'</span></em><a class="headerlink" href="#pinder.core.index.utils.SupplementaryData.monomer_neff" title="Link to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="py attribute">
<dt class="sig sig-object py" id="pinder.core.index.utils.SupplementaryData.paired_neff">
<span class="sig-name descname"><span class="pre">paired_neff</span></span><em class="property"><span class="w"> </span><span class="p"><span class="pre">=</span></span><span class="w"> </span><span class="pre">'test_split_paired_neffs.parquet'</span></em><a class="headerlink" href="#pinder.core.index.utils.SupplementaryData.paired_neff" title="Link to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="py attribute">
<dt class="sig sig-object py" id="pinder.core.index.utils.SupplementaryData.transient_interface_metadata">
<span class="sig-name descname"><span class="pre">transient_interface_metadata</span></span><em class="property"><span class="w"> </span><span class="p"><span class="pre">=</span></span><span class="w"> </span><span class="pre">'transient_interface_metadata.parquet'</span></em><a class="headerlink" href="#pinder.core.index.utils.SupplementaryData.transient_interface_metadata" title="Link to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="py attribute">
<dt class="sig sig-object py" id="pinder.core.index.utils.SupplementaryData.ialign_split_similarity_labels">
<span class="sig-name descname"><span class="pre">ialign_split_similarity_labels</span></span><em class="property"><span class="w"> </span><span class="p"><span class="pre">=</span></span><span class="w"> </span><span class="pre">'ialign_split_similarity_labels.parquet'</span></em><a class="headerlink" href="#pinder.core.index.utils.SupplementaryData.ialign_split_similarity_labels" title="Link to this definition">¶</a></dt>
<dd></dd></dl>

</dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="pinder.core.index.utils.get_supplementary_data">
<span class="sig-prename descclassname"><span class="pre">pinder.core.index.utils.</span></span><span class="sig-name descname"><span class="pre">get_supplementary_data</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">supplementary_data</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference internal" href="#pinder.core.index.utils.SupplementaryData" title="pinder.core.index.utils.SupplementaryData"><span class="pre">SupplementaryData</span></a></span></em>, <em class="sig-param"><span class="n"><span class="pre">update</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">bool</span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">False</span></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">&#x2192;</span> <span class="sig-return-typehint"><span class="pre">DataFrame</span></span></span><a class="reference internal" href="../_modules/pinder/core/index/utils.html#get_supplementary_data"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#pinder.core.index.utils.get_supplementary_data" title="Link to this definition">¶</a></dt>
<dd><p>Retrieves supplementary data file exposed in the Pinder dataset as a pandas DataFrame.
If the data is not already loaded, it reads from the local Parquet file, or downloads it if not present.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>supplementary_data</strong> (<a class="reference internal" href="#pinder.core.index.utils.SupplementaryData" title="pinder.core.index.utils.SupplementaryData"><em>SupplementaryData</em></a>) – The name of the Parquet file to load.</p></li>
<li><p><strong>update</strong> (<em>bool</em>) – Whether to force update data on disk even if it exists.
Default is False.</p></li>
</ul>
</dd>
<dt class="field-even">Returns<span class="colon">:</span></dt>
<dd class="field-even"><p>The requested supplementary Pinder data as a DataFrame.</p>
</dd>
<dt class="field-odd">Return type<span class="colon">:</span></dt>
<dd class="field-odd"><p>pd.DataFrame</p>
</dd>
</dl>
</dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="pinder.core.index.utils.get_sequence_database">
<span class="sig-prename descclassname"><span class="pre">pinder.core.index.utils.</span></span><span class="sig-name descname"><span class="pre">get_sequence_database</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">pqt_name</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">str</span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">'sequence_database.parquet'</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">update</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">bool</span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">False</span></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">&#x2192;</span> <span class="sig-return-typehint"><span class="pre">DataFrame</span></span></span><a class="reference internal" href="../_modules/pinder/core/index/utils.html#get_sequence_database"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#pinder.core.index.utils.get_sequence_database" title="Link to this definition">¶</a></dt>
<dd><p>Retrieves sequences for all PDB files in the Pinder dataset (including dimers, bound and unbound monomers) as a pandas DataFrame.
If the database is not already loaded, it reads from the local Parquet file, or downloads it if not present.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>pqt_name</strong> (<em>str</em>) – The name of the Parquet file to load. Defaults to “sequence_database.parquet”.</p></li>
<li><p><strong>update</strong> (<em>bool</em>) – Whether to force update sequence database on disk even if it exists.
Default is False.</p></li>
</ul>
</dd>
<dt class="field-even">Returns<span class="colon">:</span></dt>
<dd class="field-even"><p>The Pinder sequence database as a DataFrame.</p>
</dd>
<dt class="field-odd">Return type<span class="colon">:</span></dt>
<dd class="field-odd"><p>pd.DataFrame</p>
</dd>
</dl>
</dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="pinder.core.index.utils.download_dataset">
<span class="sig-prename descclassname"><span class="pre">pinder.core.index.utils.</span></span><span class="sig-name descname"><span class="pre">download_dataset</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">skip_inflation</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">bool</span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">False</span></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">&#x2192;</span> <span class="sig-return-typehint"><span class="pre">None</span></span></span><a class="reference internal" href="../_modules/pinder/core/index/utils.html#download_dataset"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#pinder.core.index.utils.download_dataset" title="Link to this definition">¶</a></dt>
<dd><p>Downloads the Pinder dataset archives (PDBs and mappings) and optionally inflates them.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><p><strong>skip_inflation</strong> (<em>bool</em>) – If True, the method will skip inflating (unzipping) the
downloaded archives. Defaults to False.</p>
</dd>
</dl>
<div class="admonition note">
<p class="admonition-title">Note</p>
<p>Required disk space to download the full Pinder dataset:</p>
<div class="highlight-text notranslate"><div class="highlight"><pre><span></span># compressed
144G    pdbs.zip
149M    test_set_pdbs.zip
6.8G    mappings.zip

# unpacked
672G    pdbs
705M    test_set_pdbs
25G     mappings
</pre></div>
</div>
</div>
</dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="pinder.core.index.utils.get_arg_parser_args">
<span class="sig-prename descclassname"><span class="pre">pinder.core.index.utils.</span></span><span class="sig-name descname"><span class="pre">get_arg_parser_args</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">argv</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">list</span><span class="p"><span class="pre">[</span></span><span class="pre">str</span><span class="p"><span class="pre">]</span></span><span class="w"> </span><span class="p"><span class="pre">|</span></span><span class="w"> </span><span class="pre">None</span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">title</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">str</span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">'Download</span> <span class="pre">latest</span> <span class="pre">pinder</span> <span class="pre">dataset</span> <span class="pre">to</span> <span class="pre">disk'</span></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">&#x2192;</span> <span class="sig-return-typehint"><span class="pre">dict</span><span class="p"><span class="pre">[</span></span><span class="pre">str</span><span class="p"><span class="pre">,</span></span><span class="w"> </span><span class="pre">str</span><span class="w"> </span><span class="p"><span class="pre">|</span></span><span class="w"> </span><span class="pre">bool</span><span class="p"><span class="pre">]</span></span></span></span><a class="reference internal" href="../_modules/pinder/core/index/utils.html#get_arg_parser_args"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#pinder.core.index.utils.get_arg_parser_args" title="Link to this definition">¶</a></dt>
<dd><p>The command-line arg parser for different pinder data downloads and updates.</p>
<p>It accepts command-line arguments to specify the base directory, release version,
and whether to skip inflating the compressed archives.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><p><strong>argv</strong> (<em>Optional</em><em>[</em><em>List</em><em>[</em><em>str</em><em>]</em><em>]</em>) – The command-line arguments. If None, sys.argv is used.</p>
</dd>
</dl>
</dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="pinder.core.index.utils.download_pinder_cmd">
<span class="sig-prename descclassname"><span class="pre">pinder.core.index.utils.</span></span><span class="sig-name descname"><span class="pre">download_pinder_cmd</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">argv</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">list</span><span class="p"><span class="pre">[</span></span><span class="pre">str</span><span class="p"><span class="pre">]</span></span><span class="w"> </span><span class="p"><span class="pre">|</span></span><span class="w"> </span><span class="pre">None</span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">None</span></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">&#x2192;</span> <span class="sig-return-typehint"><span class="pre">None</span></span></span><a class="reference internal" href="../_modules/pinder/core/index/utils.html#download_pinder_cmd"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#pinder.core.index.utils.download_pinder_cmd" title="Link to this definition">¶</a></dt>
<dd><p>The command-line interface for downloading the latest Pinder dataset to disk.</p>
<p>It accepts command-line arguments to specify the base directory, release version,
and whether to skip inflating the compressed archives.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><p><strong>argv</strong> (<em>Optional</em><em>[</em><em>List</em><em>[</em><em>str</em><em>]</em><em>]</em>) – The command-line arguments. If None, sys.argv is used.</p>
</dd>
</dl>
<div class="admonition note">
<p class="admonition-title">Note</p>
<p>Required disk space to download the full Pinder dataset:</p>
<div class="highlight-text notranslate"><div class="highlight"><pre><span></span># compressed
144G    pdbs.zip
149M    test_set_pdbs.zip
6.8G    mappings.zip

# unpacked
672G    pdbs
705M    test_set_pdbs
25G     mappings
</pre></div>
</div>
</div>
</dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="pinder.core.index.utils.update_index_cmd">
<span class="sig-prename descclassname"><span class="pre">pinder.core.index.utils.</span></span><span class="sig-name descname"><span class="pre">update_index_cmd</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">argv</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">list</span><span class="p"><span class="pre">[</span></span><span class="pre">str</span><span class="p"><span class="pre">]</span></span><span class="w"> </span><span class="p"><span class="pre">|</span></span><span class="w"> </span><span class="pre">None</span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">None</span></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">&#x2192;</span> <span class="sig-return-typehint"><span class="pre">None</span></span></span><a class="reference internal" href="../_modules/pinder/core/index/utils.html#update_index_cmd"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#pinder.core.index.utils.update_index_cmd" title="Link to this definition">¶</a></dt>
<dd><p>The command-line interface for downloading the latest Pinder index to disk.</p>
<p>It accepts command-line arguments to specify the base directory and release version.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><p><strong>argv</strong> (<em>Optional</em><em>[</em><em>List</em><em>[</em><em>str</em><em>]</em><em>]</em>) – The command-line arguments. If None, sys.argv is used.</p>
</dd>
</dl>
</dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="pinder.core.index.utils.get_missing_blobs">
<span class="sig-prename descclassname"><span class="pre">pinder.core.index.utils.</span></span><span class="sig-name descname"><span class="pre">get_missing_blobs</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">prefix</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">str</span></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">&#x2192;</span> <span class="sig-return-typehint"><span class="pre">tuple</span><span class="p"><span class="pre">[</span></span><span class="pre">list</span><span class="p"><span class="pre">[</span></span><span class="pre">str</span><span class="p"><span class="pre">]</span></span><span class="p"><span class="pre">,</span></span><span class="w"> </span><span class="pre">list</span><span class="p"><span class="pre">[</span></span><span class="pre">Path</span><span class="p"><span class="pre">]</span></span><span class="p"><span class="pre">]</span></span></span></span><a class="reference internal" href="../_modules/pinder/core/index/utils.html#get_missing_blobs"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#pinder.core.index.utils.get_missing_blobs" title="Link to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="pinder.core.index.utils.sync_pinder_structure_data">
<span class="sig-prename descclassname"><span class="pre">pinder.core.index.utils.</span></span><span class="sig-name descname"><span class="pre">sync_pinder_structure_data</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">argv</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">list</span><span class="p"><span class="pre">[</span></span><span class="pre">str</span><span class="p"><span class="pre">]</span></span><span class="w"> </span><span class="p"><span class="pre">|</span></span><span class="w"> </span><span class="pre">None</span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">None</span></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">&#x2192;</span> <span class="sig-return-typehint"><span class="pre">None</span></span></span><a class="reference internal" href="../_modules/pinder/core/index/utils.html#sync_pinder_structure_data"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#pinder.core.index.utils.sync_pinder_structure_data" title="Link to this definition">¶</a></dt>
<dd><p>The command-line interface for syncing any structural data files missing on disk.</p>
<p>It accepts command-line arguments to specify the base directory and release version.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><p><strong>argv</strong> (<em>Optional</em><em>[</em><em>List</em><em>[</em><em>str</em><em>]</em><em>]</em>) – The command-line arguments. If None, sys.argv is used.</p>
</dd>
</dl>
</dd></dl>

<dl class="py class">
<dt class="sig sig-object py" id="pinder.core.index.utils.IndexEntry">
<em class="property"><span class="pre">class</span><span class="w"> </span></em><span class="sig-prename descclassname"><span class="pre">pinder.core.index.utils.</span></span><span class="sig-name descname"><span class="pre">IndexEntry</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="o"><span class="pre">*</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">split</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">str</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">id</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">str</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">pdb_id</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">str</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">cluster_id</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">str</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">cluster_id_R</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">str</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">cluster_id_L</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">str</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">pinder_s</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">bool</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">pinder_xl</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">bool</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">pinder_af2</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">bool</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">uniprot_R</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">str</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">uniprot_L</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">str</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">holo_R_pdb</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">str</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">holo_L_pdb</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">str</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">predicted_R_pdb</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">str</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">predicted_L_pdb</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">str</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">apo_R_pdb</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">str</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">apo_L_pdb</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">str</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">apo_R_pdbs</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">str</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">apo_L_pdbs</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">str</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">holo_R</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">bool</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">holo_L</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">bool</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">predicted_R</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">bool</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">predicted_L</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">bool</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">apo_R</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">bool</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">apo_L</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">bool</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">apo_R_quality</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">str</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">apo_L_quality</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">str</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">chain1_neff</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">float</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">chain2_neff</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">float</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">chain_R</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">str</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">chain_L</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">str</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">contains_antibody</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">bool</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">contains_antigen</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">bool</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">contains_enzyme</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">bool</span></span></em><span class="sig-paren">)</span><a class="reference internal" href="../_modules/pinder/core/index/utils.html#IndexEntry"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#pinder.core.index.utils.IndexEntry" title="Link to this definition">¶</a></dt>
<dd><p>Bases: <code class="xref py py-class docutils literal notranslate"><span class="pre">BaseModel</span></code></p>
<p>Pydantic model representing a single entry in the Pinder index.</p>
<p>Stores all associated metadata for a particular dataset entry as attributes.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><dl class="simple">
<dt><strong>split (str):</strong></dt><dd><p>The type of data split (e.g., ‘train’, ‘test’).</p>
</dd>
<dt><strong>id (str):</strong></dt><dd><p>The unique identifier for the dataset entry.</p>
</dd>
<dt><strong>pdb_id (str):</strong></dt><dd><p>The PDB identifier associated with the entry.</p>
</dd>
<dt><strong>cluster_id (str):</strong></dt><dd><p>The cluster identifier associated with the entry.</p>
</dd>
<dt><strong>cluster_id_R (str):</strong></dt><dd><p>The cluster identifier associated with receptor dimer body.</p>
</dd>
<dt><strong>cluster_id_L (str):</strong></dt><dd><p>The cluster identifier associated with ligand dimer body.</p>
</dd>
<dt><strong>pinder_s (bool):</strong></dt><dd><p>Flag indicating if the entry is part of the Pinder-S dataset.</p>
</dd>
<dt><strong>pinder_xl (bool):</strong></dt><dd><p>Flag indicating if the entry is part of the Pinder-XL dataset.</p>
</dd>
<dt><strong>pinder_af2 (bool):</strong></dt><dd><p>Flag indicating if the entry is part of the Pinder-AF2 dataset.</p>
</dd>
<dt><strong>uniprot_R (str):</strong></dt><dd><p>The UniProt identifier for the receptor protein.</p>
</dd>
<dt><strong>uniprot_L (str):</strong></dt><dd><p>The UniProt identifier for the ligand protein.</p>
</dd>
<dt><strong>holo_R_pdb (str):</strong></dt><dd><p>The PDB identifier for the holo form of the receptor protein.</p>
</dd>
<dt><strong>holo_L_pdb (str):</strong></dt><dd><p>The PDB identifier for the holo form of the ligand protein.</p>
</dd>
<dt><strong>predicted_R_pdb (str):</strong></dt><dd><p>The PDB identifier for the predicted structure of the receptor protein.</p>
</dd>
<dt><strong>predicted_L_pdb (str):</strong></dt><dd><p>The PDB identifier for the predicted structure of the ligand protein.</p>
</dd>
<dt><strong>apo_R_pdb (str):</strong></dt><dd><p>The PDB identifier for the apo form of the receptor protein.</p>
</dd>
<dt><strong>apo_L_pdb (str):</strong></dt><dd><p>The PDB identifier for the apo form of the ligand protein.</p>
</dd>
<dt><strong>apo_R_pdbs (str):</strong></dt><dd><p>The PDB identifiers for the apo forms of the receptor protein.</p>
</dd>
<dt><strong>apo_L_pdbs (str):</strong></dt><dd><p>The PDB identifiers for the apo forms of the ligand protein.</p>
</dd>
<dt><strong>holo_R (bool):</strong></dt><dd><p>Flag indicating if the holo form of the receptor protein is available.</p>
</dd>
<dt><strong>holo_L (bool):</strong></dt><dd><p>Flag indicating if the holo form of the ligand protein is available.</p>
</dd>
<dt><strong>predicted_R (bool):</strong></dt><dd><p>Flag indicating if the predicted structure of the receptor protein is available.</p>
</dd>
<dt><strong>predicted_L (bool):</strong></dt><dd><p>Flag indicating if the predicted structure of the ligand protein is available.</p>
</dd>
<dt><strong>apo_R (bool):</strong></dt><dd><p>Flag indicating if the apo form of the receptor protein is available.</p>
</dd>
<dt><strong>apo_L (bool):</strong></dt><dd><p>Flag indicating if the apo form of the ligand protein is available.</p>
</dd>
<dt><strong>apo_R_quality (str):</strong></dt><dd><p>Classification of apo receptor pairing quality. Can be <cite>high, low, ‘’</cite>.
All test and val are labeled high. Train split is broken into <cite>high</cite> and <cite>low</cite>,
depending on whether the pairing was produced with a low-confidence quality/eval metrics
or <cite>high</cite> if the same metrics were used as for train and val.
If no pairing exists, it is labeled with an empty string.</p>
</dd>
<dt><strong>apo_L_quality (str):</strong></dt><dd><p>Classification of apo ligand pairing quality. Can be <cite>high, low, ‘’</cite>.
All test and val are labeled high. Train split is broken into <cite>high</cite> and <cite>low</cite>,
depending on whether the pairing was produced with a low-confidence quality/eval metrics
or <cite>high</cite> if the same metrics were used as for train and val.
If no pairing exists, it is labeled with an empty string.</p>
</dd>
<dt><strong>chain1_neff (float):</strong></dt><dd><p>The Neff value for the first chain in the protein complex.</p>
</dd>
<dt><strong>chain2_neff (float):</strong></dt><dd><p>The Neff value for the second chain in the protein complex.</p>
</dd>
<dt><strong>chain_R (str):</strong></dt><dd><p>The chain identifier for the receptor protein.</p>
</dd>
<dt><strong>chain_L (str):</strong></dt><dd><p>The chain identifier for the ligand protein.</p>
</dd>
<dt><strong>contains_antibody (bool):</strong></dt><dd><p>Flag indicating if the protein complex contains an antibody as per SAbDab.</p>
</dd>
<dt><strong>contains_antigen (bool):</strong></dt><dd><p>Flag indicating if the protein complex contains an antigen as per SAbDab.</p>
</dd>
<dt><strong>contains_enzyme (bool):</strong></dt><dd><p>Flag indicating if the protein complex contains an enzyme as per EC ID number.</p>
</dd>
</dl>
</dd>
</dl>
<dl class="py attribute">
<dt class="sig sig-object py" id="pinder.core.index.utils.IndexEntry.model_config">
<span class="sig-name descname"><span class="pre">model_config</span></span><em class="property"><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="pre">ClassVar</span><span class="p"><span class="pre">[</span></span><span class="pre">ConfigDict</span><span class="p"><span class="pre">]</span></span></em><em class="property"><span class="w"> </span><span class="p"><span class="pre">=</span></span><span class="w"> </span><span class="pre">{'extra':</span> <span class="pre">'forbid'}</span></em><a class="headerlink" href="#pinder.core.index.utils.IndexEntry.model_config" title="Link to this definition">¶</a></dt>
<dd><p>Configuration for the model, should be a dictionary conforming to [<cite>ConfigDict</cite>][pydantic.config.ConfigDict].</p>
</dd></dl>

<dl class="py attribute">
<dt class="sig sig-object py" id="pinder.core.index.utils.IndexEntry.split">
<span class="sig-name descname"><span class="pre">split</span></span><em class="property"><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="pre">str</span></em><a class="headerlink" href="#pinder.core.index.utils.IndexEntry.split" title="Link to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="py attribute">
<dt class="sig sig-object py" id="pinder.core.index.utils.IndexEntry.id">
<span class="sig-name descname"><span class="pre">id</span></span><em class="property"><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="pre">str</span></em><a class="headerlink" href="#pinder.core.index.utils.IndexEntry.id" title="Link to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="py attribute">
<dt class="sig sig-object py" id="pinder.core.index.utils.IndexEntry.pdb_id">
<span class="sig-name descname"><span class="pre">pdb_id</span></span><em class="property"><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="pre">str</span></em><a class="headerlink" href="#pinder.core.index.utils.IndexEntry.pdb_id" title="Link to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="py attribute">
<dt class="sig sig-object py" id="pinder.core.index.utils.IndexEntry.cluster_id">
<span class="sig-name descname"><span class="pre">cluster_id</span></span><em class="property"><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="pre">str</span></em><a class="headerlink" href="#pinder.core.index.utils.IndexEntry.cluster_id" title="Link to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="py attribute">
<dt class="sig sig-object py" id="pinder.core.index.utils.IndexEntry.cluster_id_R">
<span class="sig-name descname"><span class="pre">cluster_id_R</span></span><em class="property"><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="pre">str</span></em><a class="headerlink" href="#pinder.core.index.utils.IndexEntry.cluster_id_R" title="Link to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="py attribute">
<dt class="sig sig-object py" id="pinder.core.index.utils.IndexEntry.cluster_id_L">
<span class="sig-name descname"><span class="pre">cluster_id_L</span></span><em class="property"><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="pre">str</span></em><a class="headerlink" href="#pinder.core.index.utils.IndexEntry.cluster_id_L" title="Link to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="py attribute">
<dt class="sig sig-object py" id="pinder.core.index.utils.IndexEntry.pinder_s">
<span class="sig-name descname"><span class="pre">pinder_s</span></span><em class="property"><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="pre">bool</span></em><a class="headerlink" href="#pinder.core.index.utils.IndexEntry.pinder_s" title="Link to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="py attribute">
<dt class="sig sig-object py" id="pinder.core.index.utils.IndexEntry.pinder_xl">
<span class="sig-name descname"><span class="pre">pinder_xl</span></span><em class="property"><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="pre">bool</span></em><a class="headerlink" href="#pinder.core.index.utils.IndexEntry.pinder_xl" title="Link to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="py attribute">
<dt class="sig sig-object py" id="pinder.core.index.utils.IndexEntry.pinder_af2">
<span class="sig-name descname"><span class="pre">pinder_af2</span></span><em class="property"><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="pre">bool</span></em><a class="headerlink" href="#pinder.core.index.utils.IndexEntry.pinder_af2" title="Link to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="py attribute">
<dt class="sig sig-object py" id="pinder.core.index.utils.IndexEntry.uniprot_R">
<span class="sig-name descname"><span class="pre">uniprot_R</span></span><em class="property"><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="pre">str</span></em><a class="headerlink" href="#pinder.core.index.utils.IndexEntry.uniprot_R" title="Link to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="py attribute">
<dt class="sig sig-object py" id="pinder.core.index.utils.IndexEntry.uniprot_L">
<span class="sig-name descname"><span class="pre">uniprot_L</span></span><em class="property"><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="pre">str</span></em><a class="headerlink" href="#pinder.core.index.utils.IndexEntry.uniprot_L" title="Link to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="py attribute">
<dt class="sig sig-object py" id="pinder.core.index.utils.IndexEntry.holo_R_pdb">
<span class="sig-name descname"><span class="pre">holo_R_pdb</span></span><em class="property"><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="pre">str</span></em><a class="headerlink" href="#pinder.core.index.utils.IndexEntry.holo_R_pdb" title="Link to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="py attribute">
<dt class="sig sig-object py" id="pinder.core.index.utils.IndexEntry.holo_L_pdb">
<span class="sig-name descname"><span class="pre">holo_L_pdb</span></span><em class="property"><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="pre">str</span></em><a class="headerlink" href="#pinder.core.index.utils.IndexEntry.holo_L_pdb" title="Link to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="py attribute">
<dt class="sig sig-object py" id="pinder.core.index.utils.IndexEntry.predicted_R_pdb">
<span class="sig-name descname"><span class="pre">predicted_R_pdb</span></span><em class="property"><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="pre">str</span></em><a class="headerlink" href="#pinder.core.index.utils.IndexEntry.predicted_R_pdb" title="Link to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="py attribute">
<dt class="sig sig-object py" id="pinder.core.index.utils.IndexEntry.predicted_L_pdb">
<span class="sig-name descname"><span class="pre">predicted_L_pdb</span></span><em class="property"><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="pre">str</span></em><a class="headerlink" href="#pinder.core.index.utils.IndexEntry.predicted_L_pdb" title="Link to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="py attribute">
<dt class="sig sig-object py" id="pinder.core.index.utils.IndexEntry.apo_R_pdb">
<span class="sig-name descname"><span class="pre">apo_R_pdb</span></span><em class="property"><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="pre">str</span></em><a class="headerlink" href="#pinder.core.index.utils.IndexEntry.apo_R_pdb" title="Link to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="py attribute">
<dt class="sig sig-object py" id="pinder.core.index.utils.IndexEntry.apo_L_pdb">
<span class="sig-name descname"><span class="pre">apo_L_pdb</span></span><em class="property"><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="pre">str</span></em><a class="headerlink" href="#pinder.core.index.utils.IndexEntry.apo_L_pdb" title="Link to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="py attribute">
<dt class="sig sig-object py" id="pinder.core.index.utils.IndexEntry.apo_R_pdbs">
<span class="sig-name descname"><span class="pre">apo_R_pdbs</span></span><em class="property"><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="pre">str</span></em><a class="headerlink" href="#pinder.core.index.utils.IndexEntry.apo_R_pdbs" title="Link to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="py attribute">
<dt class="sig sig-object py" id="pinder.core.index.utils.IndexEntry.apo_L_pdbs">
<span class="sig-name descname"><span class="pre">apo_L_pdbs</span></span><em class="property"><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="pre">str</span></em><a class="headerlink" href="#pinder.core.index.utils.IndexEntry.apo_L_pdbs" title="Link to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="py attribute">
<dt class="sig sig-object py" id="pinder.core.index.utils.IndexEntry.holo_R">
<span class="sig-name descname"><span class="pre">holo_R</span></span><em class="property"><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="pre">bool</span></em><a class="headerlink" href="#pinder.core.index.utils.IndexEntry.holo_R" title="Link to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="py attribute">
<dt class="sig sig-object py" id="pinder.core.index.utils.IndexEntry.holo_L">
<span class="sig-name descname"><span class="pre">holo_L</span></span><em class="property"><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="pre">bool</span></em><a class="headerlink" href="#pinder.core.index.utils.IndexEntry.holo_L" title="Link to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="py attribute">
<dt class="sig sig-object py" id="pinder.core.index.utils.IndexEntry.predicted_R">
<span class="sig-name descname"><span class="pre">predicted_R</span></span><em class="property"><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="pre">bool</span></em><a class="headerlink" href="#pinder.core.index.utils.IndexEntry.predicted_R" title="Link to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="py attribute">
<dt class="sig sig-object py" id="pinder.core.index.utils.IndexEntry.predicted_L">
<span class="sig-name descname"><span class="pre">predicted_L</span></span><em class="property"><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="pre">bool</span></em><a class="headerlink" href="#pinder.core.index.utils.IndexEntry.predicted_L" title="Link to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="py attribute">
<dt class="sig sig-object py" id="pinder.core.index.utils.IndexEntry.apo_R">
<span class="sig-name descname"><span class="pre">apo_R</span></span><em class="property"><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="pre">bool</span></em><a class="headerlink" href="#pinder.core.index.utils.IndexEntry.apo_R" title="Link to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="py attribute">
<dt class="sig sig-object py" id="pinder.core.index.utils.IndexEntry.apo_L">
<span class="sig-name descname"><span class="pre">apo_L</span></span><em class="property"><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="pre">bool</span></em><a class="headerlink" href="#pinder.core.index.utils.IndexEntry.apo_L" title="Link to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="py attribute">
<dt class="sig sig-object py" id="pinder.core.index.utils.IndexEntry.apo_R_quality">
<span class="sig-name descname"><span class="pre">apo_R_quality</span></span><em class="property"><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="pre">str</span></em><a class="headerlink" href="#pinder.core.index.utils.IndexEntry.apo_R_quality" title="Link to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="py attribute">
<dt class="sig sig-object py" id="pinder.core.index.utils.IndexEntry.apo_L_quality">
<span class="sig-name descname"><span class="pre">apo_L_quality</span></span><em class="property"><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="pre">str</span></em><a class="headerlink" href="#pinder.core.index.utils.IndexEntry.apo_L_quality" title="Link to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="py attribute">
<dt class="sig sig-object py" id="pinder.core.index.utils.IndexEntry.chain1_neff">
<span class="sig-name descname"><span class="pre">chain1_neff</span></span><em class="property"><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="pre">float</span></em><a class="headerlink" href="#pinder.core.index.utils.IndexEntry.chain1_neff" title="Link to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="py attribute">
<dt class="sig sig-object py" id="pinder.core.index.utils.IndexEntry.chain2_neff">
<span class="sig-name descname"><span class="pre">chain2_neff</span></span><em class="property"><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="pre">float</span></em><a class="headerlink" href="#pinder.core.index.utils.IndexEntry.chain2_neff" title="Link to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="py attribute">
<dt class="sig sig-object py" id="pinder.core.index.utils.IndexEntry.chain_R">
<span class="sig-name descname"><span class="pre">chain_R</span></span><em class="property"><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="pre">str</span></em><a class="headerlink" href="#pinder.core.index.utils.IndexEntry.chain_R" title="Link to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="py attribute">
<dt class="sig sig-object py" id="pinder.core.index.utils.IndexEntry.chain_L">
<span class="sig-name descname"><span class="pre">chain_L</span></span><em class="property"><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="pre">str</span></em><a class="headerlink" href="#pinder.core.index.utils.IndexEntry.chain_L" title="Link to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="py attribute">
<dt class="sig sig-object py" id="pinder.core.index.utils.IndexEntry.contains_antibody">
<span class="sig-name descname"><span class="pre">contains_antibody</span></span><em class="property"><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="pre">bool</span></em><a class="headerlink" href="#pinder.core.index.utils.IndexEntry.contains_antibody" title="Link to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="py attribute">
<dt class="sig sig-object py" id="pinder.core.index.utils.IndexEntry.contains_antigen">
<span class="sig-name descname"><span class="pre">contains_antigen</span></span><em class="property"><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="pre">bool</span></em><a class="headerlink" href="#pinder.core.index.utils.IndexEntry.contains_antigen" title="Link to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="py attribute">
<dt class="sig sig-object py" id="pinder.core.index.utils.IndexEntry.contains_enzyme">
<span class="sig-name descname"><span class="pre">contains_enzyme</span></span><em class="property"><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="pre">bool</span></em><a class="headerlink" href="#pinder.core.index.utils.IndexEntry.contains_enzyme" title="Link to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="pinder.core.index.utils.IndexEntry.pdb_path">
<span class="sig-name descname"><span class="pre">pdb_path</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">pdb_name</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">str</span></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">&#x2192;</span> <span class="sig-return-typehint"><span class="pre">str</span></span></span><a class="reference internal" href="../_modules/pinder/core/index/utils.html#IndexEntry.pdb_path"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#pinder.core.index.utils.IndexEntry.pdb_path" title="Link to this definition">¶</a></dt>
<dd><p>Constructs the relative path for a given PDB file within the Pinder dataset.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><p><strong>pdb_name</strong> (<em>str</em>) – The name of the PDB file.</p>
</dd>
<dt class="field-even">Returns<span class="colon">:</span></dt>
<dd class="field-even"><p>The relative path as a string.</p>
</dd>
<dt class="field-odd">Return type<span class="colon">:</span></dt>
<dd class="field-odd"><p>str</p>
</dd>
</dl>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="pinder.core.index.utils.IndexEntry.mapping_path">
<span class="sig-name descname"><span class="pre">mapping_path</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">pdb_name</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">str</span></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">&#x2192;</span> <span class="sig-return-typehint"><span class="pre">str</span></span></span><a class="reference internal" href="../_modules/pinder/core/index/utils.html#IndexEntry.mapping_path"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#pinder.core.index.utils.IndexEntry.mapping_path" title="Link to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="py property">
<dt class="sig sig-object py" id="pinder.core.index.utils.IndexEntry.pdb_paths">
<em class="property"><span class="pre">property</span><span class="w"> </span></em><span class="sig-name descname"><span class="pre">pdb_paths</span></span><em class="property"><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="pre">dict</span><span class="p"><span class="pre">[</span></span><span class="pre">str</span><span class="p"><span class="pre">,</span></span><span class="w"> </span><span class="pre">str</span><span class="w"> </span><span class="p"><span class="pre">|</span></span><span class="w"> </span><span class="pre">list</span><span class="p"><span class="pre">[</span></span><span class="pre">str</span><span class="p"><span class="pre">]</span></span><span class="p"><span class="pre">]</span></span></em><a class="headerlink" href="#pinder.core.index.utils.IndexEntry.pdb_paths" title="Link to this definition">¶</a></dt>
<dd><p>Dictionary containing the PDB type as key and PDB file name as value. Alternative apo monomers (if available) are returned as a list.</p>
<dl class="field-list simple">
<dt class="field-odd">Type<span class="colon">:</span></dt>
<dd class="field-odd"><p>dict[str, str | list[str]]</p>
</dd>
</dl>
</dd></dl>

<dl class="py property">
<dt class="sig sig-object py" id="pinder.core.index.utils.IndexEntry.mapping_paths">
<em class="property"><span class="pre">property</span><span class="w"> </span></em><span class="sig-name descname"><span class="pre">mapping_paths</span></span><em class="property"><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="pre">dict</span><span class="p"><span class="pre">[</span></span><span class="pre">str</span><span class="p"><span class="pre">,</span></span><span class="w"> </span><span class="pre">str</span><span class="w"> </span><span class="p"><span class="pre">|</span></span><span class="w"> </span><span class="pre">list</span><span class="p"><span class="pre">[</span></span><span class="pre">str</span><span class="p"><span class="pre">]</span></span><span class="p"><span class="pre">]</span></span></em><a class="headerlink" href="#pinder.core.index.utils.IndexEntry.mapping_paths" title="Link to this definition">¶</a></dt>
<dd><p>Dictionary containing the PDB type as key and parquet file names corresponding to residue-level mapping information as value. are returned as a list.</p>
<dl class="field-list simple">
<dt class="field-odd">Type<span class="colon">:</span></dt>
<dd class="field-odd"><p>dict[str, str | list[str]]</p>
</dd>
</dl>
</dd></dl>

<dl class="py property">
<dt class="sig sig-object py" id="pinder.core.index.utils.IndexEntry.pinder_id">
<em class="property"><span class="pre">property</span><span class="w"> </span></em><span class="sig-name descname"><span class="pre">pinder_id</span></span><em class="property"><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="pre">str</span></em><a class="headerlink" href="#pinder.core.index.utils.IndexEntry.pinder_id" title="Link to this definition">¶</a></dt>
<dd><p>PINDER identifier for the dimer.</p>
<dl class="field-list simple">
<dt class="field-odd">Type<span class="colon">:</span></dt>
<dd class="field-odd"><p>str</p>
</dd>
</dl>
</dd></dl>

<dl class="py property">
<dt class="sig sig-object py" id="pinder.core.index.utils.IndexEntry.pinder_pdb">
<em class="property"><span class="pre">property</span><span class="w"> </span></em><span class="sig-name descname"><span class="pre">pinder_pdb</span></span><em class="property"><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="pre">str</span></em><a class="headerlink" href="#pinder.core.index.utils.IndexEntry.pinder_pdb" title="Link to this definition">¶</a></dt>
<dd><p>PINDER dimer PDB file name.</p>
<dl class="field-list simple">
<dt class="field-odd">Type<span class="colon">:</span></dt>
<dd class="field-odd"><p>str</p>
</dd>
</dl>
</dd></dl>

<dl class="py property">
<dt class="sig sig-object py" id="pinder.core.index.utils.IndexEntry.apo_R_alt">
<em class="property"><span class="pre">property</span><span class="w"> </span></em><span class="sig-name descname"><span class="pre">apo_R_alt</span></span><em class="property"><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="pre">list</span><span class="p"><span class="pre">[</span></span><span class="pre">str</span><span class="p"><span class="pre">]</span></span><span class="w"> </span><span class="p"><span class="pre">|</span></span><span class="w"> </span><span class="pre">list</span><span class="p"><span class="pre">[</span></span><span class="pre">None</span><span class="p"><span class="pre">]</span></span></em><a class="headerlink" href="#pinder.core.index.utils.IndexEntry.apo_R_alt" title="Link to this definition">¶</a></dt>
<dd><p>list of alternative apo receptor PDBs associated with entry.</p>
<dl class="field-list simple">
<dt class="field-odd">Type<span class="colon">:</span></dt>
<dd class="field-odd"><p>list[str]</p>
</dd>
</dl>
</dd></dl>

<dl class="py property">
<dt class="sig sig-object py" id="pinder.core.index.utils.IndexEntry.apo_L_alt">
<em class="property"><span class="pre">property</span><span class="w"> </span></em><span class="sig-name descname"><span class="pre">apo_L_alt</span></span><em class="property"><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="pre">list</span><span class="p"><span class="pre">[</span></span><span class="pre">str</span><span class="p"><span class="pre">]</span></span><span class="w"> </span><span class="p"><span class="pre">|</span></span><span class="w"> </span><span class="pre">list</span><span class="p"><span class="pre">[</span></span><span class="pre">None</span><span class="p"><span class="pre">]</span></span></em><a class="headerlink" href="#pinder.core.index.utils.IndexEntry.apo_L_alt" title="Link to this definition">¶</a></dt>
<dd><p>list of alternative apo ligand PDBs associated with entry.</p>
<dl class="field-list simple">
<dt class="field-odd">Type<span class="colon">:</span></dt>
<dd class="field-odd"><p>list[str]</p>
</dd>
</dl>
</dd></dl>

<dl class="py property">
<dt class="sig sig-object py" id="pinder.core.index.utils.IndexEntry.homodimer">
<em class="property"><span class="pre">property</span><span class="w"> </span></em><span class="sig-name descname"><span class="pre">homodimer</span></span><em class="property"><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="pre">bool</span></em><a class="headerlink" href="#pinder.core.index.utils.IndexEntry.homodimer" title="Link to this definition">¶</a></dt>
<dd><p>loose definition of whether the entry is a homodimer (based on identical receptor and ligand UniProt IDs).</p>
<dl class="field-list simple">
<dt class="field-odd">Type<span class="colon">:</span></dt>
<dd class="field-odd"><p>bool</p>
</dd>
</dl>
</dd></dl>

<dl class="py property">
<dt class="sig sig-object py" id="pinder.core.index.utils.IndexEntry.test_system">
<em class="property"><span class="pre">property</span><span class="w"> </span></em><span class="sig-name descname"><span class="pre">test_system</span></span><em class="property"><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="pre">bool</span></em><a class="headerlink" href="#pinder.core.index.utils.IndexEntry.test_system" title="Link to this definition">¶</a></dt>
<dd><p>whether the system is part of the test split.</p>
<dl class="field-list simple">
<dt class="field-odd">Type<span class="colon">:</span></dt>
<dd class="field-odd"><p>bool</p>
</dd>
</dl>
</dd></dl>

<dl class="py attribute">
<dt class="sig sig-object py" id="pinder.core.index.utils.IndexEntry.model_computed_fields">
<span class="sig-name descname"><span class="pre">model_computed_fields</span></span><em class="property"><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="pre">ClassVar</span><span class="p"><span class="pre">[</span></span><span class="pre">dict</span><span class="p"><span class="pre">[</span></span><span class="pre">str</span><span class="p"><span class="pre">,</span></span><span class="w"> </span><span class="pre">ComputedFieldInfo</span><span class="p"><span class="pre">]</span></span><span class="p"><span class="pre">]</span></span></em><em class="property"><span class="w"> </span><span class="p"><span class="pre">=</span></span><span class="w"> </span><span class="pre">{}</span></em><a class="headerlink" href="#pinder.core.index.utils.IndexEntry.model_computed_fields" title="Link to this definition">¶</a></dt>
<dd><p>A dictionary of computed field names and their corresponding <cite>ComputedFieldInfo</cite> objects.</p>
</dd></dl>

<dl class="py attribute">
<dt class="sig sig-object py" id="pinder.core.index.utils.IndexEntry.model_fields">
<span class="sig-name descname"><span class="pre">model_fields</span></span><em class="property"><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="pre">ClassVar</span><span class="p"><span class="pre">[</span></span><span class="pre">dict</span><span class="p"><span class="pre">[</span></span><span class="pre">str</span><span class="p"><span class="pre">,</span></span><span class="w"> </span><span class="pre">FieldInfo</span><span class="p"><span class="pre">]</span></span><span class="p"><span class="pre">]</span></span></em><em class="property"><span class="w"> </span><span class="p"><span class="pre">=</span></span><span class="w"> </span><span class="pre">{'apo_L':</span> <span class="pre">FieldInfo(annotation=bool,</span> <span class="pre">required=True),</span> <span class="pre">'apo_L_pdb':</span> <span class="pre">FieldInfo(annotation=str,</span> <span class="pre">required=True),</span> <span class="pre">'apo_L_pdbs':</span> <span class="pre">FieldInfo(annotation=str,</span> <span class="pre">required=True),</span> <span class="pre">'apo_L_quality':</span> <span class="pre">FieldInfo(annotation=str,</span> <span class="pre">required=True),</span> <span class="pre">'apo_R':</span> <span class="pre">FieldInfo(annotation=bool,</span> <span class="pre">required=True),</span> <span class="pre">'apo_R_pdb':</span> <span class="pre">FieldInfo(annotation=str,</span> <span class="pre">required=True),</span> <span class="pre">'apo_R_pdbs':</span> <span class="pre">FieldInfo(annotation=str,</span> <span class="pre">required=True),</span> <span class="pre">'apo_R_quality':</span> <span class="pre">FieldInfo(annotation=str,</span> <span class="pre">required=True),</span> <span class="pre">'chain1_neff':</span> <span class="pre">FieldInfo(annotation=float,</span> <span class="pre">required=True),</span> <span class="pre">'chain2_neff':</span> <span class="pre">FieldInfo(annotation=float,</span> <span class="pre">required=True),</span> <span class="pre">'chain_L':</span> <span class="pre">FieldInfo(annotation=str,</span> <span class="pre">required=True),</span> <span class="pre">'chain_R':</span> <span class="pre">FieldInfo(annotation=str,</span> <span class="pre">required=True),</span> <span class="pre">'cluster_id':</span> <span class="pre">FieldInfo(annotation=str,</span> <span class="pre">required=True),</span> <span class="pre">'cluster_id_L':</span> <span class="pre">FieldInfo(annotation=str,</span> <span class="pre">required=True),</span> <span class="pre">'cluster_id_R':</span> <span class="pre">FieldInfo(annotation=str,</span> <span class="pre">required=True),</span> <span class="pre">'contains_antibody':</span> <span class="pre">FieldInfo(annotation=bool,</span> <span class="pre">required=True),</span> <span class="pre">'contains_antigen':</span> <span class="pre">FieldInfo(annotation=bool,</span> <span class="pre">required=True),</span> <span class="pre">'contains_enzyme':</span> <span class="pre">FieldInfo(annotation=bool,</span> <span class="pre">required=True),</span> <span class="pre">'holo_L':</span> <span class="pre">FieldInfo(annotation=bool,</span> <span class="pre">required=True),</span> <span class="pre">'holo_L_pdb':</span> <span class="pre">FieldInfo(annotation=str,</span> <span class="pre">required=True),</span> <span class="pre">'holo_R':</span> <span class="pre">FieldInfo(annotation=bool,</span> <span class="pre">required=True),</span> <span class="pre">'holo_R_pdb':</span> <span class="pre">FieldInfo(annotation=str,</span> <span class="pre">required=True),</span> <span class="pre">'id':</span> <span class="pre">FieldInfo(annotation=str,</span> <span class="pre">required=True),</span> <span class="pre">'pdb_id':</span> <span class="pre">FieldInfo(annotation=str,</span> <span class="pre">required=True),</span> <span class="pre">'pinder_af2':</span> <span class="pre">FieldInfo(annotation=bool,</span> <span class="pre">required=True),</span> <span class="pre">'pinder_s':</span> <span class="pre">FieldInfo(annotation=bool,</span> <span class="pre">required=True),</span> <span class="pre">'pinder_xl':</span> <span class="pre">FieldInfo(annotation=bool,</span> <span class="pre">required=True),</span> <span class="pre">'predicted_L':</span> <span class="pre">FieldInfo(annotation=bool,</span> <span class="pre">required=True),</span> <span class="pre">'predicted_L_pdb':</span> <span class="pre">FieldInfo(annotation=str,</span> <span class="pre">required=True),</span> <span class="pre">'predicted_R':</span> <span class="pre">FieldInfo(annotation=bool,</span> <span class="pre">required=True),</span> <span class="pre">'predicted_R_pdb':</span> <span class="pre">FieldInfo(annotation=str,</span> <span class="pre">required=True),</span> <span class="pre">'split':</span> <span class="pre">FieldInfo(annotation=str,</span> <span class="pre">required=True),</span> <span class="pre">'uniprot_L':</span> <span class="pre">FieldInfo(annotation=str,</span> <span class="pre">required=True),</span> <span class="pre">'uniprot_R':</span> <span class="pre">FieldInfo(annotation=str,</span> <span class="pre">required=True)}</span></em><a class="headerlink" href="#pinder.core.index.utils.IndexEntry.model_fields" title="Link to this definition">¶</a></dt>
<dd><p>Metadata about the fields defined on the model,
mapping of field names to [<cite>FieldInfo</cite>][pydantic.fields.FieldInfo].</p>
<p>This replaces <cite>Model.__fields__</cite> from Pydantic V1.</p>
</dd></dl>

</dd></dl>

<dl class="py class">
<dt class="sig sig-object py" id="pinder.core.index.utils.MetadataEntry">
<em class="property"><span class="pre">class</span><span class="w"> </span></em><span class="sig-prename descclassname"><span class="pre">pinder.core.index.utils.</span></span><span class="sig-name descname"><span class="pre">MetadataEntry</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="o"><span class="pre">*</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">id</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">str</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">entry_id</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">str</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">method</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">str</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">date</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">str</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">release_date</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">str</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">resolution</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">float</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">label</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">str</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">probability</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">float</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">chain1_id</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">str</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">chain2_id</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">str</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">assembly</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">int</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">assembly_details</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">str</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">oligomeric_details</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">str</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">oligomeric_count</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">int</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">biol_details</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">str</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">complex_type</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">str</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">chain_1</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">str</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">asym_id_1</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">str</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">chain_2</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">str</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">asym_id_2</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">str</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">length1</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">int</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">length2</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">int</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">length_resolved_1</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">int</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">length_resolved_2</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">int</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">number_of_components_1</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">int</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">number_of_components_2</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">int</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">link_density</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">float</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">planarity</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">float</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">max_var_1</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">float</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">max_var_2</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">float</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">num_atom_types</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">int</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">n_residue_pairs</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">int</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">n_residues</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">int</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">buried_sasa</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">float</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">intermolecular_contacts</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">int</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">charged_charged_contacts</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">int</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">charged_polar_contacts</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">int</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">charged_apolar_contacts</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">int</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">polar_polar_contacts</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">int</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">apolar_polar_contacts</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">int</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">apolar_apolar_contacts</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">int</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">interface_atom_gaps_4A</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">int</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">missing_interface_residues_4A</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">int</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">interface_atom_gaps_8A</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">int</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">missing_interface_residues_8A</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">int</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">entity_id_R</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">int</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">entity_id_L</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">int</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">pdb_strand_id_R</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">str</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">pdb_strand_id_L</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">str</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">ECOD_names_R</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">str</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">ECOD_names_L</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">str</span></span></em>, <em class="sig-param"><span class="o"><span class="pre">**</span></span><span class="n"><span class="pre">extra_data</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">Any</span></span></em><span class="sig-paren">)</span><a class="reference internal" href="../_modules/pinder/core/index/utils.html#MetadataEntry"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#pinder.core.index.utils.MetadataEntry" title="Link to this definition">¶</a></dt>
<dd><p>Bases: <code class="xref py py-class docutils literal notranslate"><span class="pre">BaseModel</span></code></p>
<p>Pydantic model representing a single entry in the Pinder metadata.</p>
<p>Stores detailed metadata for a particular dataset entry as attributes.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>id</strong> (<em>str</em>) – The unique identifier for the PINDER entry.
It follows the convention <cite>&lt;Receptor&gt;–&lt;Ligand&gt;</cite>, where <cite>&lt;Receptor&gt;</cite> is <cite>&lt;pdbid&gt;__&lt;chain_1&gt;_&lt;uniprotid&gt;</cite> and
<cite>&lt;Ligand&gt;</cite> is <cite>&lt;pdbid&gt;__&lt;chain_2&gt;&lt;uniprotid&gt;</cite>.</p></li>
<li><p><strong>entry_id</strong> (<em>str</em>) – The RCSB entry identifier associated with the PINDER entry.</p></li>
<li><p><strong>method</strong> (<em>str</em>) – The experimental method for structure determination (XRAY, CRYO-EM, etc.).</p></li>
<li><p><strong>date</strong> (<em>str</em>) – Date of deposition into RCSB PDB.</p></li>
<li><p><strong>release_date</strong> (<em>str</em>) – Date of initial public release in RCSB PDB.</p></li>
<li><p><strong>resolution</strong> (<em>float</em>) – The resolution of the experimental structure.</p></li>
<li><p><strong>label</strong> (<em>str</em>) – Classification of the interface as likely to be biologically-relevant or a crystal contact, annotated using PRODIGY-cryst.
PRODIGY-cryst uses machine learning to compute bio-relevant/crystal contact propensity based on Intermolecular contact types and Interfacial link density.</p></li>
<li><p><strong>probability</strong> (<em>float</em>) – Probability that the protein complex is a true biological complex.</p></li>
<li><p><strong>chain1_id</strong> (<em>str</em>) – The Receptor chain identifier associated with the dimer entry. Should all be chain ‘R’.</p></li>
<li><p><strong>chain2_id</strong> (<em>str</em>) – The Ligand chain identifier associated with the dimer entry. Should all be chain ‘L’.</p></li>
<li><p><strong>assembly</strong> (<em>int</em>) – Which bioassembly is used to derive the structure. 1, 2, 3 means first, second, and third assembly, respectively.
All PINDER dimers are derived from the first biological assembly.</p></li>
<li><p><strong>assembly_details</strong> (<em>str</em>) – How the bioassembly information was derived. Is it author-defined or from another source.</p></li>
<li><p><strong>oligomeric_details</strong> (<em>str</em>) – Description of the oligomeric state of the protein complex.</p></li>
<li><p><strong>oligomeric_count</strong> (<em>int</em>) – The oligomeric count associated with the dataset entry.</p></li>
<li><p><strong>biol_details</strong> (<em>str</em>) – The biological assembly details associated with the dataset entry.</p></li>
<li><p><strong>complex_type</strong> (<em>str</em>) – The type of the complex in the dataset entry (homomer or heteromer).</p></li>
<li><p><strong>chain_1</strong> (<em>str</em>) – New chain id generated post-bioassembly generation, to reflect the asym_id of the bioassembly and also to ensure that there is no collision of chain ids, for example in homooligomers (receptor chain).</p></li>
<li><p><strong>asym_id_1</strong> (<em>str</em>) – The first asymmetric identifier (author chain ID)</p></li>
<li><p><strong>chain_2</strong> (<em>str</em>) – New chain id generated post-bioassembly generation, to reflect the asym_id of the bioassembly and also to ensure that there is no collision of chain ids, for example in homooligomers (ligand chain).</p></li>
<li><p><strong>asym_id_2</strong> (<em>str</em>) – The second asymmetric identifier (author chain ID)</p></li>
<li><p><strong>length1</strong> (<em>int</em>) – The number of amino acids in the first (receptor) chain.</p></li>
<li><p><strong>length2</strong> (<em>int</em>) – The number of amino acids in the second (ligand) chain.</p></li>
<li><p><strong>length_resolved_1</strong> (<em>int</em>) – The structurally resolved (CA) length of the first (receptor) chain in amino acids.</p></li>
<li><p><strong>length_resolved_2</strong> (<em>int</em>) – The structurally resolved (CA) length of the second (ligand) chain in amino acids.</p></li>
<li><p><strong>number_of_components_1</strong> (<em>int</em>) – The number of connected components in the first (receptor) chain (contiguous structural fragments)</p></li>
<li><p><strong>number_of_components_2</strong> (<em>int</em>) – The number of connected components in the second (receptor) chain (contiguous structural fragments)</p></li>
<li><p><strong>link_density</strong> (<em>float</em>) – Density of contacts at the interface as reported by PRODIGY-cryst.
Interfacial link density is defined as the number of interfacial contacts normalized by the maximum possible number of pairwise contacts for that interface.
Values range between 0 and 1, with higher values indicating a denser contact network at the interface.</p></li>
<li><p><strong>planarity</strong> (<em>float</em>) – Defined as the deviation of interfacial Cα atoms from the fitted plane.
This interface characteristic quantifies interfacial shape complementarity.
Transient complexes have smaller and more planar interfaces than permanent and structural scaffold complexes.</p></li>
<li><p><strong>max_var_1</strong> (<em>float</em>) – The maximum variance of coordinates projected onto the largest principal component.
This allows the detection of long end-to-end stacked complexes, likely to be repetitive with small interfaces (receptor chain).</p></li>
<li><p><strong>max_var_2</strong> (<em>float</em>) – The maximum variance of coordinates projected onto the largest principal component.
This allows the detection of long end-to-end stacked complexes, likely to be repetitive with small interfaces (ligand chain).</p></li>
<li><p><strong>num_atom_types</strong> (<em>int</em>) – Number of unique atom types in structure. This is an important annotation to identify complexes with only Cα or backbone atoms.</p></li>
<li><p><strong>n_residue_pairs</strong> (<em>int</em>) – The number of residue pairs at the interface.</p></li>
<li><p><strong>n_residues</strong> (<em>int</em>) – The number of residues at the interface.</p></li>
<li><p><strong>buried_sasa</strong> (<em>float</em>) – The buried solvent accessible surface area upon complex formation.</p></li>
<li><p><strong>intermolecular_contacts</strong> (<em>int</em>) – The total number of intermolecular contacts (pair residues with any atom within a 5Å distance cutoff) at the interface.
Annotated using PRODIGY-cryst.</p></li>
<li><p><strong>charged_charged_contacts</strong> (<em>int</em>) – Denotes intermolecular contacts between any of the charged amino acids (E, D, H, K).
Annotated using PRODIGY-cryst.</p></li>
<li><p><strong>charged_polar_contacts</strong> (<em>int</em>) – Denotes intermolecular contacts between charged amino acids (E, D, H, K, R) and polar amino acids (N, Q, S, T).
Annotated using PRODIGY-cryst.</p></li>
<li><p><strong>charged_apolar_contacts</strong> (<em>int</em>) – Denotes intermolecular contacts between charged amino acids (E, D, H, K) and apolar amino acids (A, C, G, F, I, M, L, P, W, V, Y).
Annotated using PRODIGY-cryst.</p></li>
<li><p><strong>polar_polar_contacts</strong> (<em>int</em>) – Denotes intermolecular contacts between any of the charged amino acids (N, Q, S, T).
Annotated using PRODIGY-cryst.</p></li>
<li><p><strong>apolar_polar_contacts</strong> (<em>int</em>) – Denotes intermolecular contacts between apolar amino acids (A, C, G,F, I, M, L, P, W, V, Y) and polar amino acids (N, Q, S, T).
Annotated using PRODIGY-cryst.</p></li>
<li><p><strong>apolar_apolar_contacts</strong> (<em>int</em>) – Denotes intermolecular contacts between any of the charged amino acids (A, C, G, F, I, M, L, P, W, V, Y).
Annotated using PRODIGY-cryst.</p></li>
<li><p><strong>interface_atom_gaps_4A</strong> (<em>int</em>) – Number of interface atoms within a 4Å radius of a residue gap.
A Gap is determined by residue numbering; regions where one or more of the expected residue index is missing is marked as a gap.</p></li>
<li><p><strong>missing_interface_residues_4A</strong> (<em>int</em>) – Number of interface residues within a 4Å radius of a residue gap.
A Gap is determined by residue numbering; regions where one or more of the expected residue index is missing is marked as a gap.</p></li>
<li><p><strong>interface_atom_gaps_8A</strong> (<em>int</em>) – Number of interface atoms within an 8Å radius of a residue gap.
A Gap is determined by residue numbering; regions where one or more of the expected residue index is missing is marked as a gap.</p></li>
<li><p><strong>missing_interface_residues_8A</strong> (<em>int</em>) – Number of interface residues within an 8Å radius of a residue gap.
A Gap is determined by residue numbering; regions where one or more of the expected residue index is missing is marked as a gap.</p></li>
<li><p><strong>entity_id_R</strong> (<em>int</em>) – The RCSB PDB <cite>entity_id</cite> corresponding to the receptor dimer chain.</p></li>
<li><p><strong>entity_id_L</strong> (<em>int</em>) – The RCSB PDB <cite>entity_id</cite> corresponding to the ligand dimer chain.</p></li>
<li><p><strong>pdb_strand_id_R</strong> (<em>str</em>) – The RCSB PDB <cite>pdb_strand_id</cite> (author chain) corresponding to the receptor dimer chain.</p></li>
<li><p><strong>pdb_strand_id_L</strong> (<em>str</em>) – The RCSB PDB <cite>pdb_strand_id</cite> (author chain) corresponding to the ligand dimer chain.</p></li>
<li><p><strong>ECOD_names_R</strong> (<em>str</em>) – The RCSB-derived ECOD domain protein family name(s) corresponding to the receptor dimer chain. If multiple ECOD domain annotations
were found, the domains are delimited with a comma.</p></li>
<li><p><strong>ECOD_names_L</strong> (<em>str</em>) – The RCSB-derived ECOD domain protein family name(s) corresponding to the ligand dimer chain. If multiple ECOD domain annotations
were found, the domains are delimited with a comma.</p></li>
</ul>
</dd>
</dl>
<dl class="py attribute">
<dt class="sig sig-object py" id="pinder.core.index.utils.MetadataEntry.model_config">
<span class="sig-name descname"><span class="pre">model_config</span></span><em class="property"><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="pre">ClassVar</span><span class="p"><span class="pre">[</span></span><span class="pre">ConfigDict</span><span class="p"><span class="pre">]</span></span></em><em class="property"><span class="w"> </span><span class="p"><span class="pre">=</span></span><span class="w"> </span><span class="pre">{'extra':</span> <span class="pre">'allow'}</span></em><a class="headerlink" href="#pinder.core.index.utils.MetadataEntry.model_config" title="Link to this definition">¶</a></dt>
<dd><p>Configuration for the model, should be a dictionary conforming to [<cite>ConfigDict</cite>][pydantic.config.ConfigDict].</p>
</dd></dl>

<dl class="py attribute">
<dt class="sig sig-object py" id="pinder.core.index.utils.MetadataEntry.id">
<span class="sig-name descname"><span class="pre">id</span></span><em class="property"><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="pre">str</span></em><a class="headerlink" href="#pinder.core.index.utils.MetadataEntry.id" title="Link to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="py attribute">
<dt class="sig sig-object py" id="pinder.core.index.utils.MetadataEntry.entry_id">
<span class="sig-name descname"><span class="pre">entry_id</span></span><em class="property"><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="pre">str</span></em><a class="headerlink" href="#pinder.core.index.utils.MetadataEntry.entry_id" title="Link to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="py attribute">
<dt class="sig sig-object py" id="pinder.core.index.utils.MetadataEntry.method">
<span class="sig-name descname"><span class="pre">method</span></span><em class="property"><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="pre">str</span></em><a class="headerlink" href="#pinder.core.index.utils.MetadataEntry.method" title="Link to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="py attribute">
<dt class="sig sig-object py" id="pinder.core.index.utils.MetadataEntry.date">
<span class="sig-name descname"><span class="pre">date</span></span><em class="property"><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="pre">str</span></em><a class="headerlink" href="#pinder.core.index.utils.MetadataEntry.date" title="Link to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="py attribute">
<dt class="sig sig-object py" id="pinder.core.index.utils.MetadataEntry.release_date">
<span class="sig-name descname"><span class="pre">release_date</span></span><em class="property"><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="pre">str</span></em><a class="headerlink" href="#pinder.core.index.utils.MetadataEntry.release_date" title="Link to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="py attribute">
<dt class="sig sig-object py" id="pinder.core.index.utils.MetadataEntry.resolution">
<span class="sig-name descname"><span class="pre">resolution</span></span><em class="property"><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="pre">float</span></em><a class="headerlink" href="#pinder.core.index.utils.MetadataEntry.resolution" title="Link to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="py attribute">
<dt class="sig sig-object py" id="pinder.core.index.utils.MetadataEntry.label">
<span class="sig-name descname"><span class="pre">label</span></span><em class="property"><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="pre">str</span></em><a class="headerlink" href="#pinder.core.index.utils.MetadataEntry.label" title="Link to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="py attribute">
<dt class="sig sig-object py" id="pinder.core.index.utils.MetadataEntry.probability">
<span class="sig-name descname"><span class="pre">probability</span></span><em class="property"><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="pre">float</span></em><a class="headerlink" href="#pinder.core.index.utils.MetadataEntry.probability" title="Link to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="py attribute">
<dt class="sig sig-object py" id="pinder.core.index.utils.MetadataEntry.chain1_id">
<span class="sig-name descname"><span class="pre">chain1_id</span></span><em class="property"><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="pre">str</span></em><a class="headerlink" href="#pinder.core.index.utils.MetadataEntry.chain1_id" title="Link to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="py attribute">
<dt class="sig sig-object py" id="pinder.core.index.utils.MetadataEntry.chain2_id">
<span class="sig-name descname"><span class="pre">chain2_id</span></span><em class="property"><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="pre">str</span></em><a class="headerlink" href="#pinder.core.index.utils.MetadataEntry.chain2_id" title="Link to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="py attribute">
<dt class="sig sig-object py" id="pinder.core.index.utils.MetadataEntry.assembly">
<span class="sig-name descname"><span class="pre">assembly</span></span><em class="property"><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="pre">int</span></em><a class="headerlink" href="#pinder.core.index.utils.MetadataEntry.assembly" title="Link to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="py attribute">
<dt class="sig sig-object py" id="pinder.core.index.utils.MetadataEntry.assembly_details">
<span class="sig-name descname"><span class="pre">assembly_details</span></span><em class="property"><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="pre">str</span></em><a class="headerlink" href="#pinder.core.index.utils.MetadataEntry.assembly_details" title="Link to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="py attribute">
<dt class="sig sig-object py" id="pinder.core.index.utils.MetadataEntry.oligomeric_details">
<span class="sig-name descname"><span class="pre">oligomeric_details</span></span><em class="property"><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="pre">str</span></em><a class="headerlink" href="#pinder.core.index.utils.MetadataEntry.oligomeric_details" title="Link to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="py attribute">
<dt class="sig sig-object py" id="pinder.core.index.utils.MetadataEntry.oligomeric_count">
<span class="sig-name descname"><span class="pre">oligomeric_count</span></span><em class="property"><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="pre">int</span></em><a class="headerlink" href="#pinder.core.index.utils.MetadataEntry.oligomeric_count" title="Link to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="py attribute">
<dt class="sig sig-object py" id="pinder.core.index.utils.MetadataEntry.biol_details">
<span class="sig-name descname"><span class="pre">biol_details</span></span><em class="property"><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="pre">str</span></em><a class="headerlink" href="#pinder.core.index.utils.MetadataEntry.biol_details" title="Link to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="py attribute">
<dt class="sig sig-object py" id="pinder.core.index.utils.MetadataEntry.complex_type">
<span class="sig-name descname"><span class="pre">complex_type</span></span><em class="property"><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="pre">str</span></em><a class="headerlink" href="#pinder.core.index.utils.MetadataEntry.complex_type" title="Link to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="py attribute">
<dt class="sig sig-object py" id="pinder.core.index.utils.MetadataEntry.chain_1">
<span class="sig-name descname"><span class="pre">chain_1</span></span><em class="property"><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="pre">str</span></em><a class="headerlink" href="#pinder.core.index.utils.MetadataEntry.chain_1" title="Link to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="py attribute">
<dt class="sig sig-object py" id="pinder.core.index.utils.MetadataEntry.asym_id_1">
<span class="sig-name descname"><span class="pre">asym_id_1</span></span><em class="property"><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="pre">str</span></em><a class="headerlink" href="#pinder.core.index.utils.MetadataEntry.asym_id_1" title="Link to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="py attribute">
<dt class="sig sig-object py" id="pinder.core.index.utils.MetadataEntry.chain_2">
<span class="sig-name descname"><span class="pre">chain_2</span></span><em class="property"><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="pre">str</span></em><a class="headerlink" href="#pinder.core.index.utils.MetadataEntry.chain_2" title="Link to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="py attribute">
<dt class="sig sig-object py" id="pinder.core.index.utils.MetadataEntry.asym_id_2">
<span class="sig-name descname"><span class="pre">asym_id_2</span></span><em class="property"><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="pre">str</span></em><a class="headerlink" href="#pinder.core.index.utils.MetadataEntry.asym_id_2" title="Link to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="py attribute">
<dt class="sig sig-object py" id="pinder.core.index.utils.MetadataEntry.length1">
<span class="sig-name descname"><span class="pre">length1</span></span><em class="property"><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="pre">int</span></em><a class="headerlink" href="#pinder.core.index.utils.MetadataEntry.length1" title="Link to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="py attribute">
<dt class="sig sig-object py" id="pinder.core.index.utils.MetadataEntry.length2">
<span class="sig-name descname"><span class="pre">length2</span></span><em class="property"><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="pre">int</span></em><a class="headerlink" href="#pinder.core.index.utils.MetadataEntry.length2" title="Link to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="py attribute">
<dt class="sig sig-object py" id="pinder.core.index.utils.MetadataEntry.length_resolved_1">
<span class="sig-name descname"><span class="pre">length_resolved_1</span></span><em class="property"><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="pre">int</span></em><a class="headerlink" href="#pinder.core.index.utils.MetadataEntry.length_resolved_1" title="Link to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="py attribute">
<dt class="sig sig-object py" id="pinder.core.index.utils.MetadataEntry.length_resolved_2">
<span class="sig-name descname"><span class="pre">length_resolved_2</span></span><em class="property"><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="pre">int</span></em><a class="headerlink" href="#pinder.core.index.utils.MetadataEntry.length_resolved_2" title="Link to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="py attribute">
<dt class="sig sig-object py" id="pinder.core.index.utils.MetadataEntry.number_of_components_1">
<span class="sig-name descname"><span class="pre">number_of_components_1</span></span><em class="property"><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="pre">int</span></em><a class="headerlink" href="#pinder.core.index.utils.MetadataEntry.number_of_components_1" title="Link to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="py attribute">
<dt class="sig sig-object py" id="pinder.core.index.utils.MetadataEntry.number_of_components_2">
<span class="sig-name descname"><span class="pre">number_of_components_2</span></span><em class="property"><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="pre">int</span></em><a class="headerlink" href="#pinder.core.index.utils.MetadataEntry.number_of_components_2" title="Link to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="py attribute">
<dt class="sig sig-object py" id="pinder.core.index.utils.MetadataEntry.link_density">
<span class="sig-name descname"><span class="pre">link_density</span></span><em class="property"><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="pre">float</span></em><a class="headerlink" href="#pinder.core.index.utils.MetadataEntry.link_density" title="Link to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="py attribute">
<dt class="sig sig-object py" id="pinder.core.index.utils.MetadataEntry.planarity">
<span class="sig-name descname"><span class="pre">planarity</span></span><em class="property"><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="pre">float</span></em><a class="headerlink" href="#pinder.core.index.utils.MetadataEntry.planarity" title="Link to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="py attribute">
<dt class="sig sig-object py" id="pinder.core.index.utils.MetadataEntry.max_var_1">
<span class="sig-name descname"><span class="pre">max_var_1</span></span><em class="property"><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="pre">float</span></em><a class="headerlink" href="#pinder.core.index.utils.MetadataEntry.max_var_1" title="Link to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="py attribute">
<dt class="sig sig-object py" id="pinder.core.index.utils.MetadataEntry.max_var_2">
<span class="sig-name descname"><span class="pre">max_var_2</span></span><em class="property"><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="pre">float</span></em><a class="headerlink" href="#pinder.core.index.utils.MetadataEntry.max_var_2" title="Link to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="py attribute">
<dt class="sig sig-object py" id="pinder.core.index.utils.MetadataEntry.num_atom_types">
<span class="sig-name descname"><span class="pre">num_atom_types</span></span><em class="property"><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="pre">int</span></em><a class="headerlink" href="#pinder.core.index.utils.MetadataEntry.num_atom_types" title="Link to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="py attribute">
<dt class="sig sig-object py" id="pinder.core.index.utils.MetadataEntry.n_residue_pairs">
<span class="sig-name descname"><span class="pre">n_residue_pairs</span></span><em class="property"><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="pre">int</span></em><a class="headerlink" href="#pinder.core.index.utils.MetadataEntry.n_residue_pairs" title="Link to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="py attribute">
<dt class="sig sig-object py" id="pinder.core.index.utils.MetadataEntry.n_residues">
<span class="sig-name descname"><span class="pre">n_residues</span></span><em class="property"><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="pre">int</span></em><a class="headerlink" href="#pinder.core.index.utils.MetadataEntry.n_residues" title="Link to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="py attribute">
<dt class="sig sig-object py" id="pinder.core.index.utils.MetadataEntry.buried_sasa">
<span class="sig-name descname"><span class="pre">buried_sasa</span></span><em class="property"><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="pre">float</span></em><a class="headerlink" href="#pinder.core.index.utils.MetadataEntry.buried_sasa" title="Link to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="py attribute">
<dt class="sig sig-object py" id="pinder.core.index.utils.MetadataEntry.intermolecular_contacts">
<span class="sig-name descname"><span class="pre">intermolecular_contacts</span></span><em class="property"><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="pre">int</span></em><a class="headerlink" href="#pinder.core.index.utils.MetadataEntry.intermolecular_contacts" title="Link to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="py attribute">
<dt class="sig sig-object py" id="pinder.core.index.utils.MetadataEntry.charged_charged_contacts">
<span class="sig-name descname"><span class="pre">charged_charged_contacts</span></span><em class="property"><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="pre">int</span></em><a class="headerlink" href="#pinder.core.index.utils.MetadataEntry.charged_charged_contacts" title="Link to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="py attribute">
<dt class="sig sig-object py" id="pinder.core.index.utils.MetadataEntry.charged_polar_contacts">
<span class="sig-name descname"><span class="pre">charged_polar_contacts</span></span><em class="property"><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="pre">int</span></em><a class="headerlink" href="#pinder.core.index.utils.MetadataEntry.charged_polar_contacts" title="Link to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="py attribute">
<dt class="sig sig-object py" id="pinder.core.index.utils.MetadataEntry.charged_apolar_contacts">
<span class="sig-name descname"><span class="pre">charged_apolar_contacts</span></span><em class="property"><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="pre">int</span></em><a class="headerlink" href="#pinder.core.index.utils.MetadataEntry.charged_apolar_contacts" title="Link to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="py attribute">
<dt class="sig sig-object py" id="pinder.core.index.utils.MetadataEntry.polar_polar_contacts">
<span class="sig-name descname"><span class="pre">polar_polar_contacts</span></span><em class="property"><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="pre">int</span></em><a class="headerlink" href="#pinder.core.index.utils.MetadataEntry.polar_polar_contacts" title="Link to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="py attribute">
<dt class="sig sig-object py" id="pinder.core.index.utils.MetadataEntry.apolar_polar_contacts">
<span class="sig-name descname"><span class="pre">apolar_polar_contacts</span></span><em class="property"><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="pre">int</span></em><a class="headerlink" href="#pinder.core.index.utils.MetadataEntry.apolar_polar_contacts" title="Link to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="py attribute">
<dt class="sig sig-object py" id="pinder.core.index.utils.MetadataEntry.apolar_apolar_contacts">
<span class="sig-name descname"><span class="pre">apolar_apolar_contacts</span></span><em class="property"><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="pre">int</span></em><a class="headerlink" href="#pinder.core.index.utils.MetadataEntry.apolar_apolar_contacts" title="Link to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="py attribute">
<dt class="sig sig-object py" id="pinder.core.index.utils.MetadataEntry.interface_atom_gaps_4A">
<span class="sig-name descname"><span class="pre">interface_atom_gaps_4A</span></span><em class="property"><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="pre">int</span></em><a class="headerlink" href="#pinder.core.index.utils.MetadataEntry.interface_atom_gaps_4A" title="Link to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="py attribute">
<dt class="sig sig-object py" id="pinder.core.index.utils.MetadataEntry.missing_interface_residues_4A">
<span class="sig-name descname"><span class="pre">missing_interface_residues_4A</span></span><em class="property"><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="pre">int</span></em><a class="headerlink" href="#pinder.core.index.utils.MetadataEntry.missing_interface_residues_4A" title="Link to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="py attribute">
<dt class="sig sig-object py" id="pinder.core.index.utils.MetadataEntry.interface_atom_gaps_8A">
<span class="sig-name descname"><span class="pre">interface_atom_gaps_8A</span></span><em class="property"><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="pre">int</span></em><a class="headerlink" href="#pinder.core.index.utils.MetadataEntry.interface_atom_gaps_8A" title="Link to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="py attribute">
<dt class="sig sig-object py" id="pinder.core.index.utils.MetadataEntry.missing_interface_residues_8A">
<span class="sig-name descname"><span class="pre">missing_interface_residues_8A</span></span><em class="property"><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="pre">int</span></em><a class="headerlink" href="#pinder.core.index.utils.MetadataEntry.missing_interface_residues_8A" title="Link to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="py attribute">
<dt class="sig sig-object py" id="pinder.core.index.utils.MetadataEntry.entity_id_R">
<span class="sig-name descname"><span class="pre">entity_id_R</span></span><em class="property"><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="pre">int</span></em><a class="headerlink" href="#pinder.core.index.utils.MetadataEntry.entity_id_R" title="Link to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="py attribute">
<dt class="sig sig-object py" id="pinder.core.index.utils.MetadataEntry.entity_id_L">
<span class="sig-name descname"><span class="pre">entity_id_L</span></span><em class="property"><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="pre">int</span></em><a class="headerlink" href="#pinder.core.index.utils.MetadataEntry.entity_id_L" title="Link to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="py attribute">
<dt class="sig sig-object py" id="pinder.core.index.utils.MetadataEntry.pdb_strand_id_R">
<span class="sig-name descname"><span class="pre">pdb_strand_id_R</span></span><em class="property"><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="pre">str</span></em><a class="headerlink" href="#pinder.core.index.utils.MetadataEntry.pdb_strand_id_R" title="Link to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="py attribute">
<dt class="sig sig-object py" id="pinder.core.index.utils.MetadataEntry.pdb_strand_id_L">
<span class="sig-name descname"><span class="pre">pdb_strand_id_L</span></span><em class="property"><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="pre">str</span></em><a class="headerlink" href="#pinder.core.index.utils.MetadataEntry.pdb_strand_id_L" title="Link to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="py attribute">
<dt class="sig sig-object py" id="pinder.core.index.utils.MetadataEntry.ECOD_names_R">
<span class="sig-name descname"><span class="pre">ECOD_names_R</span></span><em class="property"><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="pre">str</span></em><a class="headerlink" href="#pinder.core.index.utils.MetadataEntry.ECOD_names_R" title="Link to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="py attribute">
<dt class="sig sig-object py" id="pinder.core.index.utils.MetadataEntry.ECOD_names_L">
<span class="sig-name descname"><span class="pre">ECOD_names_L</span></span><em class="property"><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="pre">str</span></em><a class="headerlink" href="#pinder.core.index.utils.MetadataEntry.ECOD_names_L" title="Link to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="py property">
<dt class="sig sig-object py" id="pinder.core.index.utils.MetadataEntry.pinder_id">
<em class="property"><span class="pre">property</span><span class="w"> </span></em><span class="sig-name descname"><span class="pre">pinder_id</span></span><em class="property"><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="pre">str</span></em><a class="headerlink" href="#pinder.core.index.utils.MetadataEntry.pinder_id" title="Link to this definition">¶</a></dt>
<dd><p>PINDER identifier for the dimer.</p>
<dl class="field-list simple">
<dt class="field-odd">Type<span class="colon">:</span></dt>
<dd class="field-odd"><p>str</p>
</dd>
</dl>
</dd></dl>

<dl class="py attribute">
<dt class="sig sig-object py" id="pinder.core.index.utils.MetadataEntry.model_computed_fields">
<span class="sig-name descname"><span class="pre">model_computed_fields</span></span><em class="property"><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="pre">ClassVar</span><span class="p"><span class="pre">[</span></span><span class="pre">dict</span><span class="p"><span class="pre">[</span></span><span class="pre">str</span><span class="p"><span class="pre">,</span></span><span class="w"> </span><span class="pre">ComputedFieldInfo</span><span class="p"><span class="pre">]</span></span><span class="p"><span class="pre">]</span></span></em><em class="property"><span class="w"> </span><span class="p"><span class="pre">=</span></span><span class="w"> </span><span class="pre">{}</span></em><a class="headerlink" href="#pinder.core.index.utils.MetadataEntry.model_computed_fields" title="Link to this definition">¶</a></dt>
<dd><p>A dictionary of computed field names and their corresponding <cite>ComputedFieldInfo</cite> objects.</p>
</dd></dl>

<dl class="py attribute">
<dt class="sig sig-object py" id="pinder.core.index.utils.MetadataEntry.model_fields">
<span class="sig-name descname"><span class="pre">model_fields</span></span><em class="property"><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="pre">ClassVar</span><span class="p"><span class="pre">[</span></span><span class="pre">dict</span><span class="p"><span class="pre">[</span></span><span class="pre">str</span><span class="p"><span class="pre">,</span></span><span class="w"> </span><span class="pre">FieldInfo</span><span class="p"><span class="pre">]</span></span><span class="p"><span class="pre">]</span></span></em><em class="property"><span class="w"> </span><span class="p"><span class="pre">=</span></span><span class="w"> </span><span class="pre">{'ECOD_names_L':</span> <span class="pre">FieldInfo(annotation=str,</span> <span class="pre">required=True),</span> <span class="pre">'ECOD_names_R':</span> <span class="pre">FieldInfo(annotation=str,</span> <span class="pre">required=True),</span> <span class="pre">'apolar_apolar_contacts':</span> <span class="pre">FieldInfo(annotation=int,</span> <span class="pre">required=True),</span> <span class="pre">'apolar_polar_contacts':</span> <span class="pre">FieldInfo(annotation=int,</span> <span class="pre">required=True),</span> <span class="pre">'assembly':</span> <span class="pre">FieldInfo(annotation=int,</span> <span class="pre">required=True),</span> <span class="pre">'assembly_details':</span> <span class="pre">FieldInfo(annotation=str,</span> <span class="pre">required=True),</span> <span class="pre">'asym_id_1':</span> <span class="pre">FieldInfo(annotation=str,</span> <span class="pre">required=True),</span> <span class="pre">'asym_id_2':</span> <span class="pre">FieldInfo(annotation=str,</span> <span class="pre">required=True),</span> <span class="pre">'biol_details':</span> <span class="pre">FieldInfo(annotation=str,</span> <span class="pre">required=True),</span> <span class="pre">'buried_sasa':</span> <span class="pre">FieldInfo(annotation=float,</span> <span class="pre">required=True),</span> <span class="pre">'chain1_id':</span> <span class="pre">FieldInfo(annotation=str,</span> <span class="pre">required=True),</span> <span class="pre">'chain2_id':</span> <span class="pre">FieldInfo(annotation=str,</span> <span class="pre">required=True),</span> <span class="pre">'chain_1':</span> <span class="pre">FieldInfo(annotation=str,</span> <span class="pre">required=True),</span> <span class="pre">'chain_2':</span> <span class="pre">FieldInfo(annotation=str,</span> <span class="pre">required=True),</span> <span class="pre">'charged_apolar_contacts':</span> <span class="pre">FieldInfo(annotation=int,</span> <span class="pre">required=True),</span> <span class="pre">'charged_charged_contacts':</span> <span class="pre">FieldInfo(annotation=int,</span> <span class="pre">required=True),</span> <span class="pre">'charged_polar_contacts':</span> <span class="pre">FieldInfo(annotation=int,</span> <span class="pre">required=True),</span> <span class="pre">'complex_type':</span> <span class="pre">FieldInfo(annotation=str,</span> <span class="pre">required=True),</span> <span class="pre">'date':</span> <span class="pre">FieldInfo(annotation=str,</span> <span class="pre">required=True),</span> <span class="pre">'entity_id_L':</span> <span class="pre">FieldInfo(annotation=int,</span> <span class="pre">required=True),</span> <span class="pre">'entity_id_R':</span> <span class="pre">FieldInfo(annotation=int,</span> <span class="pre">required=True),</span> <span class="pre">'entry_id':</span> <span class="pre">FieldInfo(annotation=str,</span> <span class="pre">required=True),</span> <span class="pre">'id':</span> <span class="pre">FieldInfo(annotation=str,</span> <span class="pre">required=True),</span> <span class="pre">'interface_atom_gaps_4A':</span> <span class="pre">FieldInfo(annotation=int,</span> <span class="pre">required=True),</span> <span class="pre">'interface_atom_gaps_8A':</span> <span class="pre">FieldInfo(annotation=int,</span> <span class="pre">required=True),</span> <span class="pre">'intermolecular_contacts':</span> <span class="pre">FieldInfo(annotation=int,</span> <span class="pre">required=True),</span> <span class="pre">'label':</span> <span class="pre">FieldInfo(annotation=str,</span> <span class="pre">required=True),</span> <span class="pre">'length1':</span> <span class="pre">FieldInfo(annotation=int,</span> <span class="pre">required=True),</span> <span class="pre">'length2':</span> <span class="pre">FieldInfo(annotation=int,</span> <span class="pre">required=True),</span> <span class="pre">'length_resolved_1':</span> <span class="pre">FieldInfo(annotation=int,</span> <span class="pre">required=True),</span> <span class="pre">'length_resolved_2':</span> <span class="pre">FieldInfo(annotation=int,</span> <span class="pre">required=True),</span> <span class="pre">'link_density':</span> <span class="pre">FieldInfo(annotation=float,</span> <span class="pre">required=True),</span> <span class="pre">'max_var_1':</span> <span class="pre">FieldInfo(annotation=float,</span> <span class="pre">required=True),</span> <span class="pre">'max_var_2':</span> <span class="pre">FieldInfo(annotation=float,</span> <span class="pre">required=True),</span> <span class="pre">'method':</span> <span class="pre">FieldInfo(annotation=str,</span> <span class="pre">required=True),</span> <span class="pre">'missing_interface_residues_4A':</span> <span class="pre">FieldInfo(annotation=int,</span> <span class="pre">required=True),</span> <span class="pre">'missing_interface_residues_8A':</span> <span class="pre">FieldInfo(annotation=int,</span> <span class="pre">required=True),</span> <span class="pre">'n_residue_pairs':</span> <span class="pre">FieldInfo(annotation=int,</span> <span class="pre">required=True),</span> <span class="pre">'n_residues':</span> <span class="pre">FieldInfo(annotation=int,</span> <span class="pre">required=True),</span> <span class="pre">'num_atom_types':</span> <span class="pre">FieldInfo(annotation=int,</span> <span class="pre">required=True),</span> <span class="pre">'number_of_components_1':</span> <span class="pre">FieldInfo(annotation=int,</span> <span class="pre">required=True),</span> <span class="pre">'number_of_components_2':</span> <span class="pre">FieldInfo(annotation=int,</span> <span class="pre">required=True),</span> <span class="pre">'oligomeric_count':</span> <span class="pre">FieldInfo(annotation=int,</span> <span class="pre">required=True),</span> <span class="pre">'oligomeric_details':</span> <span class="pre">FieldInfo(annotation=str,</span> <span class="pre">required=True),</span> <span class="pre">'pdb_strand_id_L':</span> <span class="pre">FieldInfo(annotation=str,</span> <span class="pre">required=True),</span> <span class="pre">'pdb_strand_id_R':</span> <span class="pre">FieldInfo(annotation=str,</span> <span class="pre">required=True),</span> <span class="pre">'planarity':</span> <span class="pre">FieldInfo(annotation=float,</span> <span class="pre">required=True),</span> <span class="pre">'polar_polar_contacts':</span> <span class="pre">FieldInfo(annotation=int,</span> <span class="pre">required=True),</span> <span class="pre">'probability':</span> <span class="pre">FieldInfo(annotation=float,</span> <span class="pre">required=True),</span> <span class="pre">'release_date':</span> <span class="pre">FieldInfo(annotation=str,</span> <span class="pre">required=True),</span> <span class="pre">'resolution':</span> <span class="pre">FieldInfo(annotation=float,</span> <span class="pre">required=True)}</span></em><a class="headerlink" href="#pinder.core.index.utils.MetadataEntry.model_fields" title="Link to this definition">¶</a></dt>
<dd><p>Metadata about the fields defined on the model,
mapping of field names to [<cite>FieldInfo</cite>][pydantic.fields.FieldInfo].</p>
<p>This replaces <cite>Model.__fields__</cite> from Pydantic V1.</p>
</dd></dl>

</dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="pinder.core.index.utils.fix_index">
<span class="sig-prename descclassname"><span class="pre">pinder.core.index.utils.</span></span><span class="sig-name descname"><span class="pre">fix_index</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">df_index</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">DataFrame</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">index_fields</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">list</span><span class="p"><span class="pre">[</span></span><span class="pre">str</span><span class="p"><span class="pre">]</span></span></span></em>, <em class="sig-param"><span class="n"><span class="pre">cast_types</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">bool</span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">True</span></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">&#x2192;</span> <span class="sig-return-typehint"><span class="pre">DataFrame</span></span></span><a class="reference internal" href="../_modules/pinder/core/index/utils.html#fix_index"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#pinder.core.index.utils.fix_index" title="Link to this definition">¶</a></dt>
<dd><p>Fix the index dataframe according to the IndexEntry schema</p>
</dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="pinder.core.index.utils.fix_metadata">
<span class="sig-prename descclassname"><span class="pre">pinder.core.index.utils.</span></span><span class="sig-name descname"><span class="pre">fix_metadata</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">df_metadata</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">DataFrame</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">metadata_fields</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">list</span><span class="p"><span class="pre">[</span></span><span class="pre">str</span><span class="p"><span class="pre">]</span></span></span></em>, <em class="sig-param"><span class="n"><span class="pre">cast_types</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">bool</span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">True</span></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">&#x2192;</span> <span class="sig-return-typehint"><span class="pre">DataFrame</span></span></span><a class="reference internal" href="../_modules/pinder/core/index/utils.html#fix_metadata"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#pinder.core.index.utils.fix_metadata" title="Link to this definition">¶</a></dt>
<dd><p>Fix the metadata dataframe according to the MetadataEntry schema</p>
</dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="pinder.core.index.utils.downcast_dtypes">
<span class="sig-prename descclassname"><span class="pre">pinder.core.index.utils.</span></span><span class="sig-name descname"><span class="pre">downcast_dtypes</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">df</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">DataFrame</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">str_as_cat</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">bool</span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">True</span></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">&#x2192;</span> <span class="sig-return-typehint"><span class="pre">DataFrame</span></span></span><a class="reference internal" href="../_modules/pinder/core/index/utils.html#downcast_dtypes"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#pinder.core.index.utils.downcast_dtypes" title="Link to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="pinder.core.index.utils.set_mapping_column_types">
<span class="sig-prename descclassname"><span class="pre">pinder.core.index.utils.</span></span><span class="sig-name descname"><span class="pre">set_mapping_column_types</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">mapping_df</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">DataFrame</span></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">&#x2192;</span> <span class="sig-return-typehint"><span class="pre">DataFrame</span></span></span><a class="reference internal" href="../_modules/pinder/core/index/utils.html#set_mapping_column_types"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#pinder.core.index.utils.set_mapping_column_types" title="Link to this definition">¶</a></dt>
<dd><p>Set the column types for the mapping dataframe to avoid 1.0 2.0 “integers”.
Int64 is supposed to support NaNs. This function requires Pandas 1.0+.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><dl class="simple">
<dt><strong>mapping_df</strong><span class="classifier">pd.DataFrame</span></dt><dd><p>The dataframe whose column types are to be set.</p>
</dd>
</dl>
</dd>
</dl>
</dd></dl>

</section>
<section id="module-pinder.core.index">
<span id="module-contents"></span><h2>Module contents<a class="headerlink" href="#module-pinder.core.index" title="Link to this heading">¶</a></h2>
</section>
</section>


            <div class="clearer"></div>
          </div>
        </div>
      </div>
    
        <div id="show_right_sidebar">
            <p><a class="toggle_right_sidebar" href="#"><span class="icon">&lt;</span><span>Page contents</span></a></p>
        </div>

        <div id="right_sidebar">
            <p><a class="toggle_right_sidebar" href="#"><span class="icon">&gt;</span><span>Page contents:</span></a></p>
            <div class="page_toc">
                <ul>
<li><a class="reference internal" href="#">pinder.core.index package</a><ul>
<li><a class="reference internal" href="#submodules">Submodules</a></li>
<li><a class="reference internal" href="#module-pinder.core.index.id">pinder.core.index.id module</a><ul>
<li><a class="reference internal" href="#pinder.core.index.id.Protein"><code class="docutils literal notranslate"><span class="pre">Protein</span></code></a><ul>
<li><a class="reference internal" href="#pinder.core.index.id.Protein.source"><code class="docutils literal notranslate"><span class="pre">Protein.source</span></code></a></li>
<li><a class="reference internal" href="#pinder.core.index.id.Protein.uniprot"><code class="docutils literal notranslate"><span class="pre">Protein.uniprot</span></code></a></li>
<li><a class="reference internal" href="#pinder.core.index.id.Protein.chain"><code class="docutils literal notranslate"><span class="pre">Protein.chain</span></code></a></li>
<li><a class="reference internal" href="#pinder.core.index.id.Protein.from_residue"><code class="docutils literal notranslate"><span class="pre">Protein.from_residue</span></code></a></li>
<li><a class="reference internal" href="#pinder.core.index.id.Protein.to_residue"><code class="docutils literal notranslate"><span class="pre">Protein.to_residue</span></code></a></li>
</ul>
</li>
<li><a class="reference internal" href="#pinder.core.index.id.Monomer"><code class="docutils literal notranslate"><span class="pre">Monomer</span></code></a><ul>
<li><a class="reference internal" href="#pinder.core.index.id.Monomer.proteins"><code class="docutils literal notranslate"><span class="pre">Monomer.proteins</span></code></a></li>
<li><a class="reference internal" href="#pinder.core.index.id.Monomer.side"><code class="docutils literal notranslate"><span class="pre">Monomer.side</span></code></a></li>
<li><a class="reference internal" href="#pinder.core.index.id.Monomer.from_string"><code class="docutils literal notranslate"><span class="pre">Monomer.from_string()</span></code></a></li>
</ul>
</li>
<li><a class="reference internal" href="#pinder.core.index.id.Dimer"><code class="docutils literal notranslate"><span class="pre">Dimer</span></code></a><ul>
<li><a class="reference internal" href="#pinder.core.index.id.Dimer.monomer1"><code class="docutils literal notranslate"><span class="pre">Dimer.monomer1</span></code></a></li>
<li><a class="reference internal" href="#pinder.core.index.id.Dimer.monomer2"><code class="docutils literal notranslate"><span class="pre">Dimer.monomer2</span></code></a></li>
<li><a class="reference internal" href="#pinder.core.index.id.Dimer.from_string"><code class="docutils literal notranslate"><span class="pre">Dimer.from_string()</span></code></a></li>
</ul>
</li>
</ul>
</li>
<li><a class="reference internal" href="#module-pinder.core.index.system">pinder.core.index.system module</a><ul>
<li><a class="reference internal" href="#pinder.core.index.system.FolderNames"><code class="docutils literal notranslate"><span class="pre">FolderNames</span></code></a><ul>
<li><a class="reference internal" href="#pinder.core.index.system.FolderNames.apo"><code class="docutils literal notranslate"><span class="pre">FolderNames.apo</span></code></a></li>
<li><a class="reference internal" href="#pinder.core.index.system.FolderNames.holo"><code class="docutils literal notranslate"><span class="pre">FolderNames.holo</span></code></a></li>
<li><a class="reference internal" href="#pinder.core.index.system.FolderNames.predicted"><code class="docutils literal notranslate"><span class="pre">FolderNames.predicted</span></code></a></li>
<li><a class="reference internal" href="#pinder.core.index.system.FolderNames.alphafold"><code class="docutils literal notranslate"><span class="pre">FolderNames.alphafold</span></code></a></li>
<li><a class="reference internal" href="#pinder.core.index.system.FolderNames.af2"><code class="docutils literal notranslate"><span class="pre">FolderNames.af2</span></code></a></li>
</ul>
</li>
<li><a class="reference internal" href="#pinder.core.index.system.PinderSystem"><code class="docutils literal notranslate"><span class="pre">PinderSystem</span></code></a><ul>
<li><a class="reference internal" href="#pinder.core.index.system.PinderSystem.entry"><code class="docutils literal notranslate"><span class="pre">PinderSystem.entry</span></code></a></li>
<li><a class="reference internal" href="#pinder.core.index.system.PinderSystem.pdbs_path"><code class="docutils literal notranslate"><span class="pre">PinderSystem.pdbs_path</span></code></a></li>
<li><a class="reference internal" href="#pinder.core.index.system.PinderSystem.mappings_path"><code class="docutils literal notranslate"><span class="pre">PinderSystem.mappings_path</span></code></a></li>
<li><a class="reference internal" href="#pinder.core.index.system.PinderSystem.native"><code class="docutils literal notranslate"><span class="pre">PinderSystem.native</span></code></a></li>
<li><a class="reference internal" href="#pinder.core.index.system.PinderSystem.holo_dir"><code class="docutils literal notranslate"><span class="pre">PinderSystem.holo_dir</span></code></a></li>
<li><a class="reference internal" href="#pinder.core.index.system.PinderSystem.apo_dir"><code class="docutils literal notranslate"><span class="pre">PinderSystem.apo_dir</span></code></a></li>
<li><a class="reference internal" href="#pinder.core.index.system.PinderSystem.pred_dir"><code class="docutils literal notranslate"><span class="pre">PinderSystem.pred_dir</span></code></a></li>
<li><a class="reference internal" href="#pinder.core.index.system.PinderSystem.holo_receptor"><code class="docutils literal notranslate"><span class="pre">PinderSystem.holo_receptor</span></code></a></li>
<li><a class="reference internal" href="#pinder.core.index.system.PinderSystem.holo_ligand"><code class="docutils literal notranslate"><span class="pre">PinderSystem.holo_ligand</span></code></a></li>
<li><a class="reference internal" href="#pinder.core.index.system.PinderSystem.apo_receptor"><code class="docutils literal notranslate"><span class="pre">PinderSystem.apo_receptor</span></code></a></li>
<li><a class="reference internal" href="#pinder.core.index.system.PinderSystem.apo_ligand"><code class="docutils literal notranslate"><span class="pre">PinderSystem.apo_ligand</span></code></a></li>
<li><a class="reference internal" href="#pinder.core.index.system.PinderSystem.pred_receptor"><code class="docutils literal notranslate"><span class="pre">PinderSystem.pred_receptor</span></code></a></li>
<li><a class="reference internal" href="#pinder.core.index.system.PinderSystem.pred_ligand"><code class="docutils literal notranslate"><span class="pre">PinderSystem.pred_ligand</span></code></a></li>
<li><a class="reference internal" href="#pinder.core.index.system.PinderSystem.filter_common_uniprot_res"><code class="docutils literal notranslate"><span class="pre">PinderSystem.filter_common_uniprot_res()</span></code></a></li>
<li><a class="reference internal" href="#pinder.core.index.system.PinderSystem.create_masked_bound_unbound_complexes"><code class="docutils literal notranslate"><span class="pre">PinderSystem.create_masked_bound_unbound_complexes()</span></code></a></li>
<li><a class="reference internal" href="#pinder.core.index.system.PinderSystem.create_complex"><code class="docutils literal notranslate"><span class="pre">PinderSystem.create_complex()</span></code></a></li>
<li><a class="reference internal" href="#pinder.core.index.system.PinderSystem.create_apo_complex"><code class="docutils literal notranslate"><span class="pre">PinderSystem.create_apo_complex()</span></code></a></li>
<li><a class="reference internal" href="#pinder.core.index.system.PinderSystem.create_pred_complex"><code class="docutils literal notranslate"><span class="pre">PinderSystem.create_pred_complex()</span></code></a></li>
<li><a class="reference internal" href="#pinder.core.index.system.PinderSystem.unbound_rmsd"><code class="docutils literal notranslate"><span class="pre">PinderSystem.unbound_rmsd()</span></code></a></li>
<li><a class="reference internal" href="#pinder.core.index.system.PinderSystem.unbound_difficulty"><code class="docutils literal notranslate"><span class="pre">PinderSystem.unbound_difficulty()</span></code></a></li>
<li><a class="reference internal" href="#pinder.core.index.system.PinderSystem.apo_monomer_difficulty"><code class="docutils literal notranslate"><span class="pre">PinderSystem.apo_monomer_difficulty()</span></code></a></li>
<li><a class="reference internal" href="#pinder.core.index.system.PinderSystem.download_entry"><code class="docutils literal notranslate"><span class="pre">PinderSystem.download_entry()</span></code></a></li>
<li><a class="reference internal" href="#pinder.core.index.system.PinderSystem.filepaths"><code class="docutils literal notranslate"><span class="pre">PinderSystem.filepaths</span></code></a></li>
<li><a class="reference internal" href="#pinder.core.index.system.PinderSystem.metadata"><code class="docutils literal notranslate"><span class="pre">PinderSystem.metadata</span></code></a></li>
<li><a class="reference internal" href="#pinder.core.index.system.PinderSystem.pymol_script"><code class="docutils literal notranslate"><span class="pre">PinderSystem.pymol_script</span></code></a></li>
<li><a class="reference internal" href="#pinder.core.index.system.PinderSystem.load_alt_apo_structure"><code class="docutils literal notranslate"><span class="pre">PinderSystem.load_alt_apo_structure()</span></code></a></li>
<li><a class="reference internal" href="#pinder.core.index.system.PinderSystem.load_structure"><code class="docutils literal notranslate"><span class="pre">PinderSystem.load_structure()</span></code></a></li>
</ul>
</li>
</ul>
</li>
<li><a class="reference internal" href="#module-pinder.core.index.utils">pinder.core.index.utils module</a><ul>
<li><a class="reference internal" href="#pinder.core.index.utils.get_pinder_location"><code class="docutils literal notranslate"><span class="pre">get_pinder_location()</span></code></a></li>
<li><a class="reference internal" href="#pinder.core.index.utils.get_pinder_bucket_root"><code class="docutils literal notranslate"><span class="pre">get_pinder_bucket_root()</span></code></a></li>
<li><a class="reference internal" href="#pinder.core.index.utils.get_index_location"><code class="docutils literal notranslate"><span class="pre">get_index_location()</span></code></a></li>
<li><a class="reference internal" href="#pinder.core.index.utils.get_index"><code class="docutils literal notranslate"><span class="pre">get_index()</span></code></a></li>
<li><a class="reference internal" href="#pinder.core.index.utils.get_metadata"><code class="docutils literal notranslate"><span class="pre">get_metadata()</span></code></a></li>
<li><a class="reference internal" href="#pinder.core.index.utils.get_extra_metadata"><code class="docutils literal notranslate"><span class="pre">get_extra_metadata()</span></code></a></li>
<li><a class="reference internal" href="#pinder.core.index.utils.SupplementaryData"><code class="docutils literal notranslate"><span class="pre">SupplementaryData</span></code></a><ul>
<li><a class="reference internal" href="#pinder.core.index.utils.SupplementaryData.sequence_database"><code class="docutils literal notranslate"><span class="pre">SupplementaryData.sequence_database</span></code></a></li>
<li><a class="reference internal" href="#pinder.core.index.utils.SupplementaryData.supplementary_metadata"><code class="docutils literal notranslate"><span class="pre">SupplementaryData.supplementary_metadata</span></code></a></li>
<li><a class="reference internal" href="#pinder.core.index.utils.SupplementaryData.entity_metadata"><code class="docutils literal notranslate"><span class="pre">SupplementaryData.entity_metadata</span></code></a></li>
<li><a class="reference internal" href="#pinder.core.index.utils.SupplementaryData.chain_metadata"><code class="docutils literal notranslate"><span class="pre">SupplementaryData.chain_metadata</span></code></a></li>
<li><a class="reference internal" href="#pinder.core.index.utils.SupplementaryData.ecod_metadata"><code class="docutils literal notranslate"><span class="pre">SupplementaryData.ecod_metadata</span></code></a></li>
<li><a class="reference internal" href="#pinder.core.index.utils.SupplementaryData.enzyme_classification_metadata"><code class="docutils literal notranslate"><span class="pre">SupplementaryData.enzyme_classification_metadata</span></code></a></li>
<li><a class="reference internal" href="#pinder.core.index.utils.SupplementaryData.interface_annotations"><code class="docutils literal notranslate"><span class="pre">SupplementaryData.interface_annotations</span></code></a></li>
<li><a class="reference internal" href="#pinder.core.index.utils.SupplementaryData.sabdab_metadata"><code class="docutils literal notranslate"><span class="pre">SupplementaryData.sabdab_metadata</span></code></a></li>
<li><a class="reference internal" href="#pinder.core.index.utils.SupplementaryData.monomer_neff"><code class="docutils literal notranslate"><span class="pre">SupplementaryData.monomer_neff</span></code></a></li>
<li><a class="reference internal" href="#pinder.core.index.utils.SupplementaryData.paired_neff"><code class="docutils literal notranslate"><span class="pre">SupplementaryData.paired_neff</span></code></a></li>
<li><a class="reference internal" href="#pinder.core.index.utils.SupplementaryData.transient_interface_metadata"><code class="docutils literal notranslate"><span class="pre">SupplementaryData.transient_interface_metadata</span></code></a></li>
<li><a class="reference internal" href="#pinder.core.index.utils.SupplementaryData.ialign_split_similarity_labels"><code class="docutils literal notranslate"><span class="pre">SupplementaryData.ialign_split_similarity_labels</span></code></a></li>
</ul>
</li>
<li><a class="reference internal" href="#pinder.core.index.utils.get_supplementary_data"><code class="docutils literal notranslate"><span class="pre">get_supplementary_data()</span></code></a></li>
<li><a class="reference internal" href="#pinder.core.index.utils.get_sequence_database"><code class="docutils literal notranslate"><span class="pre">get_sequence_database()</span></code></a></li>
<li><a class="reference internal" href="#pinder.core.index.utils.download_dataset"><code class="docutils literal notranslate"><span class="pre">download_dataset()</span></code></a></li>
<li><a class="reference internal" href="#pinder.core.index.utils.get_arg_parser_args"><code class="docutils literal notranslate"><span class="pre">get_arg_parser_args()</span></code></a></li>
<li><a class="reference internal" href="#pinder.core.index.utils.download_pinder_cmd"><code class="docutils literal notranslate"><span class="pre">download_pinder_cmd()</span></code></a></li>
<li><a class="reference internal" href="#pinder.core.index.utils.update_index_cmd"><code class="docutils literal notranslate"><span class="pre">update_index_cmd()</span></code></a></li>
<li><a class="reference internal" href="#pinder.core.index.utils.get_missing_blobs"><code class="docutils literal notranslate"><span class="pre">get_missing_blobs()</span></code></a></li>
<li><a class="reference internal" href="#pinder.core.index.utils.sync_pinder_structure_data"><code class="docutils literal notranslate"><span class="pre">sync_pinder_structure_data()</span></code></a></li>
<li><a class="reference internal" href="#pinder.core.index.utils.IndexEntry"><code class="docutils literal notranslate"><span class="pre">IndexEntry</span></code></a><ul>
<li><a class="reference internal" href="#pinder.core.index.utils.IndexEntry.model_config"><code class="docutils literal notranslate"><span class="pre">IndexEntry.model_config</span></code></a></li>
<li><a class="reference internal" href="#pinder.core.index.utils.IndexEntry.split"><code class="docutils literal notranslate"><span class="pre">IndexEntry.split</span></code></a></li>
<li><a class="reference internal" href="#pinder.core.index.utils.IndexEntry.id"><code class="docutils literal notranslate"><span class="pre">IndexEntry.id</span></code></a></li>
<li><a class="reference internal" href="#pinder.core.index.utils.IndexEntry.pdb_id"><code class="docutils literal notranslate"><span class="pre">IndexEntry.pdb_id</span></code></a></li>
<li><a class="reference internal" href="#pinder.core.index.utils.IndexEntry.cluster_id"><code class="docutils literal notranslate"><span class="pre">IndexEntry.cluster_id</span></code></a></li>
<li><a class="reference internal" href="#pinder.core.index.utils.IndexEntry.cluster_id_R"><code class="docutils literal notranslate"><span class="pre">IndexEntry.cluster_id_R</span></code></a></li>
<li><a class="reference internal" href="#pinder.core.index.utils.IndexEntry.cluster_id_L"><code class="docutils literal notranslate"><span class="pre">IndexEntry.cluster_id_L</span></code></a></li>
<li><a class="reference internal" href="#pinder.core.index.utils.IndexEntry.pinder_s"><code class="docutils literal notranslate"><span class="pre">IndexEntry.pinder_s</span></code></a></li>
<li><a class="reference internal" href="#pinder.core.index.utils.IndexEntry.pinder_xl"><code class="docutils literal notranslate"><span class="pre">IndexEntry.pinder_xl</span></code></a></li>
<li><a class="reference internal" href="#pinder.core.index.utils.IndexEntry.pinder_af2"><code class="docutils literal notranslate"><span class="pre">IndexEntry.pinder_af2</span></code></a></li>
<li><a class="reference internal" href="#pinder.core.index.utils.IndexEntry.uniprot_R"><code class="docutils literal notranslate"><span class="pre">IndexEntry.uniprot_R</span></code></a></li>
<li><a class="reference internal" href="#pinder.core.index.utils.IndexEntry.uniprot_L"><code class="docutils literal notranslate"><span class="pre">IndexEntry.uniprot_L</span></code></a></li>
<li><a class="reference internal" href="#pinder.core.index.utils.IndexEntry.holo_R_pdb"><code class="docutils literal notranslate"><span class="pre">IndexEntry.holo_R_pdb</span></code></a></li>
<li><a class="reference internal" href="#pinder.core.index.utils.IndexEntry.holo_L_pdb"><code class="docutils literal notranslate"><span class="pre">IndexEntry.holo_L_pdb</span></code></a></li>
<li><a class="reference internal" href="#pinder.core.index.utils.IndexEntry.predicted_R_pdb"><code class="docutils literal notranslate"><span class="pre">IndexEntry.predicted_R_pdb</span></code></a></li>
<li><a class="reference internal" href="#pinder.core.index.utils.IndexEntry.predicted_L_pdb"><code class="docutils literal notranslate"><span class="pre">IndexEntry.predicted_L_pdb</span></code></a></li>
<li><a class="reference internal" href="#pinder.core.index.utils.IndexEntry.apo_R_pdb"><code class="docutils literal notranslate"><span class="pre">IndexEntry.apo_R_pdb</span></code></a></li>
<li><a class="reference internal" href="#pinder.core.index.utils.IndexEntry.apo_L_pdb"><code class="docutils literal notranslate"><span class="pre">IndexEntry.apo_L_pdb</span></code></a></li>
<li><a class="reference internal" href="#pinder.core.index.utils.IndexEntry.apo_R_pdbs"><code class="docutils literal notranslate"><span class="pre">IndexEntry.apo_R_pdbs</span></code></a></li>
<li><a class="reference internal" href="#pinder.core.index.utils.IndexEntry.apo_L_pdbs"><code class="docutils literal notranslate"><span class="pre">IndexEntry.apo_L_pdbs</span></code></a></li>
<li><a class="reference internal" href="#pinder.core.index.utils.IndexEntry.holo_R"><code class="docutils literal notranslate"><span class="pre">IndexEntry.holo_R</span></code></a></li>
<li><a class="reference internal" href="#pinder.core.index.utils.IndexEntry.holo_L"><code class="docutils literal notranslate"><span class="pre">IndexEntry.holo_L</span></code></a></li>
<li><a class="reference internal" href="#pinder.core.index.utils.IndexEntry.predicted_R"><code class="docutils literal notranslate"><span class="pre">IndexEntry.predicted_R</span></code></a></li>
<li><a class="reference internal" href="#pinder.core.index.utils.IndexEntry.predicted_L"><code class="docutils literal notranslate"><span class="pre">IndexEntry.predicted_L</span></code></a></li>
<li><a class="reference internal" href="#pinder.core.index.utils.IndexEntry.apo_R"><code class="docutils literal notranslate"><span class="pre">IndexEntry.apo_R</span></code></a></li>
<li><a class="reference internal" href="#pinder.core.index.utils.IndexEntry.apo_L"><code class="docutils literal notranslate"><span class="pre">IndexEntry.apo_L</span></code></a></li>
<li><a class="reference internal" href="#pinder.core.index.utils.IndexEntry.apo_R_quality"><code class="docutils literal notranslate"><span class="pre">IndexEntry.apo_R_quality</span></code></a></li>
<li><a class="reference internal" href="#pinder.core.index.utils.IndexEntry.apo_L_quality"><code class="docutils literal notranslate"><span class="pre">IndexEntry.apo_L_quality</span></code></a></li>
<li><a class="reference internal" href="#pinder.core.index.utils.IndexEntry.chain1_neff"><code class="docutils literal notranslate"><span class="pre">IndexEntry.chain1_neff</span></code></a></li>
<li><a class="reference internal" href="#pinder.core.index.utils.IndexEntry.chain2_neff"><code class="docutils literal notranslate"><span class="pre">IndexEntry.chain2_neff</span></code></a></li>
<li><a class="reference internal" href="#pinder.core.index.utils.IndexEntry.chain_R"><code class="docutils literal notranslate"><span class="pre">IndexEntry.chain_R</span></code></a></li>
<li><a class="reference internal" href="#pinder.core.index.utils.IndexEntry.chain_L"><code class="docutils literal notranslate"><span class="pre">IndexEntry.chain_L</span></code></a></li>
<li><a class="reference internal" href="#pinder.core.index.utils.IndexEntry.contains_antibody"><code class="docutils literal notranslate"><span class="pre">IndexEntry.contains_antibody</span></code></a></li>
<li><a class="reference internal" href="#pinder.core.index.utils.IndexEntry.contains_antigen"><code class="docutils literal notranslate"><span class="pre">IndexEntry.contains_antigen</span></code></a></li>
<li><a class="reference internal" href="#pinder.core.index.utils.IndexEntry.contains_enzyme"><code class="docutils literal notranslate"><span class="pre">IndexEntry.contains_enzyme</span></code></a></li>
<li><a class="reference internal" href="#pinder.core.index.utils.IndexEntry.pdb_path"><code class="docutils literal notranslate"><span class="pre">IndexEntry.pdb_path()</span></code></a></li>
<li><a class="reference internal" href="#pinder.core.index.utils.IndexEntry.mapping_path"><code class="docutils literal notranslate"><span class="pre">IndexEntry.mapping_path()</span></code></a></li>
<li><a class="reference internal" href="#pinder.core.index.utils.IndexEntry.pdb_paths"><code class="docutils literal notranslate"><span class="pre">IndexEntry.pdb_paths</span></code></a></li>
<li><a class="reference internal" href="#pinder.core.index.utils.IndexEntry.mapping_paths"><code class="docutils literal notranslate"><span class="pre">IndexEntry.mapping_paths</span></code></a></li>
<li><a class="reference internal" href="#pinder.core.index.utils.IndexEntry.pinder_id"><code class="docutils literal notranslate"><span class="pre">IndexEntry.pinder_id</span></code></a></li>
<li><a class="reference internal" href="#pinder.core.index.utils.IndexEntry.pinder_pdb"><code class="docutils literal notranslate"><span class="pre">IndexEntry.pinder_pdb</span></code></a></li>
<li><a class="reference internal" href="#pinder.core.index.utils.IndexEntry.apo_R_alt"><code class="docutils literal notranslate"><span class="pre">IndexEntry.apo_R_alt</span></code></a></li>
<li><a class="reference internal" href="#pinder.core.index.utils.IndexEntry.apo_L_alt"><code class="docutils literal notranslate"><span class="pre">IndexEntry.apo_L_alt</span></code></a></li>
<li><a class="reference internal" href="#pinder.core.index.utils.IndexEntry.homodimer"><code class="docutils literal notranslate"><span class="pre">IndexEntry.homodimer</span></code></a></li>
<li><a class="reference internal" href="#pinder.core.index.utils.IndexEntry.test_system"><code class="docutils literal notranslate"><span class="pre">IndexEntry.test_system</span></code></a></li>
<li><a class="reference internal" href="#pinder.core.index.utils.IndexEntry.model_computed_fields"><code class="docutils literal notranslate"><span class="pre">IndexEntry.model_computed_fields</span></code></a></li>
<li><a class="reference internal" href="#pinder.core.index.utils.IndexEntry.model_fields"><code class="docutils literal notranslate"><span class="pre">IndexEntry.model_fields</span></code></a></li>
</ul>
</li>
<li><a class="reference internal" href="#pinder.core.index.utils.MetadataEntry"><code class="docutils literal notranslate"><span class="pre">MetadataEntry</span></code></a><ul>
<li><a class="reference internal" href="#pinder.core.index.utils.MetadataEntry.model_config"><code class="docutils literal notranslate"><span class="pre">MetadataEntry.model_config</span></code></a></li>
<li><a class="reference internal" href="#pinder.core.index.utils.MetadataEntry.id"><code class="docutils literal notranslate"><span class="pre">MetadataEntry.id</span></code></a></li>
<li><a class="reference internal" href="#pinder.core.index.utils.MetadataEntry.entry_id"><code class="docutils literal notranslate"><span class="pre">MetadataEntry.entry_id</span></code></a></li>
<li><a class="reference internal" href="#pinder.core.index.utils.MetadataEntry.method"><code class="docutils literal notranslate"><span class="pre">MetadataEntry.method</span></code></a></li>
<li><a class="reference internal" href="#pinder.core.index.utils.MetadataEntry.date"><code class="docutils literal notranslate"><span class="pre">MetadataEntry.date</span></code></a></li>
<li><a class="reference internal" href="#pinder.core.index.utils.MetadataEntry.release_date"><code class="docutils literal notranslate"><span class="pre">MetadataEntry.release_date</span></code></a></li>
<li><a class="reference internal" href="#pinder.core.index.utils.MetadataEntry.resolution"><code class="docutils literal notranslate"><span class="pre">MetadataEntry.resolution</span></code></a></li>
<li><a class="reference internal" href="#pinder.core.index.utils.MetadataEntry.label"><code class="docutils literal notranslate"><span class="pre">MetadataEntry.label</span></code></a></li>
<li><a class="reference internal" href="#pinder.core.index.utils.MetadataEntry.probability"><code class="docutils literal notranslate"><span class="pre">MetadataEntry.probability</span></code></a></li>
<li><a class="reference internal" href="#pinder.core.index.utils.MetadataEntry.chain1_id"><code class="docutils literal notranslate"><span class="pre">MetadataEntry.chain1_id</span></code></a></li>
<li><a class="reference internal" href="#pinder.core.index.utils.MetadataEntry.chain2_id"><code class="docutils literal notranslate"><span class="pre">MetadataEntry.chain2_id</span></code></a></li>
<li><a class="reference internal" href="#pinder.core.index.utils.MetadataEntry.assembly"><code class="docutils literal notranslate"><span class="pre">MetadataEntry.assembly</span></code></a></li>
<li><a class="reference internal" href="#pinder.core.index.utils.MetadataEntry.assembly_details"><code class="docutils literal notranslate"><span class="pre">MetadataEntry.assembly_details</span></code></a></li>
<li><a class="reference internal" href="#pinder.core.index.utils.MetadataEntry.oligomeric_details"><code class="docutils literal notranslate"><span class="pre">MetadataEntry.oligomeric_details</span></code></a></li>
<li><a class="reference internal" href="#pinder.core.index.utils.MetadataEntry.oligomeric_count"><code class="docutils literal notranslate"><span class="pre">MetadataEntry.oligomeric_count</span></code></a></li>
<li><a class="reference internal" href="#pinder.core.index.utils.MetadataEntry.biol_details"><code class="docutils literal notranslate"><span class="pre">MetadataEntry.biol_details</span></code></a></li>
<li><a class="reference internal" href="#pinder.core.index.utils.MetadataEntry.complex_type"><code class="docutils literal notranslate"><span class="pre">MetadataEntry.complex_type</span></code></a></li>
<li><a class="reference internal" href="#pinder.core.index.utils.MetadataEntry.chain_1"><code class="docutils literal notranslate"><span class="pre">MetadataEntry.chain_1</span></code></a></li>
<li><a class="reference internal" href="#pinder.core.index.utils.MetadataEntry.asym_id_1"><code class="docutils literal notranslate"><span class="pre">MetadataEntry.asym_id_1</span></code></a></li>
<li><a class="reference internal" href="#pinder.core.index.utils.MetadataEntry.chain_2"><code class="docutils literal notranslate"><span class="pre">MetadataEntry.chain_2</span></code></a></li>
<li><a class="reference internal" href="#pinder.core.index.utils.MetadataEntry.asym_id_2"><code class="docutils literal notranslate"><span class="pre">MetadataEntry.asym_id_2</span></code></a></li>
<li><a class="reference internal" href="#pinder.core.index.utils.MetadataEntry.length1"><code class="docutils literal notranslate"><span class="pre">MetadataEntry.length1</span></code></a></li>
<li><a class="reference internal" href="#pinder.core.index.utils.MetadataEntry.length2"><code class="docutils literal notranslate"><span class="pre">MetadataEntry.length2</span></code></a></li>
<li><a class="reference internal" href="#pinder.core.index.utils.MetadataEntry.length_resolved_1"><code class="docutils literal notranslate"><span class="pre">MetadataEntry.length_resolved_1</span></code></a></li>
<li><a class="reference internal" href="#pinder.core.index.utils.MetadataEntry.length_resolved_2"><code class="docutils literal notranslate"><span class="pre">MetadataEntry.length_resolved_2</span></code></a></li>
<li><a class="reference internal" href="#pinder.core.index.utils.MetadataEntry.number_of_components_1"><code class="docutils literal notranslate"><span class="pre">MetadataEntry.number_of_components_1</span></code></a></li>
<li><a class="reference internal" href="#pinder.core.index.utils.MetadataEntry.number_of_components_2"><code class="docutils literal notranslate"><span class="pre">MetadataEntry.number_of_components_2</span></code></a></li>
<li><a class="reference internal" href="#pinder.core.index.utils.MetadataEntry.link_density"><code class="docutils literal notranslate"><span class="pre">MetadataEntry.link_density</span></code></a></li>
<li><a class="reference internal" href="#pinder.core.index.utils.MetadataEntry.planarity"><code class="docutils literal notranslate"><span class="pre">MetadataEntry.planarity</span></code></a></li>
<li><a class="reference internal" href="#pinder.core.index.utils.MetadataEntry.max_var_1"><code class="docutils literal notranslate"><span class="pre">MetadataEntry.max_var_1</span></code></a></li>
<li><a class="reference internal" href="#pinder.core.index.utils.MetadataEntry.max_var_2"><code class="docutils literal notranslate"><span class="pre">MetadataEntry.max_var_2</span></code></a></li>
<li><a class="reference internal" href="#pinder.core.index.utils.MetadataEntry.num_atom_types"><code class="docutils literal notranslate"><span class="pre">MetadataEntry.num_atom_types</span></code></a></li>
<li><a class="reference internal" href="#pinder.core.index.utils.MetadataEntry.n_residue_pairs"><code class="docutils literal notranslate"><span class="pre">MetadataEntry.n_residue_pairs</span></code></a></li>
<li><a class="reference internal" href="#pinder.core.index.utils.MetadataEntry.n_residues"><code class="docutils literal notranslate"><span class="pre">MetadataEntry.n_residues</span></code></a></li>
<li><a class="reference internal" href="#pinder.core.index.utils.MetadataEntry.buried_sasa"><code class="docutils literal notranslate"><span class="pre">MetadataEntry.buried_sasa</span></code></a></li>
<li><a class="reference internal" href="#pinder.core.index.utils.MetadataEntry.intermolecular_contacts"><code class="docutils literal notranslate"><span class="pre">MetadataEntry.intermolecular_contacts</span></code></a></li>
<li><a class="reference internal" href="#pinder.core.index.utils.MetadataEntry.charged_charged_contacts"><code class="docutils literal notranslate"><span class="pre">MetadataEntry.charged_charged_contacts</span></code></a></li>
<li><a class="reference internal" href="#pinder.core.index.utils.MetadataEntry.charged_polar_contacts"><code class="docutils literal notranslate"><span class="pre">MetadataEntry.charged_polar_contacts</span></code></a></li>
<li><a class="reference internal" href="#pinder.core.index.utils.MetadataEntry.charged_apolar_contacts"><code class="docutils literal notranslate"><span class="pre">MetadataEntry.charged_apolar_contacts</span></code></a></li>
<li><a class="reference internal" href="#pinder.core.index.utils.MetadataEntry.polar_polar_contacts"><code class="docutils literal notranslate"><span class="pre">MetadataEntry.polar_polar_contacts</span></code></a></li>
<li><a class="reference internal" href="#pinder.core.index.utils.MetadataEntry.apolar_polar_contacts"><code class="docutils literal notranslate"><span class="pre">MetadataEntry.apolar_polar_contacts</span></code></a></li>
<li><a class="reference internal" href="#pinder.core.index.utils.MetadataEntry.apolar_apolar_contacts"><code class="docutils literal notranslate"><span class="pre">MetadataEntry.apolar_apolar_contacts</span></code></a></li>
<li><a class="reference internal" href="#pinder.core.index.utils.MetadataEntry.interface_atom_gaps_4A"><code class="docutils literal notranslate"><span class="pre">MetadataEntry.interface_atom_gaps_4A</span></code></a></li>
<li><a class="reference internal" href="#pinder.core.index.utils.MetadataEntry.missing_interface_residues_4A"><code class="docutils literal notranslate"><span class="pre">MetadataEntry.missing_interface_residues_4A</span></code></a></li>
<li><a class="reference internal" href="#pinder.core.index.utils.MetadataEntry.interface_atom_gaps_8A"><code class="docutils literal notranslate"><span class="pre">MetadataEntry.interface_atom_gaps_8A</span></code></a></li>
<li><a class="reference internal" href="#pinder.core.index.utils.MetadataEntry.missing_interface_residues_8A"><code class="docutils literal notranslate"><span class="pre">MetadataEntry.missing_interface_residues_8A</span></code></a></li>
<li><a class="reference internal" href="#pinder.core.index.utils.MetadataEntry.entity_id_R"><code class="docutils literal notranslate"><span class="pre">MetadataEntry.entity_id_R</span></code></a></li>
<li><a class="reference internal" href="#pinder.core.index.utils.MetadataEntry.entity_id_L"><code class="docutils literal notranslate"><span class="pre">MetadataEntry.entity_id_L</span></code></a></li>
<li><a class="reference internal" href="#pinder.core.index.utils.MetadataEntry.pdb_strand_id_R"><code class="docutils literal notranslate"><span class="pre">MetadataEntry.pdb_strand_id_R</span></code></a></li>
<li><a class="reference internal" href="#pinder.core.index.utils.MetadataEntry.pdb_strand_id_L"><code class="docutils literal notranslate"><span class="pre">MetadataEntry.pdb_strand_id_L</span></code></a></li>
<li><a class="reference internal" href="#pinder.core.index.utils.MetadataEntry.ECOD_names_R"><code class="docutils literal notranslate"><span class="pre">MetadataEntry.ECOD_names_R</span></code></a></li>
<li><a class="reference internal" href="#pinder.core.index.utils.MetadataEntry.ECOD_names_L"><code class="docutils literal notranslate"><span class="pre">MetadataEntry.ECOD_names_L</span></code></a></li>
<li><a class="reference internal" href="#pinder.core.index.utils.MetadataEntry.pinder_id"><code class="docutils literal notranslate"><span class="pre">MetadataEntry.pinder_id</span></code></a></li>
<li><a class="reference internal" href="#pinder.core.index.utils.MetadataEntry.model_computed_fields"><code class="docutils literal notranslate"><span class="pre">MetadataEntry.model_computed_fields</span></code></a></li>
<li><a class="reference internal" href="#pinder.core.index.utils.MetadataEntry.model_fields"><code class="docutils literal notranslate"><span class="pre">MetadataEntry.model_fields</span></code></a></li>
</ul>
</li>
<li><a class="reference internal" href="#pinder.core.index.utils.fix_index"><code class="docutils literal notranslate"><span class="pre">fix_index()</span></code></a></li>
<li><a class="reference internal" href="#pinder.core.index.utils.fix_metadata"><code class="docutils literal notranslate"><span class="pre">fix_metadata()</span></code></a></li>
<li><a class="reference internal" href="#pinder.core.index.utils.downcast_dtypes"><code class="docutils literal notranslate"><span class="pre">downcast_dtypes()</span></code></a></li>
<li><a class="reference internal" href="#pinder.core.index.utils.set_mapping_column_types"><code class="docutils literal notranslate"><span class="pre">set_mapping_column_types()</span></code></a></li>
</ul>
</li>
<li><a class="reference internal" href="#module-pinder.core.index">Module contents</a></li>
</ul>
</li>
</ul>

            </div>
        </div>
    

      <div class="clearer"></div>
    </div>
    <div class="button_nav_wrapper">
        <div class="button_nav">
            <div class="left">
                
                <a href="pinder.core.html">
                    <span class="icon">&lt;</span><span>pinder.core package</span></a>
                
            </div>

            <div class="right">
                
                    <a href="pinder.core.loader.html"><span>pinder.core.loader package</span><span class="icon">&gt;</span></a>
                
            </div>
        </div>
    </div>


    <div class="footer" role="contentinfo">
    &#169; Copyright 2023, VantAI.
      Created using <a href="https://www.sphinx-doc.org/">Sphinx</a> 7.4.7.
    </div>

<p id="theme_credit"></p>
  </body>
</html>